// react-crossword/lib/index.js
module.exports = function (t)
{
	var e = {};

	function n(r)
	{
		if (e[r]) return e[r].exports;
		var o = e[r] = {
			i: r,
			l: !1,
			exports:
			{}
		};
		return t[r].call(o.exports, o, o.exports, n), o.l = !0, o.exports
	}
	return n.m = t, n.c = e, n.d = function (t, e, r)
	{
		n.o(t, e) || Object.defineProperty(t, e,
		{
			configurable: !1,
			enumerable: !0,
			get: r
		})
	}, n.r = function (t)
	{
		Object.defineProperty(t, "__esModule",
		{
			value: !0
		})
	}, n.n = function (t)
	{
		var e = t && t.__esModule ? function ()
		{
			return t.default
		} : function ()
		{
			return t
		};
		return n.d(e, "a", e), e
	}, n.o = function (t, e)
	{
		return Object.prototype.hasOwnProperty.call(t, e)
	}, n.p = "", n(n.s = 110)
}([function (t, e)
{
	t.exports = require("react")
}, function (t, e, n)
{
	"use strict";
	e.__esModule = !0;
	var r, o = n(102),
		i = (r = o) && r.__esModule ? r :
		{
			default: r
		};
	e.default = function ()
	{
		function t(t, e)
		{
			for (var n = 0; n < e.length; n++)
			{
				var r = e[n];
				r.enumerable = r.enumerable || !1, r.configurable = !0, "value" in r && (r.writable = !0), (0, i.default)(t, r.key, r)
			}
		}
		return function (e, n, r)
		{
			return n && t(e.prototype, n), r && t(e, r), e
		}
	}()
}, function (t, e, n)
{
	"use strict";
	e.__esModule = !0, e.default = function (t, e)
	{
		if (!(t instanceof e)) throw new TypeError("Cannot call a class as a function")
	}
}, function (t, e, n)
{
	"use strict";
	e.__esModule = !0;
	var r = s(n(234)),
		o = s(n(230)),
		i = s(n(99));

	function s(t)
	{
		return t && t.__esModule ? t :
		{
			default: t
		}
	}
	e.default = function (t, e)
	{
		if ("function" != typeof e && null !== e) throw new TypeError("Super expression must either be null or a function, not " + (void 0 === e ? "undefined" : (0, i.default)(e)));
		t.prototype = (0, o.default)(e && e.prototype,
		{
			constructor:
			{
				value: t,
				enumerable: !1,
				writable: !0,
				configurable: !0
			}
		}), e && (r.default ? (0, r.default)(t, e) : t.__proto__ = e)
	}
}, function (t, e, n)
{
	"use strict";
	e.__esModule = !0;
	var r, o = n(99),
		i = (r = o) && r.__esModule ? r :
		{
			default: r
		};
	e.default = function (t, e)
	{
		if (!t) throw new ReferenceError("this hasn't been initialised - super() hasn't been called");
		return !e || "object" !== (void 0 === e ? "undefined" : (0, i.default)(e)) && "function" != typeof e ? t : e
	}
}, function (t, e)
{
	t.exports = require("react-dom")
}, function (t, e)
{
	var n = Array.isArray;
	t.exports = n
}, function (t, e, n)
{
	var r = n(88),
		o = "object" == typeof self && self && self.Object === Object && self,
		i = r || o || Function("return this")();
	t.exports = i
}, function (t, e, n)
{
	var r = n(55)("wks"),
		o = n(35),
		i = n(12).Symbol,
		s = "function" == typeof i;
	(t.exports = function (t)
	{
		return r[t] || (r[t] = s && i[t] || (s ? i : o)("Symbol." + t))
	}).store = r
}, function (t, e, n)
{
	var r, o, i;
	i = function ()
	{
		var t, e, n, r, o = window,
			i = o.document,
			s = i.documentElement,
			a = /^(checked|value|selected|disabled)$/i,
			c = /^(select|fieldset|table|tbody|tfoot|td|tr|colgroup)$/i,
			u = /\s*<script +src=['"]([^'"]+)['"]>/,
			l = ["<table>", "</table>", 1],
			f = ["<table><tbody><tr>", "</tr></tbody></table>", 3],
			d = ["<select>", "</select>", 1],
			p = ["_", "", 0, 1],
			h = {
				thead: l,
				tbody: l,
				tfoot: l,
				colgroup: l,
				caption: l,
				tr: ["<table><tbody>", "</tbody></table>", 2],
				th: f,
				td: f,
				col: ["<table><colgroup>", "</colgroup></table>", 2],
				fieldset: ["<form>", "</form>", 1],
				legend: ["<form><fieldset>", "</fieldset></form>", 2],
				option: d,
				optgroup: d,
				script: p,
				style: p,
				link: p,
				param: p,
				base: p
			},
			v = /^(checked|selected|disabled)$/,
			_ = {},
			m = 0,
			g = /^-?[\d\.]+$/,
			y = /^data-(.+)$/,
			w = "px",
			b = "setAttribute",
			x = "getAttribute",
			k = (r = i.createElement("p"),
			{
				transform: function ()
				{
					var t, e = ["transform", "webkitTransform", "MozTransform", "OTransform", "msTransform"];
					for (t = 0; t < e.length; t++)
						if (e[t] in r.style) return e[t]
				}(),
				classList: "classList" in r
			}),
			E = /\s+/,
			O = String.prototype.toString,
			C = {
				lineHeight: 1,
				zoom: 1,
				zIndex: 1,
				opacity: 1,
				boxFlex: 1,
				WebkitBoxFlex: 1,
				MozBoxFlex: 1
			},
			j = i.querySelectorAll && function (t)
			{
				return i.querySelectorAll(t)
			};

		function S(t)
		{
			return t && t.nodeName && (1 == t.nodeType || 11 == t.nodeType)
		}

		function N(t, e, n)
		{
			var r, o, i;
			if ("string" == typeof t) return U.create(t);
			if (S(t) && (t = [t]), n)
			{
				for (i = [], r = 0, o = t.length; r < o; r++) i[r] = q(e, t[r]);
				return i
			}
			return t
		}

		function I(t)
		{
			return new RegExp("(^|\\s+)" + t + "(\\s+|$)")
		}

		function A(t, e, n, r)
		{
			for (var o, i = 0, s = t.length; i < s; i++) o = r ? t.length - i - 1 : i, e.call(n || t[o], t[o], o, t);
			return t
		}

		function T(t, e, n)
		{
			for (var r = 0, o = t.length; r < o; r++) S(t[r]) && (T(t[r].childNodes, e, n), e.call(n || t[r], t[r], r, t));
			return t
		}

		function L(t)
		{
			return t.replace(/-(.)/g, function (t, e)
			{
				return e.toUpperCase()
			})
		}

		function P(t)
		{
			t[x]("data-node-uid") || t[b]("data-node-uid", ++m);
			var e = t[x]("data-node-uid");
			return _[e] || (_[e] = {})
		}

		function M(t)
		{
			var e = t[x]("data-node-uid");
			e && delete _[e]
		}

		function F(t)
		{
			var e;
			try
			{
				return null === t || void 0 === t ? void 0 : "true" === t || "false" !== t && ("null" === t ? null : (e = parseFloat(t)) == t ? e : t)
			}
			catch (t)
			{}
		}

		function D(t, e, n)
		{
			for (var r = 0, o = t.length; r < o; ++r)
				if (e.call(n || null, t[r], r, t)) return !0;
			return !1
		}

		function z(t)
		{
			return "transform" == t && (t = k.transform) || /^transform-?[Oo]rigin$/.test(t) && (t = k.transform + "Origin"), t ? L(t) : null
		}

		function H(t, e, n, r)
		{
			var o = 0,
				i = e || this,
				s = [];
			return A(N(j && "string" == typeof t && "<" != t.charAt(0) ? j(t) : t), function (t, e)
			{
				A(i, function (r)
				{
					n(t, s[o++] = e > 0 ? q(i, r) : r)
				}, null, r)
			}, this, r), i.length = o, A(s, function (t)
			{
				i[--o] = t
			}, null, !r), i
		}

		function $(t, e, n)
		{
			var r = U(t),
				o = r.css("position"),
				i = r.offset(),
				s = "relative",
				a = o == s,
				c = [parseInt(r.css("left"), 10), parseInt(r.css("top"), 10)];
			"static" == o && (r.css("position", s), o = s), isNaN(c[0]) && (c[0] = a ? 0 : t.offsetLeft), isNaN(c[1]) && (c[1] = a ? 0 : t.offsetTop), null != e && (t.style.left = e - i.left + c[0] + w), null != n && (t.style.top = n - i.top + c[1] + w)
		}

		function R(t, e)
		{
			return "function" == typeof e ? e.call(t, t) : e
		}

		function B(t, e, n)
		{
			var r = this[0];
			return r ? null == t && null == e ? (V(r) ? W() :
			{
				x: r.scrollLeft,
				y: r.scrollTop
			})[n] : (V(r) ? o.scrollTo(t, e) : (null != t && (r.scrollLeft = t), null != e && (r.scrollTop = e)), this) : this
		}

		function G(t)
		{
			if (this.length = 0, t)
			{
				t = "string" == typeof t || t.nodeType || void 0 === t.length ? [t] : t, this.length = t.length;
				for (var e = 0; e < t.length; e++) this[e] = t[e]
			}
		}

		function q(t, e)
		{
			var n, r, o, i = e.cloneNode(!0);
			if (t.$ && "function" == typeof t.cloneEvents)
				for (t.$(i).cloneEvents(e), n = t.$(i).find("*"), r = t.$(e).find("*"), o = 0; o < r.length; o++) t.$(n[o]).cloneEvents(r[o]);
			return i
		}

		function V(t)
		{
			return t === o || /^(?:body|html)$/i.test(t.tagName)
		}

		function W()
		{
			return {
				x: o.pageXOffset || s.scrollLeft,
				y: o.pageYOffset || s.scrollTop
			}
		}

		function U(t)
		{
			return new G(t)
		}
		return k.classList ? (t = function (t, e)
		{
			return t.classList.contains(e)
		}, e = function (t, e)
		{
			t.classList.add(e)
		}, n = function (t, e)
		{
			t.classList.remove(e)
		}) : (t = function (t, e)
		{
			return I(e).test(t.className)
		}, e = function (t, e)
		{
			t.className = (t.className + " " + e).trim()
		}, n = function (t, e)
		{
			t.className = t.className.replace(I(e), " ").trim()
		}), G.prototype = {
			get: function (t)
			{
				return this[t] || null
			},
			each: function (t, e)
			{
				return A(this, t, e)
			},
			deepEach: function (t, e)
			{
				return T(this, t, e)
			},
			map: function (t, e)
			{
				var n, r, o = [];
				for (r = 0; r < this.length; r++) n = t.call(this, this[r], r), e ? e(n) && o.push(n) : o.push(n);
				return o
			},
			html: function (t, e)
			{
				var n = e ? "textContent" : "innerHTML",
					r = this;
				return void 0 !== t ? this.empty().each(function (o, i)
				{
					try
					{
						if (e || "string" == typeof t && !c.test(o.tagName)) return o[n] = t
					}
					catch (t)
					{}! function (e, n)
					{
						A(N(t, r, n), function (t)
						{
							e.appendChild(t)
						})
					}(o, i)
				}) : this[0] ? this[0][n] : ""
			},
			text: function (t)
			{
				return this.html(t, !0)
			},
			append: function (t)
			{
				var e = this;
				return this.each(function (n, r)
				{
					A(N(t, e, r), function (t)
					{
						n.appendChild(t)
					})
				})
			},
			prepend: function (t)
			{
				var e = this;
				return this.each(function (n, r)
				{
					var o = n.firstChild;
					A(N(t, e, r), function (t)
					{
						n.insertBefore(t, o)
					})
				})
			},
			appendTo: function (t, e)
			{
				return H.call(this, t, e, function (t, e)
				{
					t.appendChild(e)
				})
			},
			prependTo: function (t, e)
			{
				return H.call(this, t, e, function (t, e)
				{
					t.insertBefore(e, t.firstChild)
				}, 1)
			},
			before: function (t)
			{
				var e = this;
				return this.each(function (n, r)
				{
					A(N(t, e, r), function (t)
					{
						n.parentNode.insertBefore(t, n)
					})
				})
			},
			after: function (t)
			{
				var e = this;
				return this.each(function (n, r)
				{
					A(N(t, e, r), function (t)
					{
						n.parentNode.insertBefore(t, n.nextSibling)
					}, null, 1)
				})
			},
			insertBefore: function (t, e)
			{
				return H.call(this, t, e, function (t, e)
				{
					t.parentNode.insertBefore(e, t)
				})
			},
			insertAfter: function (t, e)
			{
				return H.call(this, t, e, function (t, e)
				{
					var n = t.nextSibling;
					n ? t.parentNode.insertBefore(e, n) : t.parentNode.appendChild(e)
				}, 1)
			},
			replaceWith: function (t)
			{
				var e = this;
				return this.each(function (n, r)
				{
					A(N(t, e, r), function (t)
					{
						n.parentNode && n.parentNode.replaceChild(t, n)
					})
				})
			},
			clone: function (t)
			{
				var e, n, r = [];
				for (n = 0, e = this.length; n < e; n++) r[n] = q(t || this, this[n]);
				return U(r)
			},
			addClass: function (n)
			{
				return n = O.call(n).split(E), this.each(function (r)
				{
					A(n, function (n)
					{
						n && !t(r, R(r, n)) && e(r, R(r, n))
					})
				})
			},
			removeClass: function (e)
			{
				return e = O.call(e).split(E), this.each(function (r)
				{
					A(e, function (e)
					{
						e && t(r, R(r, e)) && n(r, R(r, e))
					})
				})
			},
			hasClass: function (e)
			{
				return e = O.call(e).split(E), D(this, function (n)
				{
					return D(e, function (e)
					{
						return e && t(n, e)
					})
				})
			},
			toggleClass: function (r, o)
			{
				return r = O.call(r).split(E), this.each(function (i)
				{
					A(r, function (r)
					{
						r && (void 0 !== o ? o ? !t(i, r) && e(i, r) : n(i, r) : t(i, r) ? n(i, r) : e(i, r))
					})
				})
			},
			show: function (t)
			{
				return t = "string" == typeof t ? t : "", this.each(function (e)
				{
					e.style.display = t
				})
			},
			hide: function ()
			{
				return this.each(function (t)
				{
					t.style.display = "none"
				})
			},
			toggle: function (t, e)
			{
				return e = "string" == typeof e ? e : "", "function" != typeof t && (t = null), this.each(function (n)
				{
					n.style.display = n.offsetWidth || n.offsetHeight ? "none" : e, t && t.call(n)
				})
			},
			first: function ()
			{
				return U(this.length ? this[0] : [])
			},
			last: function ()
			{
				return U(this.length ? this[this.length - 1] : [])
			},
			next: function ()
			{
				return this.related("nextSibling")
			},
			previous: function ()
			{
				return this.related("previousSibling")
			},
			parent: function ()
			{
				return this.related("parentNode")
			},
			related: function (t)
			{
				return U(this.map(function (e)
				{
					for (e = e[t]; e && 1 !== e.nodeType;) e = e[t];
					return e || 0
				}, function (t)
				{
					return t
				}))
			},
			focus: function ()
			{
				return this.length && this[0].focus(), this
			},
			blur: function ()
			{
				return this.length && this[0].blur(), this
			},
			css: function (t, e)
			{
				var n, r, s, a, c, u = t;
				if (void 0 === e && "string" == typeof t) return (e = this[0]) ? e === i || e === o ? (n = e === i ? U.doc() : U.viewport(), "width" == t ? n.width : "height" == t ? n.height : "") : (t = z(t)) ? (r = e, s = t, a = null, (c = i.defaultView.getComputedStyle(r, "")) && (a = c[s]), r.style[s] || a) : null : null;
				return "string" == typeof t && ((u = {})[t] = e), this.each(function (t, e, n)
				{
					for (var r in u)
						if (u.hasOwnProperty(r))
						{
							n = u[r], (e = z(r)) && g.test(n) && !(e in C) && (n += w);
							try
							{
								t.style[e] = R(t, n)
							}
							catch (t)
							{}
						}
				})
			},
			offset: function (t, e)
			{
				if (t && "object" == typeof t && ("number" == typeof t.top || "number" == typeof t.left)) return this.each(function (e)
				{
					$(e, t.left, t.top)
				});
				if ("number" == typeof t || "number" == typeof e) return this.each(function (n)
				{
					$(n, t, e)
				});
				if (!this[0]) return {
					top: 0,
					left: 0,
					height: 0,
					width: 0
				};
				var n = this[0],
					r = n.ownerDocument.documentElement,
					o = n.getBoundingClientRect(),
					s = W(),
					a = n.offsetWidth,
					c = n.offsetHeight;
				return {
					top: o.top + s.y - Math.max(0, r && r.clientTop, i.body.clientTop),
					left: o.left + s.x - Math.max(0, r && r.clientLeft, i.body.clientLeft),
					height: c,
					width: a
				}
			},
			dim: function ()
			{
				if (!this.length) return {
					height: 0,
					width: 0
				};
				var t, e, n = this[0],
					r = 9 == n.nodeType && n.documentElement,
					o = r || !n.style || n.offsetWidth || n.offsetHeight ? null : (t = this, e = {
						position: n.style.position || "",
						visibility: n.style.visibility || "",
						display: n.style.display || ""
					}, t.first().css(
					{
						position: "absolute",
						visibility: "hidden",
						display: "block"
					}), e),
					i = r ? Math.max(n.body.scrollWidth, n.body.offsetWidth, r.scrollWidth, r.offsetWidth, r.clientWidth) : n.offsetWidth,
					s = r ? Math.max(n.body.scrollHeight, n.body.offsetHeight, r.scrollHeight, r.offsetHeight, r.clientHeight) : n.offsetHeight;
				return o && this.first().css(o),
				{
					height: s,
					width: i
				}
			},
			attr: function (t, e)
			{
				var n, r = this[0];
				if ("string" != typeof t && !(t instanceof String))
				{
					for (n in t) t.hasOwnProperty(n) && this.attr(n, t[n]);
					return this
				}
				return void 0 === e ? r ? a.test(t) ? !(!v.test(t) || "string" != typeof r[t]) || r[t] : r[x](t) : null : this.each(function (n)
				{
					a.test(t) ? n[t] = R(n, e) : n[b](t, R(n, e))
				})
			},
			removeAttr: function (t)
			{
				return this.each(function (e)
				{
					v.test(t) ? e[t] = !1 : e.removeAttribute(t)
				})
			},
			val: function (t)
			{
				return "string" == typeof t || "number" == typeof t ? this.attr("value", t) : this.length ? this[0].value : null
			},
			data: function (t, e)
			{
				var n, r, o, i = this[0];
				return void 0 === e ? i ? (n = P(i), void 0 === t ? (A(i.attributes, function (t)
				{
					(r = ("" + t.name).match(y)) && (n[L(r[1])] = F(t.value))
				}), n) : (void 0 === n[t] && (n[t] = F(this.attr("data-" + ((o = t) ? o.replace(/([a-z])([A-Z])/g, "$1-$2").toLowerCase() : o)))), n[t])) : null : this.each(function (n)
				{
					P(n)[t] = e
				})
			},
			remove: function ()
			{
				return this.deepEach(M), this.detach()
			},
			empty: function ()
			{
				return this.each(function (t)
				{
					for (T(t.childNodes, M); t.firstChild;) t.removeChild(t.firstChild)
				})
			},
			detach: function ()
			{
				return this.each(function (t)
				{
					t.parentNode && t.parentNode.removeChild(t)
				})
			},
			scrollTop: function (t)
			{
				return B.call(this, null, t, "y")
			},
			scrollLeft: function (t)
			{
				return B.call(this, t, null, "x")
			}
		}, U.setQueryEngine = function (t)
		{
			j = t, delete U.setQueryEngine
		}, U.aug = function (t, e)
		{
			for (var n in t) t.hasOwnProperty(n) && ((e || G.prototype)[n] = t[n])
		}, U.create = function (t)
		{
			return "string" == typeof t && "" !== t ? function ()
			{
				if (u.test(t)) return [(e = t, n = document.createElement("script"), r = e.match(u), n.src = r[1], n)];
				var e, n, r, o = t.match(/^\s*<([^\s>]+)/),
					s = i.createElement("div"),
					a = [],
					c = o ? h[o[1].toLowerCase()] : null,
					l = c ? c[2] + 1 : 1,
					f = c && c[3],
					d = "parentNode";
				for (s.innerHTML = c ? c[0] + t + c[1] : t; l--;) s = s.firstChild;
				f && s && 1 !== s.nodeType && (s = s.nextSibling);
				do {
					o && 1 != s.nodeType || a.push(s)
				} while (s = s.nextSibling);
				return A(a, function (t)
				{
					t[d] && t[d].removeChild(t)
				}), a
			}() : S(t) ? [t.cloneNode(!0)] : []
		}, U.doc = function ()
		{
			var t = U.viewport();
			return {
				width: Math.max(i.body.scrollWidth, s.scrollWidth, t.width),
				height: Math.max(i.body.scrollHeight, s.scrollHeight, t.height)
			}
		}, U.firstChild = function (t)
		{
			for (var e, n = t.childNodes, r = 0, o = n && n.length || 0; r < o; r++) 1 === n[r].nodeType && (e = n[o = r]);
			return e
		}, U.viewport = function ()
		{
			return {
				width: o.innerWidth,
				height: o.innerHeight
			}
		}, U.isAncestor = "compareDocumentPosition" in s ? function (t, e)
		{
			return 16 == (16 & t.compareDocumentPosition(e))
		} : function (t, e)
		{
			return t !== e && t.contains(e)
		}, U
	}, void 0 !== t && t.exports ? t.exports = i() : void 0 === (o = "function" == typeof (r = i) ? r.call(e, n, e, t) : r) || (t.exports = o)
}, function (t, e, n)
{
	var r = n(21),
		o = n(101),
		i = n(63),
		s = Object.defineProperty;
	e.f = n(16) ? Object.defineProperty : function (t, e, n)
	{
		if (r(t), e = i(e, !0), r(n), o) try
		{
			return s(t, e, n)
		}
		catch (t)
		{}
		if ("get" in n || "set" in n) throw TypeError("Accessors not supported!");
		return "value" in n && (t[e] = n.value), t
	}
}, function (t, e)
{
	var n = t.exports = {
		version: "2.5.3"
	};
	"number" == typeof __e && (__e = n)
}, function (t, e)
{
	var n = t.exports = "undefined" != typeof window && window.Math == Math ? window : "undefined" != typeof self && self.Math == Math ? self : Function("return this")();
	"number" == typeof __g && (__g = n)
}, function (t, e, n)
{
	var r;
	! function (o)
	{
		"use strict";
		var i = window.requestAnimationFrame || window.webkitRequestAnimationFrame || window.mozRequestAnimationFrame || window.msRequestAnimationFrame || function (t)
		{
			return window.setTimeout(t, 1e3 / 60)
		};

		function s()
		{
			this.frames = [], this.lastId = 0, this.raf = i, this.batch = {
				hash:
				{},
				read: [],
				write: [],
				mode: null
			}
		}
		s.prototype.read = function (t, e)
		{
			var n = this.add("read", t, e),
				r = n.id;
			return this.batch.read.push(n.id), "reading" === this.batch.mode || this.batch.scheduled ? r : (this.scheduleBatch(), r)
		}, s.prototype.write = function (t, e)
		{
			var n = this.add("write", t, e),
				r = this.batch.mode,
				o = n.id;
			return this.batch.write.push(n.id), "writing" === r || "reading" === r || this.batch.scheduled ? o : (this.scheduleBatch(), o)
		}, s.prototype.defer = function (t, e, n)
		{
			"function" == typeof t && (n = e, e = t, t = 1);
			var r = this,
				o = t - 1;
			return this.schedule(o, function ()
			{
				r.run(
				{
					fn: e,
					ctx: n
				})
			})
		}, s.prototype.clear = function (t)
		{
			if ("function" == typeof t) return this.clearFrame(t);
			var e = this.batch.hash[t];
			if (e)
			{
				var n = this.batch[e.type],
					r = n.indexOf(t);
				delete this.batch.hash[t], ~r && n.splice(r, 1)
			}
		}, s.prototype.clearFrame = function (t)
		{
			var e = this.frames.indexOf(t);
			~e && this.frames.splice(e, 1)
		}, s.prototype.scheduleBatch = function ()
		{
			var t = this;
			this.schedule(0, function ()
			{
				t.batch.scheduled = !1, t.runBatch()
			}), this.batch.scheduled = !0
		}, s.prototype.uniqueId = function ()
		{
			return ++this.lastId
		}, s.prototype.flush = function (t)
		{
			for (var e; e = t.shift();) this.run(this.batch.hash[e])
		}, s.prototype.runBatch = function ()
		{
			try
			{
				this.batch.mode = "reading", this.flush(this.batch.read), this.batch.mode = "writing", this.flush(this.batch.write), this.batch.mode = null
			}
			catch (t)
			{
				throw this.runBatch(), t
			}
		}, s.prototype.add = function (t, e, n)
		{
			var r = this.uniqueId();
			return this.batch.hash[r] = {
				id: r,
				fn: e,
				ctx: n,
				type: t
			}
		}, s.prototype.run = function (t)
		{
			var e = t.ctx || this,
				n = t.fn;
			if (delete this.batch.hash[t.id], !this.onError) return n.call(e);
			try
			{
				n.call(e)
			}
			catch (t)
			{
				this.onError(t)
			}
		}, s.prototype.loop = function ()
		{
			var t = this,
				e = this.raf;
			this.looping || (e(function n()
			{
				var r = t.frames.shift();
				t.frames.length ? e(n) : t.looping = !1, r && r()
			}), this.looping = !0)
		}, s.prototype.schedule = function (t, e)
		{
			return this.frames[t] ? this.schedule(t + 1, e) : (this.loop(), this.frames[t] = e)
		}, o = o || new s, void 0 !== t && t.exports ? t.exports = o : void 0 === (r = function ()
		{
			return o
		}.call(e, n, e, t)) || (t.exports = r)
	}(window.fastdom)
}, function (t, e, n)
{
	var r = n(190),
		o = n(187);
	t.exports = function (t, e)
	{
		var n = o(t, e);
		return r(n) ? n : void 0
	}
}, function (t, e)
{
	var n = {}.hasOwnProperty;
	t.exports = function (t, e)
	{
		return n.call(t, e)
	}
}, function (t, e, n)
{
	t.exports = !n(36)(function ()
	{
		return 7 != Object.defineProperty(
		{}, "a",
		{
			get: function ()
			{
				return 7
			}
		}).a
	})
}, function (t, e)
{
	t.exports = function (t)
	{
		var e = typeof t;
		return null != t && ("object" == e || "function" == e)
	}
}, function (t, e)
{
	t.exports = function (t)
	{
		return null != t && "object" == typeof t
	}
}, function (t, e, n)
{
	var r = n(253),
		o = n(61);
	t.exports = function (t)
	{
		return r(o(t))
	}
}, function (t, e)
{
	t.exports = function (t)
	{
		return "object" == typeof t ? null !== t : "function" == typeof t
	}
}, function (t, e, n)
{
	var r = n(20);
	t.exports = function (t)
	{
		if (!r(t)) throw TypeError(t + " is not an object!");
		return t
	}
}, function (t, e, n)
{
	var r = n(10),
		o = n(27);
	t.exports = n(16) ? function (t, e, n)
	{
		return r.f(t, e, o(1, n))
	} : function (t, e, n)
	{
		return t[e] = n, t
	}
}, function (t, e, n)
{
	var r = n(12),
		o = n(11),
		i = n(64),
		s = n(22),
		a = function (t, e, n)
		{
			var c, u, l, f = t & a.F,
				d = t & a.G,
				p = t & a.S,
				h = t & a.P,
				v = t & a.B,
				_ = t & a.W,
				m = d ? o : o[e] || (o[e] = {}),
				g = m.prototype,
				y = d ? r : p ? r[e] : (r[e] ||
				{}).prototype;
			for (c in d && (n = e), n)(u = !f && y && void 0 !== y[c]) && c in m || (l = u ? y[c] : n[c], m[c] = d && "function" != typeof y[c] ? n[c] : v && u ? i(l, r) : _ && y[c] == l ? function (t)
			{
				var e = function (e, n, r)
				{
					if (this instanceof t)
					{
						switch (arguments.length)
						{
						case 0:
							return new t;
						case 1:
							return new t(e);
						case 2:
							return new t(e, n)
						}
						return new t(e, n, r)
					}
					return t.apply(this, arguments)
				};
				return e.prototype = t.prototype, e
			}(l) : h && "function" == typeof l ? i(Function.call, l) : l, h && ((m.virtual || (m.virtual = {}))[c] = l, t & a.R && g && !g[c] && s(g, c, l)))
		};
	a.F = 1, a.G = 2, a.S = 4, a.P = 8, a.B = 16, a.W = 32, a.U = 64, a.R = 128, t.exports = a
}, function (t, e, n)
{
	var r = n(25),
		o = n(209),
		i = n(208),
		s = "[object Null]",
		a = "[object Undefined]",
		c = r ? r.toStringTag : void 0;
	t.exports = function (t)
	{
		return null == t ? void 0 === t ? a : s : c && c in Object(t) ? o(t) : i(t)
	}
}, function (t, e, n)
{
	var r = n(7).Symbol;
	t.exports = r
}, function (t, e)
{
	t.exports = {}
}, function (t, e)
{
	t.exports = function (t, e)
	{
		return {
			enumerable: !(1 & t),
			configurable: !(2 & t),
			writable: !(4 & t),
			value: e
		}
	}
}, function (t, e, n)
{
	var r = n(207)();
	t.exports = r
}, function (t, e, n)
{
	var r = n(34),
		o = 1 / 0;
	t.exports = function (t)
	{
		if ("string" == typeof t || r(t)) return t;
		var e = t + "";
		return "0" == e && 1 / t == -o ? "-0" : e
	}
}, function (t, e, n)
{
	var r = n(178);
	t.exports = function (t, e)
	{
		var n = t.__data__;
		return r(e) ? n["string" == typeof e ? "string" : "hash"] : n.map
	}
}, function (t, e, n)
{
	var r = n(14)(Object, "create");
	t.exports = r
}, function (t, e, n)
{
	var r = n(48);
	t.exports = function (t, e)
	{
		for (var n = t.length; n--;)
			if (r(t[n][0], e)) return n;
		return -1
	}
}, function (t, e, n)
{
	var r = n(200),
		o = n(199),
		i = n(198),
		s = n(197),
		a = n(196);

	function c(t)
	{
		var e = -1,
			n = null == t ? 0 : t.length;
		for (this.clear(); ++e < n;)
		{
			var r = t[e];
			this.set(r[0], r[1])
		}
	}
	c.prototype.clear = r, c.prototype.delete = o, c.prototype.get = i, c.prototype.has = s, c.prototype.set = a, t.exports = c
}, function (t, e, n)
{
	var r = n(24),
		o = n(18),
		i = "[object Symbol]";
	t.exports = function (t)
	{
		return "symbol" == typeof t || o(t) && r(t) == i
	}
}, function (t, e)
{
	var n = 0,
		r = Math.random();
	t.exports = function (t)
	{
		return "Symbol(".concat(void 0 === t ? "" : t, ")_", (++n + r).toString(36))
	}
}, function (t, e)
{
	t.exports = function (t)
	{
		try
		{
			return !!t()
		}
		catch (t)
		{
			return !0
		}
	}
}, function (t, e, n)
{
	var r = n(213),
		o = 1 / 0;
	t.exports = function (t)
	{
		return null != t && t.length ? r(t, o) : []
	}
}, function (t, e)
{
	t.exports = function (t)
	{
		return t
	}
}, function (t, e)
{
	t.exports = function (t, e)
	{
		for (var n = -1, r = null == t ? 0 : t.length, o = Array(r); ++n < r;) o[n] = e(t[n], n, t);
		return o
	}
}, function (t, e, n)
{
	var r = n(6),
		o = n(34),
		i = /\.|\[(?:[^[\]]*|(["'])(?:(?!\1)[^\\]|\\.)*?\1)\]/,
		s = /^\w*$/;
	t.exports = function (t, e)
	{
		if (r(t)) return !1;
		var n = typeof t;
		return !("number" != n && "symbol" != n && "boolean" != n && null != t && !o(t)) || s.test(t) || !i.test(t) || null != e && t in Object(e)
	}
}, function (t, e, n)
{
	var r = n(162),
		o = n(157),
		i = n(47);
	t.exports = function (t)
	{
		return i(t) ? r(t) : o(t)
	}
}, function (t, e)
{
	t.exports = function (t)
	{
		var e = -1,
			n = Array(t.size);
		return t.forEach(function (t)
		{
			n[++e] = t
		}), n
	}
}, function (t, e, n)
{
	var r = n(186),
		o = n(179),
		i = n(177),
		s = n(176),
		a = n(175);

	function c(t)
	{
		var e = -1,
			n = null == t ? 0 : t.length;
		for (this.clear(); ++e < n;)
		{
			var r = t[e];
			this.set(r[0], r[1])
		}
	}
	c.prototype.clear = r, c.prototype.delete = o, c.prototype.get = i, c.prototype.has = s, c.prototype.set = a, t.exports = c
}, function (t, e, n)
{
	var r = n(14)(n(7), "Map");
	t.exports = r
}, function (t, e)
{
	var n = 9007199254740991,
		r = /^(?:0|[1-9]\d*)$/;
	t.exports = function (t, e)
	{
		var o = typeof t;
		return !!(e = null == e ? n : e) && ("number" == o || "symbol" != o && r.test(t)) && t > -1 && t % 1 == 0 && t < e
	}
}, function (t, e)
{
	var n = 9007199254740991;
	t.exports = function (t)
	{
		return "number" == typeof t && t > -1 && t % 1 == 0 && t <= n
	}
}, function (t, e, n)
{
	var r = n(87),
		o = n(46);
	t.exports = function (t)
	{
		return null != t && o(t.length) && !r(t)
	}
}, function (t, e)
{
	t.exports = function (t, e)
	{
		return t === e || t != t && e != e
	}
}, function (t, e, n)
{
	var r = n(210),
		o = n(18),
		i = Object.prototype,
		s = i.hasOwnProperty,
		a = i.propertyIsEnumerable,
		c = r(function ()
		{
			return arguments
		}()) ? r : function (t)
		{
			return o(t) && s.call(t, "callee") && !a.call(t, "callee")
		};
	t.exports = c
}, function (t, e)
{
	e.f = {}.propertyIsEnumerable
}, function (t, e, n)
{
	var r = n(12),
		o = n(11),
		i = n(60),
		s = n(52),
		a = n(10).f;
	t.exports = function (t)
	{
		var e = o.Symbol || (o.Symbol = i ?
		{} : r.Symbol ||
		{});
		"_" == t.charAt(0) || t in e || a(e, t,
		{
			value: s.f(t)
		})
	}
}, function (t, e, n)
{
	e.f = n(8)
}, function (t, e, n)
{
	var r = n(10).f,
		o = n(15),
		i = n(8)("toStringTag");
	t.exports = function (t, e, n)
	{
		t && !o(t = n ? t : t.prototype, i) && r(t, i,
		{
			configurable: !0,
			value: e
		})
	}
}, function (t, e)
{
	t.exports = "constructor,hasOwnProperty,isPrototypeOf,propertyIsEnumerable,toLocaleString,toString,valueOf".split(",")
}, function (t, e, n)
{
	var r = n(12),
		o = r["__core-js_shared__"] || (r["__core-js_shared__"] = {});
	t.exports = function (t)
	{
		return o[t] || (o[t] = {})
	}
}, function (t, e, n)
{
	var r = n(55)("keys"),
		o = n(35);
	t.exports = function (t)
	{
		return r[t] || (r[t] = o(t))
	}
}, function (t, e)
{
	var n = {}.toString;
	t.exports = function (t)
	{
		return n.call(t).slice(8, -1)
	}
}, function (t, e, n)
{
	var r = n(95),
		o = n(54);
	t.exports = Object.keys || function (t)
	{
		return r(t, o)
	}
}, function (t, e, n)
{
	var r = n(21),
		o = n(254),
		i = n(54),
		s = n(56)("IE_PROTO"),
		a = function () {},
		c = function ()
		{
			var t, e = n(100)("iframe"),
				r = i.length;
			for (e.style.display = "none", n(250).appendChild(e), e.src = "javascript:", (t = e.contentWindow.document).open(), t.write("<script>document.F=Object<\/script>"), t.close(), c = t.F; r--;) delete c.prototype[i[r]];
			return c()
		};
	t.exports = Object.create || function (t, e)
	{
		var n;
		return null !== t ? (a.prototype = r(t), n = new a, a.prototype = null, n[s] = t) : n = c(), void 0 === e ? n : o(n, e)
	}
}, function (t, e)
{
	t.exports = !0
}, function (t, e)
{
	t.exports = function (t)
	{
		if (void 0 == t) throw TypeError("Can't call method on  " + t);
		return t
	}
}, function (t, e)
{
	var n = Math.ceil,
		r = Math.floor;
	t.exports = function (t)
	{
		return isNaN(t = +t) ? 0 : (t > 0 ? r : n)(t)
	}
}, function (t, e, n)
{
	var r = n(20);
	t.exports = function (t, e)
	{
		if (!r(t)) return t;
		var n, o;
		if (e && "function" == typeof (n = t.toString) && !r(o = n.call(t))) return o;
		if ("function" == typeof (n = t.valueOf) && !r(o = n.call(t))) return o;
		if (!e && "function" == typeof (n = t.toString) && !r(o = n.call(t))) return o;
		throw TypeError("Can't convert object to primitive value")
	}
}, function (t, e, n)
{
	var r = n(259);
	t.exports = function (t, e, n)
	{
		if (r(t), void 0 === e) return t;
		switch (n)
		{
		case 1:
			return function (n)
			{
				return t.call(e, n)
			};
		case 2:
			return function (n, r)
			{
				return t.call(e, n, r)
			};
		case 3:
			return function (n, r, o)
			{
				return t.call(e, n, r, o)
			}
		}
		return function ()
		{
			return t.apply(e, arguments)
		}
	}
}, function (t, e, n)
{
	var r = n(17),
		o = n(121),
		i = n(86),
		s = "Expected a function",
		a = Math.max,
		c = Math.min;
	t.exports = function (t, e, n)
	{
		var u, l, f, d, p, h, v = 0,
			_ = !1,
			m = !1,
			g = !0;
		if ("function" != typeof t) throw new TypeError(s);

		function y(e)
		{
			var n = u,
				r = l;
			return u = l = void 0, v = e, d = t.apply(r, n)
		}

		function w(t)
		{
			var n = t - h;
			return void 0 === h || n >= e || n < 0 || m && t - v >= f
		}

		function b()
		{
			var t = o();
			if (w(t)) return x(t);
			p = setTimeout(b, function (t)
			{
				var n = e - (t - h);
				return m ? c(n, f - (t - v)) : n
			}(t))
		}

		function x(t)
		{
			return p = void 0, g && u ? y(t) : (u = l = void 0, d)
		}

		function k()
		{
			var t = o(),
				n = w(t);
			if (u = arguments, l = this, h = t, n)
			{
				if (void 0 === p) return function (t)
				{
					return v = t, p = setTimeout(b, e), _ ? y(t) : d
				}(h);
				if (m) return p = setTimeout(b, e), y(h)
			}
			return void 0 === p && (p = setTimeout(b, e)), d
		}
		return e = i(e) || 0, r(n) && (_ = !!n.leading, f = (m = "maxWait" in n) ? a(i(n.maxWait) || 0, e) : f, g = "trailing" in n ? !!n.trailing : g), k.cancel = function ()
		{
			void 0 !== p && clearTimeout(p), v = 0, u = h = l = p = void 0
		}, k.flush = function ()
		{
			return void 0 === p ? d : x(o())
		}, k
	}
}, function (t, e, n)
{
	"use strict";
	e.__esModule = !0;
	var r, o = n(102),
		i = (r = o) && r.__esModule ? r :
		{
			default: r
		};
	e.default = function (t, e, n)
	{
		return e in t ? (0, i.default)(t, e,
		{
			value: n,
			enumerable: !0,
			configurable: !0,
			writable: !0
		}) : t[e] = n, t
	}
}, function (t, e, n)
{
	var r, o, i;
	i = function ()
	{
		var t, e = document,
			n = e.documentElement,
			r = "getElementsByClassName",
			o = "getElementsByTagName",
			i = "querySelectorAll",
			s = "tagName",
			a = "nodeType",
			c = /#([\w\-]+)/,
			u = /\.[\w\-]+/g,
			l = /^#([\w\-]+)$/,
			f = /^([\w]+)?\.([\w\-]+)$/,
			d = /(^|,)\s*[>~+]/,
			p = /^\s+|\s*([,\s\+\~>]|$)\s*/g,
			h = /[\s\>\+\~]/,
			v = /(?![\s\w\-\/\?\&\=\:\.\(\)\!,@#%<>\{\}\$\*\^'"]*\]|[\s\w\+\-]*\))/,
			_ = /([.*+?\^=!:${}()|\[\]\/\\])/g,
			m = new RegExp(l.source + "|" + /^([\w\-]+)$/.source + "|" + /^\.([\w\-]+)$/.source),
			g = new RegExp("(" + h.source + ")" + v.source, "g"),
			y = new RegExp(h.source + v.source),
			w = new RegExp(/^(\*|[a-z0-9]+)?(?:([\.\#]+[\w\-\.#]+)?)/.source + "(" + /\[([\w\-]+)(?:([\|\^\$\*\~]?\=)['"]?([ \w\-\/\?\&\=\:\.\(\)\!,@#%<>\{\}\$\*\^]+)["']?)?\]/.source + ")?(" + /:([\w\-]+)(\(['"]?([^()]+)['"]?\))?/.source + ")?"),
			b = {
				" ": function (t)
				{
					return t && t !== n && t.parentNode
				},
				">": function (t, e)
				{
					return t && t.parentNode == e.parentNode && t.parentNode
				},
				"~": function (t)
				{
					return t && t.previousSibling
				},
				"+": function (t, e, n, r)
				{
					return !!t && ((n = A(t)) && (r = A(e)) && n == r && n)
				}
			};

		function x()
		{
			this.c = {}
		}
		x.prototype = {
			g: function (t)
			{
				return this.c[t] || void 0
			},
			s: function (t, e, n)
			{
				return e = n ? new RegExp(e) : e, this.c[t] = e
			}
		};
		var k = new x,
			E = new x,
			O = new x,
			C = new x;

		function j(t)
		{
			return k.g(t) || k.s(t, "(^|\\s+)" + t + "(\\s+|$)", 1)
		}

		function S(t, e)
		{
			for (var n = 0, r = t.length; n < r; n++) e(t[n])
		}

		function N(t)
		{
			for (var e = [], n = 0, r = t.length; n < r; ++n) z(t[n]) ? e = e.concat(t[n]) : e[e.length] = t[n];
			return e
		}

		function I(t)
		{
			for (var e = 0, n = t.length, r = []; e < n; e++) r[e] = t[e];
			return r
		}

		function A(t)
		{
			for (;
				(t = t.previousSibling) && 1 != t[a];);
			return t
		}

		function T(t)
		{
			return t.match(w)
		}

		function L(t, e, n, r, o, i, l, f, d, p, h)
		{
			var v, _, m, g, y;
			if (1 !== this[a]) return !1;
			if (e && "*" !== e && this[s] && this[s].toLowerCase() !== e) return !1;
			if (n && (_ = n.match(c)) && _[1] !== this.id) return !1;
			if (n && (y = n.match(u)))
				for (v = y.length; v--;)
					if (!j(y[v].slice(1)).test(this.className)) return !1;
			if (d && $.pseudos[d] && !$.pseudos[d](this, h)) return !1;
			if (r && !l)
				for (m in g = this.attributes)
					if (Object.prototype.hasOwnProperty.call(g, m) && (g[m].name || m) == o) return this;
			return !(r && ! function (t, e, n)
			{
				switch (t)
				{
				case "=":
					return e == n;
				case "^=":
					return e.match(O.g("^=" + n) || O.s("^=" + n, "^" + P(n), 1));
				case "$=":
					return e.match(O.g("$=" + n) || O.s("$=" + n, P(n) + "$", 1));
				case "*=":
					return e.match(O.g(n) || O.s(n, P(n), 1));
				case "~=":
					return e.match(O.g("~=" + n) || O.s("~=" + n, "(?:^|\\s+)" + P(n) + "(?:\\s+|$)", 1));
				case "|=":
					return e.match(O.g("|=" + n) || O.s("|=" + n, "^" + P(n) + "(-|$)", 1))
				}
				return 0
			}(i, q(this, o) || "", l)) && this
		}

		function P(t)
		{
			return E.g(t) || E.s(t, t.replace(_, "\\$1"))
		}

		function M(t, e, n, r)
		{
			var o;
			return (o = function t(r, i, s)
			{
				for (; s = b[n[i]](s, r);)
					if (F(s) && L.apply(s, T(e[i])))
					{
						if (!i) return s;
						if (o = t(s, i - 1, s)) return o
					}
			}(t, e.length - 1, t)) && (!r || G(o, r))
		}

		function F(t, e)
		{
			return t && "object" == typeof t && (e = t[a]) && (1 == e || 9 == e)
		}

		function D(t)
		{
			var e, n, r = [];
			t: for (e = 0; e < t.length; ++e)
			{
				for (n = 0; n < r.length; ++n)
					if (r[n] == t[e]) continue t;
				r[r.length] = t[e]
			}
			return r
		}

		function z(t)
		{
			return "object" == typeof t && isFinite(t.length)
		}

		function H(e, n, r)
		{
			return 9 === e[a] ? e.getElementById(n) : e.ownerDocument && ((r = e.ownerDocument.getElementById(n)) && G(r, e) && r || !G(e, e.ownerDocument) && t('[id="' + n + '"]', e)[0])
		}

		function $(n, i)
		{
			var s, c, u = function (t)
			{
				return t ? "string" == typeof t ? $(t)[0] : !t[a] && z(t) ? t[0] : t : e
			}(i);
			if (!u || !n) return [];
			if (n === window || F(n)) return !i || n !== window && F(u) && G(n, u) ? [n] : [];
			if (n && z(n)) return N(n);
			if (s = n.match(m))
			{
				if (s[1]) return (c = H(u, s[1])) ? [c] : [];
				if (s[2]) return I(u[o](s[2]));
				if (V && s[3]) return I(u[r](s[3]))
			}
			return t(n, u)
		}

		function R(t, e)
		{
			return function (n)
			{
				var r, o;
				d.test(n) ? 9 !== t[a] && ((o = r = t.getAttribute("id")) || t.setAttribute("id", o = "__qwerymeupscotty"), n = '[id="' + o + '"]' + n, e(t.parentNode || t, n, !0), r || t.removeAttribute("id")) : n.length && e(t, n, !1)
			}
		}
		var B, G = "compareDocumentPosition" in n ? function (t, e)
			{
				return 16 == (16 & e.compareDocumentPosition(t))
			} : "contains" in n ? function (t, e)
			{
				return (e = 9 === e[a] || e == window ? n : e) !== t && e.contains(t)
			} : function (t, e)
			{
				for (; t = t.parentNode;)
					if (t === e) return 1;
				return 0
			},
			q = ((B = e.createElement("p")).innerHTML = '<a href="#x">x</a>') && "#x" != B.firstChild.getAttribute("href") ? function (t, e)
			{
				return "class" === e ? t.className : "href" === e || "src" === e ? t.getAttribute(e, 2) : t.getAttribute(e)
			} : function (t, e)
			{
				return t.getAttribute(e)
			},
			V = !!e[r],
			W = e.querySelector && e[i],
			U = function (t, e)
			{
				var n, r, o = [];
				try
				{
					return 9 !== e[a] && d.test(t) ? (S(n = t.split(","), R(e, function (t, e)
					{
						1 == (r = t[i](e)).length ? o[o.length] = r.item(0) : r.length && (o = o.concat(I(r)))
					})), n.length > 1 && o.length > 1 ? D(o) : o) : I(e[i](t))
				}
				catch (t)
				{}
				return K(t, e)
			},
			K = function (t, e)
			{
				var n, r, i, c, u, d, h = [];
				if (r = (t = t.replace(p, "$1")).match(f))
				{
					for (u = j(r[2]), n = e[o](r[1] || "*"), i = 0, c = n.length; i < c; i++) u.test(n[i].className) && (h[h.length] = n[i]);
					return h
				}
				return S(d = t.split(","), R(e, function (t, n, r)
				{
					for (u = function (t, e)
						{
							var n, r, i, c, u, f, d, p = [],
								h = [],
								v = e,
								_ = C.g(t) || C.s(t, t.split(y)),
								m = t.match(g);
							if (!_.length) return p;
							if (c = (_ = _.slice(0)).pop(), _.length && (i = _[_.length - 1].match(l)) && (v = H(e, i[1])), !v) return p;
							for (f = T(c), n = 0, r = (u = v !== e && 9 !== v[a] && m && /^[+~]$/.test(m[m.length - 1]) ? function (t)
								{
									for (; v = v.nextSibling;) 1 == v[a] && (!f[1] || f[1] == v[s].toLowerCase()) && (t[t.length] = v);
									return t
								}([]) : v[o](f[1] || "*")).length; n < r; n++)(d = L.apply(u[n], f)) && (p[p.length] = d);
							return _.length ? (S(p, function (t)
							{
								M(t, _, m) && (h[h.length] = t)
							}), h) : p
						}(n, t), i = 0, c = u.length; i < c; i++)(9 === t[a] || r || G(u[i], e)) && (h[h.length] = u[i])
				})), d.length > 1 && h.length > 1 ? D(h) : h
			},
			Q = function (e)
			{
				void 0 !== e.useNativeQSA && (t = e.useNativeQSA && W ? U : K)
			};
		return Q(
		{
			useNativeQSA: !0
		}), $.configure = Q, $.uniq = D, $.is = function (t, e, n)
		{
			if (F(e)) return t == e;
			if (z(e)) return !!~N(e).indexOf(t);
			for (var r, o, i = e.split(","); e = i.pop();)
				if (r = C.g(e) || C.s(e, e.split(y)), o = e.match(g), r = r.slice(0), L.apply(t, T(r.pop())) && (!r.length || M(t, r, o, n))) return !0;
			return !1
		}, $.pseudos = {}, $
	}, void 0 !== t && t.exports ? t.exports = i() : void 0 === (o = "function" == typeof (r = i) ? r.call(e, n, e, t) : r) || (t.exports = o)
}, function (t, e, n)
{
	var r = n(125);
	t.exports = function (t, e)
	{
		var n = -1,
			o = t.length,
			i = o - 1;
		for (e = void 0 === e ? o : e; ++n < e;)
		{
			var s = r(n, i),
				a = t[s];
			t[s] = t[n], t[n] = a
		}
		return t.length = e, t
	}
}, function (t, e)
{
	t.exports = function (t)
	{
		return function (e)
		{
			return null == e ? void 0 : e[t]
		}
	}
}, function (t, e, n)
{
	var r = n(6),
		o = n(40),
		i = n(146),
		s = n(143);
	t.exports = function (t, e)
	{
		return r(t) ? t : o(t, e) ? [t] : i(s(t))
	}
}, function (t, e, n)
{
	var r = n(70),
		o = n(29);
	t.exports = function (t, e)
	{
		for (var n = 0, i = (e = r(e, t)).length; null != t && n < i;) t = t[o(e[n++])];
		return n && n == i ? t : void 0
	}
}, function (t, e)
{
	t.exports = function (t, e)
	{
		return function (n)
		{
			return null != n && n[t] === e && (void 0 !== e || t in Object(n))
		}
	}
}, function (t, e, n)
{
	var r = n(17);
	t.exports = function (t)
	{
		return t == t && !r(t)
	}
}, function (t, e, n)
{
	var r = n(14)(n(7), "Set");
	t.exports = r
}, function (t, e, n)
{
	var r = n(160),
		o = n(159),
		i = n(158),
		s = i && i.isTypedArray,
		a = s ? o(s) : r;
	t.exports = a
}, function (t, e)
{
	t.exports = function (t)
	{
		return t.webpackPolyfill || (t.deprecate = function () {}, t.paths = [], t.children || (t.children = []), Object.defineProperty(t, "loaded",
		{
			enumerable: !0,
			get: function ()
			{
				return t.l
			}
		}), Object.defineProperty(t, "id",
		{
			enumerable: !0,
			get: function ()
			{
				return t.i
			}
		}), t.webpackPolyfill = 1), t
	}
}, function (t, e, n)
{
	(function (t)
	{
		var r = n(7),
			o = n(161),
			i = "object" == typeof e && e && !e.nodeType && e,
			s = i && "object" == typeof t && t && !t.nodeType && t,
			a = s && s.exports === i ? r.Buffer : void 0,
			c = (a ? a.isBuffer : void 0) || o;
		t.exports = c
	}).call(this, n(76)(t))
}, function (t, e)
{
	t.exports = function (t, e)
	{
		for (var n = -1, r = Array(t); ++n < t;) r[n] = e(n);
		return r
	}
}, function (t, e)
{
	t.exports = function (t, e)
	{
		for (var n = -1, r = null == t ? 0 : t.length, o = 0, i = []; ++n < r;)
		{
			var s = t[n];
			e(s, n, t) && (i[o++] = s)
		}
		return i
	}
}, function (t, e)
{
	t.exports = function (t, e)
	{
		return t.has(e)
	}
}, function (t, e, n)
{
	var r = n(43),
		o = n(173),
		i = n(172);

	function s(t)
	{
		var e = -1,
			n = null == t ? 0 : t.length;
		for (this.__data__ = new r; ++e < n;) this.add(t[e])
	}
	s.prototype.add = s.prototype.push = o, s.prototype.has = i, t.exports = s
}, function (t, e, n)
{
	var r = n(81),
		o = n(171),
		i = n(80),
		s = 1,
		a = 2;
	t.exports = function (t, e, n, c, u, l)
	{
		var f = n & s,
			d = t.length,
			p = e.length;
		if (d != p && !(f && p > d)) return !1;
		var h = l.get(t);
		if (h && l.get(e)) return h == e;
		var v = -1,
			_ = !0,
			m = n & a ? new r : void 0;
		for (l.set(t, e), l.set(e, t); ++v < d;)
		{
			var g = t[v],
				y = e[v];
			if (c) var w = f ? c(y, g, v, e, t, l) : c(g, y, v, t, e, l);
			if (void 0 !== w)
			{
				if (w) continue;
				_ = !1;
				break
			}
			if (m)
			{
				if (!o(e, function (t, e)
					{
						if (!i(m, e) && (g === t || u(g, t, n, c, l))) return m.push(e)
					}))
				{
					_ = !1;
					break
				}
			}
			else if (g !== y && !u(g, y, n, c, l))
			{
				_ = !1;
				break
			}
		}
		return l.delete(t), l.delete(e), _
	}
}, function (t, e, n)
{
	var r = n(174),
		o = n(18);
	t.exports = function t(e, n, i, s, a)
	{
		return e === n || (null == e || null == n || !o(e) && !o(n) ? e != e && n != n : r(e, n, i, s, t, a))
	}
}, function (t, e)
{
	var n = Function.prototype.toString;
	t.exports = function (t)
	{
		if (null != t)
		{
			try
			{
				return n.call(t)
			}
			catch (t)
			{}
			try
			{
				return t + ""
			}
			catch (t)
			{}
		}
		return ""
	}
}, function (t, e, n)
{
	var r = n(33),
		o = n(195),
		i = n(194),
		s = n(193),
		a = n(192),
		c = n(191);

	function u(t)
	{
		var e = this.__data__ = new r(t);
		this.size = e.size
	}
	u.prototype.clear = o, u.prototype.delete = i, u.prototype.get = s, u.prototype.has = a, u.prototype.set = c, t.exports = u
}, function (t, e, n)
{
	var r = n(17),
		o = n(34),
		i = NaN,
		s = /^\s+|\s+$/g,
		a = /^[-+]0x[0-9a-f]+$/i,
		c = /^0b[01]+$/i,
		u = /^0o[0-7]+$/i,
		l = parseInt;
	t.exports = function (t)
	{
		if ("number" == typeof t) return t;
		if (o(t)) return i;
		if (r(t))
		{
			var e = "function" == typeof t.valueOf ? t.valueOf() : t;
			t = r(e) ? e + "" : e
		}
		if ("string" != typeof t) return 0 === t ? t : +t;
		t = t.replace(s, "");
		var n = c.test(t);
		return n || u.test(t) ? l(t.slice(2), n ? 2 : 8) : a.test(t) ? i : +t
	}
}, function (t, e, n)
{
	var r = n(24),
		o = n(17),
		i = "[object AsyncFunction]",
		s = "[object Function]",
		a = "[object GeneratorFunction]",
		c = "[object Proxy]";
	t.exports = function (t)
	{
		if (!o(t)) return !1;
		var e = r(t);
		return e == s || e == a || e == i || e == c
	}
}, function (t, e, n)
{
	(function (e)
	{
		var n = "object" == typeof e && e && e.Object === Object && e;
		t.exports = n
	}).call(this, n(211))
}, function (t, e)
{
	t.exports = function (t, e)
	{
		for (var n = -1, r = e.length, o = t.length; ++n < r;) t[o + n] = e[n];
		return t
	}
}, function (t, e, n)
{
	var r = n(50),
		o = n(27),
		i = n(19),
		s = n(63),
		a = n(15),
		c = n(101),
		u = Object.getOwnPropertyDescriptor;
	e.f = n(16) ? u : function (t, e)
	{
		if (t = i(t), e = s(e, !0), c) try
		{
			return u(t, e)
		}
		catch (t)
		{}
		if (a(t, e)) return o(!r.f.call(t, e), t[e])
	}
}, function (t, e, n)
{
	var r = n(95),
		o = n(54).concat("length", "prototype");
	e.f = Object.getOwnPropertyNames || function (t)
	{
		return r(t, o)
	}
}, function (t, e)
{
	e.f = Object.getOwnPropertySymbols
}, function (t, e, n)
{
	var r = n(61);
	t.exports = function (t)
	{
		return Object(r(t))
	}
}, function (t, e, n)
{
	var r = n(62),
		o = Math.min;
	t.exports = function (t)
	{
		return t > 0 ? o(r(t), 9007199254740991) : 0
	}
}, function (t, e, n)
{
	var r = n(15),
		o = n(19),
		i = n(252)(!1),
		s = n(56)("IE_PROTO");
	t.exports = function (t, e)
	{
		var n, a = o(t),
			c = 0,
			u = [];
		for (n in a) n != s && r(a, n) && u.push(n);
		for (; e.length > c;) r(a, n = e[c++]) && (~i(u, n) || u.push(n));
		return u
	}
}, function (t, e, n)
{
	t.exports = n(22)
}, function (t, e, n)
{
	"use strict";
	var r = n(60),
		o = n(23),
		i = n(96),
		s = n(22),
		a = n(15),
		c = n(26),
		u = n(255),
		l = n(53),
		f = n(249),
		d = n(8)("iterator"),
		p = !([].keys && "next" in [].keys()),
		h = function ()
		{
			return this
		};
	t.exports = function (t, e, n, v, _, m, g)
	{
		u(n, e, v);
		var y, w, b, x = function (t)
			{
				if (!p && t in C) return C[t];
				switch (t)
				{
				case "keys":
				case "values":
					return function ()
					{
						return new n(this, t)
					}
				}
				return function ()
				{
					return new n(this, t)
				}
			},
			k = e + " Iterator",
			E = "values" == _,
			O = !1,
			C = t.prototype,
			j = C[d] || C["@@iterator"] || _ && C[_],
			S = !p && j || x(_),
			N = _ ? E ? x("entries") : S : void 0,
			I = "Array" == e && C.entries || j;
		if (I && (b = f(I.call(new t))) !== Object.prototype && b.next && (l(b, k, !0), r || a(b, d) || s(b, d, h)), E && j && "values" !== j.name && (O = !0, S = function ()
			{
				return j.call(this)
			}), r && !g || !p && !O && C[d] || s(C, d, S), c[e] = S, c[k] = h, _)
			if (y = {
					values: E ? S : x("values"),
					keys: m ? S : x("keys"),
					entries: N
				}, g)
				for (w in y) w in C || i(C, w, y[w]);
			else o(o.P + o.F * (p || O), e, y);
		return y
	}
}, function (t, e, n)
{
	"use strict";
	var r = n(256)(!0);
	n(97)(String, "String", function (t)
	{
		this._t = String(t), this._i = 0
	}, function ()
	{
		var t, e = this._t,
			n = this._i;
		return n >= e.length ?
		{
			value: void 0,
			done: !0
		} : (t = r(e, n), this._i += t.length,
		{
			value: t,
			done: !1
		})
	})
}, function (t, e, n)
{
	"use strict";
	e.__esModule = !0;
	var r = s(n(258)),
		o = s(n(244)),
		i = "function" == typeof o.default && "symbol" == typeof r.default ? function (t)
		{
			return typeof t
		} : function (t)
		{
			return t && "function" == typeof o.default && t.constructor === o.default && t !== o.default.prototype ? "symbol" : typeof t
		};

	function s(t)
	{
		return t && t.__esModule ? t :
		{
			default: t
		}
	}
	e.default = "function" == typeof o.default && "symbol" === i(r.default) ? function (t)
	{
		return void 0 === t ? "undefined" : i(t)
	} : function (t)
	{
		return t && "function" == typeof o.default && t.constructor === o.default && t !== o.default.prototype ? "symbol" : void 0 === t ? "undefined" : i(t)
	}
}, function (t, e, n)
{
	var r = n(20),
		o = n(12).document,
		i = r(o) && r(o.createElement);
	t.exports = function (t)
	{
		return i ? o.createElement(t) :
		{}
	}
}, function (t, e, n)
{
	t.exports = !n(16) && !n(36)(function ()
	{
		return 7 != Object.defineProperty(n(100)("div"), "a",
		{
			get: function ()
			{
				return 7
			}
		}).a
	})
}, function (t, e, n)
{
	t.exports = {
		default: n(261),
		__esModule: !0
	}
}, function (t, e, n)
{
	var r, o, i;
	i = function (t, e)
	{
		t = t || "bean", e = e || this;
		var n, r, o, i, s, a = window,
			c = e[t],
			u = /[^\.]*(?=\..*)\.|.*/,
			l = /\..*/,
			f = "addEventListener",
			d = document ||
			{},
			p = d.documentElement ||
			{},
			h = p[f],
			v = h ? f : "attachEvent",
			_ = {},
			m = Array.prototype.slice,
			g = function (t, e)
			{
				return t.split(e || " ")
			},
			y = function (t)
			{
				return "string" == typeof t
			},
			w = function (t)
			{
				return "function" == typeof t
			},
			b = function (t, e, n)
			{
				for (n = 0; n < e.length; n++) e[n] && (t[e[n]] = 1);
				return t
			}(
			{}, g("click dblclick mouseup mousedown contextmenu mousewheel mousemultiwheel DOMMouseScroll mouseover mouseout mousemove selectstart selectend keydown keypress keyup orientationchange focus blur change reset select submit load unload beforeunload resize move DOMContentLoaded readystatechange message error abort scroll " + (h ? "show input invalid touchstart touchmove touchend touchcancel gesturestart gesturechange gestureend textinput readystatechange pageshow pagehide popstate hashchange offline online afterprint beforeprint dragstart dragenter dragover dragleave drag drop dragend loadstart progress suspend emptied stalled loadmetadata loadeddata canplay canplaythrough playing waiting seeking seeked ended durationchange timeupdate play pause ratechange volumechange cuechange checking noupdate downloading cached updateready obsolete " : ""))),
			x = (i = "compareDocumentPosition" in p ? function (t, e)
			{
				return e.compareDocumentPosition && 16 == (16 & e.compareDocumentPosition(t))
			} : "contains" in p ? function (t, e)
			{
				return (e = 9 === e.nodeType || e === window ? p : e) !== t && e.contains(t)
			} : function (t, e)
			{
				for (; t = t.parentNode;)
					if (t === e) return 1;
				return 0
			},
			{
				mouseenter:
				{
					base: "mouseover",
					condition: s = function (t)
					{
						var e = t.relatedTarget;
						return e ? e !== this && "xul" !== e.prefix && !/document/.test(this.toString()) && !i(e, this) : null == e
					}
				},
				mouseleave:
				{
					base: "mouseout",
					condition: s
				},
				mousewheel:
				{
					base: /Firefox/.test(navigator.userAgent) ? "DOMMouseScroll" : "mousewheel"
				}
			}),
			k = function ()
			{
				var t = g("altKey attrChange attrName bubbles cancelable ctrlKey currentTarget detail eventPhase getModifierState isTrusted metaKey relatedNode relatedTarget shiftKey srcElement target timeStamp type view which propertyName"),
					e = t.concat(g("button buttons clientX clientY dataTransfer fromElement offsetX offsetY pageX pageY screenX screenY toElement")),
					n = e.concat(g("wheelDelta wheelDeltaX wheelDeltaY wheelDeltaZ axis")),
					r = t.concat(g("char charCode key keyCode keyIdentifier keyLocation location")),
					o = t.concat(g("data")),
					i = t.concat(g("touches targetTouches changedTouches scale rotation")),
					s = t.concat(g("data origin source")),
					c = t.concat(g("state")),
					u = /over|out/,
					l = [
					{
						reg: /key/i,
						fix: function (t, e)
						{
							return e.keyCode = t.keyCode || t.which, r
						}
					},
					{
						reg: /click|mouse(?!(.*wheel|scroll))|menu|drag|drop/i,
						fix: function (t, n, r)
						{
							return n.rightClick = 3 === t.which || 2 === t.button, n.pos = {
								x: 0,
								y: 0
							}, t.pageX || t.pageY ? (n.clientX = t.pageX, n.clientY = t.pageY) : (t.clientX || t.clientY) && (n.clientX = t.clientX + d.body.scrollLeft + p.scrollLeft, n.clientY = t.clientY + d.body.scrollTop + p.scrollTop), u.test(r) && (n.relatedTarget = t.relatedTarget || t[("mouseover" == r ? "from" : "to") + "Element"]), e
						}
					},
					{
						reg: /mouse.*(wheel|scroll)/i,
						fix: function ()
						{
							return n
						}
					},
					{
						reg: /^text/i,
						fix: function ()
						{
							return o
						}
					},
					{
						reg: /^touch|^gesture/i,
						fix: function ()
						{
							return i
						}
					},
					{
						reg: /^message$/i,
						fix: function ()
						{
							return s
						}
					},
					{
						reg: /^popstate$/i,
						fix: function ()
						{
							return c
						}
					},
					{
						reg: /.*/,
						fix: function ()
						{
							return t
						}
					}],
					f = {},
					h = function (t, e, n)
					{
						if (arguments.length && (t = t || ((e.ownerDocument || e.document || e).parentWindow || a).event, this.originalEvent = t, this.isNative = n, this.isBean = !0, t))
						{
							var r, o, i, s, c, u = t.type,
								d = t.target || t.srcElement;
							if (this.target = d && 3 === d.nodeType ? d.parentNode : d, n)
							{
								if (!(c = f[u]))
									for (r = 0, o = l.length; r < o; r++)
										if (l[r].reg.test(u))
										{
											f[u] = c = l[r].fix;
											break
										} for (r = (s = c(t, this, u)).length; r--;) !((i = s[r]) in this) && i in t && (this[i] = t[i])
							}
						}
					};
				return h.prototype.preventDefault = function ()
				{
					this.originalEvent.preventDefault ? this.originalEvent.preventDefault() : this.originalEvent.returnValue = !1
				}, h.prototype.stopPropagation = function ()
				{
					this.originalEvent.stopPropagation ? this.originalEvent.stopPropagation() : this.originalEvent.cancelBubble = !0
				}, h.prototype.stop = function ()
				{
					this.preventDefault(), this.stopPropagation(), this.stopped = !0
				}, h.prototype.stopImmediatePropagation = function ()
				{
					this.originalEvent.stopImmediatePropagation && this.originalEvent.stopImmediatePropagation(), this.isImmediatePropagationStopped = function ()
					{
						return !0
					}
				}, h.prototype.isImmediatePropagationStopped = function ()
				{
					return this.originalEvent.isImmediatePropagationStopped && this.originalEvent.isImmediatePropagationStopped()
				}, h.prototype.clone = function (t)
				{
					var e = new h(this, this.element, this.isNative);
					return e.currentTarget = t, e
				}, h
			}(),
			E = function (t, e)
			{
				return h || e || t !== d && t !== a ? t : p
			},
			O = function ()
			{
				var t = function (t, e, n, r)
					{
						var o = function (n, o)
							{
								return e.apply(t, r ? m.call(o, n ? 0 : 1).concat(r) : o)
							},
							i = function (n, r)
							{
								return e.__beanDel ? e.__beanDel.ft(n.target, t) : r
							},
							s = n ? function (t)
							{
								var e = i(t, this);
								if (n.apply(e, arguments)) return t && (t.currentTarget = e), o(t, arguments)
							} : function (t)
							{
								return e.__beanDel && (t = t.clone(i(t))), o(t, arguments)
							};
						return s.__beanDel = e.__beanDel, s
					},
					e = function (e, n, r, o, i, s, a)
					{
						var c, u = x[n];
						"unload" == n && (r = I(A, e, n, r, o)), u && (u.condition && (r = t(e, r, u.condition, s)), n = u.base || n), this.isNative = c = b[n] && !!e[v], this.customType = !h && !c && n, this.element = e, this.type = n, this.original = o, this.namespaces = i, this.eventType = h || c ? n : "propertychange", this.target = E(e, c), this[v] = !!this.target[v], this.root = a, this.handler = t(e, r, null, s)
					};
				return e.prototype.inNamespaces = function (t)
				{
					var e, n, r = 0;
					if (!t) return !0;
					if (!this.namespaces) return !1;
					for (e = t.length; e--;)
						for (n = this.namespaces.length; n--;) t[e] == this.namespaces[n] && r++;
					return t.length === r
				}, e.prototype.matches = function (t, e, n)
				{
					return !(this.element !== t || e && this.original !== e || n && this.handler !== n)
				}, e
			}(),
			C = (r = {}, o = function (t, e, n, i, s, a)
			{
				var c = s ? "r" : "$";
				if (e && "*" != e)
				{
					var u, l = 0,
						f = r[c + e],
						d = "*" == t;
					if (!f) return;
					for (u = f.length; l < u; l++)
						if ((d || f[l].matches(t, n, i)) && !a(f[l], f, l, e)) return
				}
				else
					for (var p in r) p.charAt(0) == c && o(t, p.substr(1), n, i, s, a)
			},
			{
				has: function (t, e, n, o)
				{
					var i, s = r[(o ? "r" : "$") + e];
					if (s)
						for (i = s.length; i--;)
							if (!s[i].root && s[i].matches(t, n, null)) return !0;
					return !1
				},
				get: function (t, e, n, r)
				{
					var i = [];
					return o(t, e, n, null, r, function (t)
					{
						return i.push(t)
					}), i
				},
				put: function (t)
				{
					var e = !t.root && !this.has(t.element, t.type, null, !1),
						n = (t.root ? "r" : "$") + t.type;
					return (r[n] || (r[n] = [])).push(t), e
				},
				del: function (t)
				{
					o(t.element, t.type, null, t.handler, t.root, function (t, e, n)
					{
						return e.splice(n, 1), t.removed = !0, 0 === e.length && delete r[(t.root ? "r" : "$") + t.type], !1
					})
				},
				entries: function ()
				{
					var t, e = [];
					for (t in r) "$" == t.charAt(0) && (e = e.concat(r[t]));
					return e
				}
			}),
			j = function (t)
			{
				n = arguments.length ? t : d.querySelectorAll ? function (t, e)
				{
					return e.querySelectorAll(t)
				} : function ()
				{
					throw new Error("Bean: No selector engine installed")
				}
			},
			S = function (t, e)
			{
				if (h || !e || !t || t.propertyName == "_on" + e)
				{
					var n = C.get(this, e || t.type, null, !1),
						r = n.length,
						o = 0;
					for (t = new k(t, this, !0), e && (t.type = e); o < r && !t.isImmediatePropagationStopped(); o++) n[o].removed || n[o].handler.call(this, t)
				}
			},
			N = h ? function (t, e, n)
			{
				t[n ? f : "removeEventListener"](e, S, !1)
			} : function (t, e, n, r)
			{
				var o;
				n ? (C.put(o = new O(t, r || e, function (e)
				{
					S.call(t, e, r)
				}, S, null, null, !0)), r && null == t["_on" + r] && (t["_on" + r] = 0), o.target.attachEvent("on" + o.eventType, o.handler)) : (o = C.get(t, r || e, S, !0)[0]) && (o.target.detachEvent("on" + o.eventType, o.handler), C.del(o))
			},
			I = function (t, e, n, r, o)
			{
				return function ()
				{
					r.apply(this, arguments), t(e, n, o)
				}
			},
			A = function (t, e, n, r)
			{
				var o, i, s = e && e.replace(l, ""),
					a = C.get(t, s, null, !1),
					c = {};
				for (o = 0, i = a.length; o < i; o++) n && a[o].original !== n || !a[o].inNamespaces(r) || (C.del(a[o]), !c[a[o].eventType] && a[o][v] && (c[a[o].eventType] = {
					t: a[o].eventType,
					c: a[o].type
				}));
				for (o in c) C.has(t, c[o].t, null, !1) || N(t, c[o].t, !1, c[o].c)
			},
			T = h ? function (t, e, n)
			{
				var r = d.createEvent(t ? "HTMLEvents" : "UIEvents");
				r[t ? "initEvent" : "initUIEvent"](e, !0, !0, a, 1), n.dispatchEvent(r)
			} : function (t, e, n)
			{
				n = E(n, t), t ? n.fireEvent("on" + e, d.createEventObject()) : n["_on" + e]++
			},
			L = function (t, e, n)
			{
				var r, o, i, s, a = y(e);
				if (a && e.indexOf(" ") > 0)
				{
					for (s = (e = g(e)).length; s--;) L(t, e[s], n);
					return t
				}
				if ((o = a && e.replace(l, "")) && x[o] && (o = x[o].base), !e || a)(i = a && e.replace(u, "")) && (i = g(i, ".")), A(t, o, n, i);
				else if (w(e)) A(t, null, e);
				else
					for (r in e) e.hasOwnProperty(r) && L(t, r, e[r]);
				return t
			},
			P = function (t, e, r, o)
			{
				var i, s, a, c, f, d, p;
				if (void 0 !== r || "object" != typeof e)
				{
					for (w(r) ? (f = m.call(arguments, 3), o = i = r) : (i = o, f = m.call(arguments, 4), o = function (t, e)
						{
							var r = function (e, r)
								{
									for (var o, i = y(t) ? n(t, r) : t; e && e !== r; e = e.parentNode)
										for (o = i.length; o--;)
											if (i[o] === e) return e
								},
								o = function (t)
								{
									var n = r(t.target, this);
									n && e.apply(n, arguments)
								};
							return o.__beanDel = {
								ft: r,
								selector: t
							}, o
						}(r, i)), a = g(e), this === _ && (o = I(L, t, e, o, i)), c = a.length; c--;) p = C.put(d = new O(t, a[c].replace(l, ""), o, i, g(a[c].replace(u, ""), "."), f, !1)), d[v] && p && N(t, d.eventType, !0, d.customType);
					return t
				}
				for (s in e) e.hasOwnProperty(s) && P.call(this, t, s, e[s])
			},
			M = {
				on: P,
				add: function (t, e, n, r)
				{
					return P.apply(null, y(n) ? [t, n, e, r].concat(arguments.length > 3 ? m.call(arguments, 5) : []) : m.call(arguments))
				},
				one: function ()
				{
					return P.apply(_, arguments)
				},
				off: L,
				remove: L,
				clone: function (t, e, n)
				{
					for (var r, o, i = C.get(e, n, null, !1), s = i.length, a = 0; a < s; a++) i[a].original && (r = [t, i[a].type], (o = i[a].handler.__beanDel) && r.push(o.selector), r.push(i[a].original), P.apply(null, r));
					return t
				},
				fire: function (t, e, n)
				{
					var r, o, i, s, a, c = g(e);
					for (r = c.length; r--;)
						if (e = c[r].replace(l, ""), (s = c[r].replace(u, "")) && (s = g(s, ".")), s || n || !t[v])
							for (a = C.get(t, e, null, !1), n = [!1].concat(n), o = 0, i = a.length; o < i; o++) a[o].inNamespaces(s) && a[o].handler.apply(t, n);
						else T(b[e], e, t);
					return t
				},
				Event: k,
				setSelectorEngine: j,
				noConflict: function ()
				{
					return e[t] = c, this
				}
			};
		if (a.attachEvent)
		{
			var F = function ()
			{
				var t, e = C.entries();
				for (t in e) e[t].type && "unload" !== e[t].type && L(e[t].element, e[t].type);
				a.detachEvent("onunload", F), a.CollectGarbage && a.CollectGarbage()
			};
			a.attachEvent("onunload", F)
		}
		return j(), M
	}, void 0 !== t && t.exports ? t.exports = i() : void 0 === (o = "function" == typeof (r = i) ? r.call(e, n, e, t) : r) || (t.exports = o)
}, function (t, e, n)
{
	var r = n(120)(n(112));
	t.exports = r
}, function (t, e, n)
{
	var r = n(127),
		o = n(124),
		i = n(6);
	t.exports = function (t)
	{
		return (i(t) ? r : o)(t)
	}
}, function (t, e, n)
{
	var r = n(203),
		o = n(136);
	t.exports = function (t, e)
	{
		return t && t.length ? o(t, r(e, 2)) : []
	}
}, function (t, e)
{
	t.exports = '<svg><path d="M21 9.8l-.8-.8-5.2 4.8-5.2-4.8-.8.8 4.8 5.2-4.8 5.2.8.8 5.2-4.8 5.2 4.8.8-.8-4.8-5.2 4.8-5.2"></path></svg>'
}, function (t, e, n)
{
	"use strict";
	e.__esModule = !0;
	var r, o = n(222),
		i = (r = o) && r.__esModule ? r :
		{
			default: r
		};
	e.default = function (t)
	{
		return Array.isArray(t) ? t : (0, i.default)(t)
	}
}, function (t, e, n)
{
	var r;
	! function (e)
	{
		"use strict";

		function o()
		{}
		var i = o.prototype,
			s = e.EventEmitter;

		function a(t, e)
		{
			for (var n = t.length; n--;)
				if (t[n].listener === e) return n;
			return -1
		}

		function c(t)
		{
			return function ()
			{
				return this[t].apply(this, arguments)
			}
		}
		i.getListeners = function (t)
		{
			var e, n, r = this._getEvents();
			if (t instanceof RegExp)
				for (n in e = {}, r) r.hasOwnProperty(n) && t.test(n) && (e[n] = r[n]);
			else e = r[t] || (r[t] = []);
			return e
		}, i.flattenListeners = function (t)
		{
			var e, n = [];
			for (e = 0; e < t.length; e += 1) n.push(t[e].listener);
			return n
		}, i.getListenersAsObject = function (t)
		{
			var e, n = this.getListeners(t);
			return n instanceof Array && ((e = {})[t] = n), e || n
		}, i.addListener = function (t, e)
		{
			if (! function t(e)
				{
					return "function" == typeof e || e instanceof RegExp || !(!e || "object" != typeof e) && t(e.listener)
				}(e)) throw new TypeError("listener must be a function");
			var n, r = this.getListenersAsObject(t),
				o = "object" == typeof e;
			for (n in r) r.hasOwnProperty(n) && -1 === a(r[n], e) && r[n].push(o ? e :
			{
				listener: e,
				once: !1
			});
			return this
		}, i.on = c("addListener"), i.addOnceListener = function (t, e)
		{
			return this.addListener(t,
			{
				listener: e,
				once: !0
			})
		}, i.once = c("addOnceListener"), i.defineEvent = function (t)
		{
			return this.getListeners(t), this
		}, i.defineEvents = function (t)
		{
			for (var e = 0; e < t.length; e += 1) this.defineEvent(t[e]);
			return this
		}, i.removeListener = function (t, e)
		{
			var n, r, o = this.getListenersAsObject(t);
			for (r in o) o.hasOwnProperty(r) && -1 !== (n = a(o[r], e)) && o[r].splice(n, 1);
			return this
		}, i.off = c("removeListener"), i.addListeners = function (t, e)
		{
			return this.manipulateListeners(!1, t, e)
		}, i.removeListeners = function (t, e)
		{
			return this.manipulateListeners(!0, t, e)
		}, i.manipulateListeners = function (t, e, n)
		{
			var r, o, i = t ? this.removeListener : this.addListener,
				s = t ? this.removeListeners : this.addListeners;
			if ("object" != typeof e || e instanceof RegExp)
				for (r = n.length; r--;) i.call(this, e, n[r]);
			else
				for (r in e) e.hasOwnProperty(r) && (o = e[r]) && ("function" == typeof o ? i.call(this, r, o) : s.call(this, r, o));
			return this
		}, i.removeEvent = function (t)
		{
			var e, n = typeof t,
				r = this._getEvents();
			if ("string" === n) delete r[t];
			else if (t instanceof RegExp)
				for (e in r) r.hasOwnProperty(e) && t.test(e) && delete r[e];
			else delete this._events;
			return this
		}, i.removeAllListeners = c("removeEvent"), i.emitEvent = function (t, e)
		{
			var n, r, o, i, s = this.getListenersAsObject(t);
			for (i in s)
				if (s.hasOwnProperty(i))
					for (n = s[i].slice(0), o = 0; o < n.length; o++) !0 === (r = n[o]).once && this.removeListener(t, r.listener), r.listener.apply(this, e || []) === this._getOnceReturnValue() && this.removeListener(t, r.listener);
			return this
		}, i.trigger = c("emitEvent"), i.emit = function (t)
		{
			var e = Array.prototype.slice.call(arguments, 1);
			return this.emitEvent(t, e)
		}, i.setOnceReturnValue = function (t)
		{
			return this._onceReturnValue = t, this
		}, i._getOnceReturnValue = function ()
		{
			return !this.hasOwnProperty("_onceReturnValue") || this._onceReturnValue
		}, i._getEvents = function ()
		{
			return this._events || (this._events = {})
		}, o.noConflict = function ()
		{
			return e.EventEmitter = s, o
		}, void 0 === (r = function ()
		{
			return o
		}.call(e, n, e, t)) || (t.exports = r)
	}(this ||
	{})
}, function (t, e, n)
{
	"use strict";
	n.r(e);
	var r = n(2),
		o = n.n(r),
		i = n(1),
		s = n.n(i),
		a = n(4),
		c = n.n(a),
		u = n(3),
		l = n.n(u),
		f = (n(227), n(0)),
		d = n.n(f),
		p = n(5),
		h = n(13),
		v = n.n(h),
		_ = n(9),
		m = n.n(_),
		g = n(67),
		y = n.n(g);
	m.a.aug(
	{
		height: function ()
		{
			return this.dim().height
		}
	});
	var w = function (t, e)
	{
		return m()(y()(t, e))
	};
	w.create = function (t)
	{
		return m()(m.a.create(t))
	}, w.ancestor = function (t, e)
	{
		return null === t || void 0 === t || "html" === t.nodeName.toLowerCase() ? null : !t.parentNode || m()(t.parentNode).hasClass(e) ? t.parentNode : w.ancestor(t.parentNode, e)
	}, w.forEachElement = function (t, e)
	{
		var n = y()(t);
		return n.forEach(e), n
	};
	var b = w,
		x = n(109),
		k = new(n.n(x).a),
		E = [
		{
			name: "mobile",
			isTweakpoint: !1,
			width: 0
		},
		{
			name: "mobileMedium",
			isTweakpoint: !0,
			width: 375
		},
		{
			name: "mobileLandscape",
			isTweakpoint: !0,
			width: 480
		},
		{
			name: "phablet",
			isTweakpoint: !0,
			width: 660
		},
		{
			name: "tablet",
			isTweakpoint: !1,
			width: 740
		},
		{
			name: "desktop",
			isTweakpoint: !1,
			width: 980
		},
		{
			name: "leftCol",
			isTweakpoint: !0,
			width: 1140
		},
		{
			name: "wide",
			isTweakpoint: !1,
			width: 1300
		}],
		O = void 0,
		C = void 0,
		j = (document.visibilityState || document.webkitVisibilityState || document.mozVisibilityState || document.msVisibilityState, E.map(function (t)
		{
			return t.name
		})),
		S = function (t)
		{
			C = t.name, O = t.isTweakpoint ? function (t)
			{
				for (var e = j.indexOf(t), n = E[e]; e >= 0 && n.isTweakpoint;) n = E[e -= 1];
				return n.name
			}(C) : C
		},
		N = function ()
		{
			var t = window.getComputedStyle(document.body, "::after"),
				e = t.content.substring(1, t.content.length - 1),
				n = j.indexOf(e);
			S(E[n])
		},
		I = function ()
		{
			E.forEach(function (t, e, n)
			{
				var r = "(min-width: " + t.width + "px)";
				t.mql = e < n.length - 1 ? window.matchMedia(r + " and (max-width: " + (n[e + 1].width - 1) + "px)") : window.matchMedia(r), t.listener = function (t)
				{
					t && t.matches && S(this)
				}.bind(t), t.mql && t.mql.addListener(t.listener), t.mql && t.listener && t.listener(t.mql)
			})
		},
		A = function (t)
		{
			var e = t.min ? j.indexOf(t.min) : 0,
				n = t.max ? j.indexOf(t.max) : j.length - 1,
				r = j.indexOf(C || O);
			return e <= r && r <= n
		};
	! function ()
	{
		if (!navigator && !navigator.userAgent) return "";
		var t = navigator.userAgent,
			e = void 0,
			n = t.match(/(opera|chrome|safari|firefox|msie|trident(?=\/))\/?\s*(\d+)/i) || [];
		n && n[1] && /trident/i.test(n[1]) && (e = /\brv[ :]+(\d+)/g.exec(t)) && e[1] ? e[1] : n && "Chrome" === n[1] && (e = t.match(/\bOPR\/(\d+)/)) && e[1] ? e[1] : (n = n && n[2] ? [n[1], n[2]] : [navigator.appName, navigator.appVersion, "-?"], (e = t.match(/version\/(\d+)/i)) && e[1] && n.splice(1, 1, e[1]), n[0], n[1])
	}();
	"matchMedia" in window ? I() : (N(), k.on("window:throttledResize", N));
	var T = function (t)
		{
			return function (e)
			{
				return Math.pow(e, t)
			}
		},
		L = function (t)
		{
			return function (e)
			{
				return 1 - Math.abs(Math.pow(e - 1, t))
			}
		},
		P = function (t)
		{
			return function (e)
			{
				return e < .5 ? T(t)(2 * e) / 2 : L(t)(2 * e - 1) / 2 + .5
			}
		},
		M = {
			linear: P(1),
			easeInQuad: T(2),
			easeOutQuad: L(2),
			easeInOutQuad: P(2),
			easeInCubic: T(3),
			easeOutCubic: L(3),
			easeInOutCubic: P(3),
			easeInQuart: T(4),
			easeOutQuart: L(4),
			easeInOutQuart: P(4),
			easeInQuint: T(5),
			easeOutQuint: L(5),
			easeInOutQuint: P(5)
		},
		F = function (t)
		{
			var e = arguments.length > 1 && void 0 !== arguments[1] ? arguments[1] : 0,
				n = arguments.length > 2 && void 0 !== arguments[2] ? arguments[2] : "easeOutQuad",
				r = arguments.length > 3 && void 0 !== arguments[3] ? arguments[3] : document.body,
				o = m()(r),
				i = o.scrollTop(),
				s = t - i,
				a = function (t, e)
				{
					var n = new Date,
						r = M[t];
					return function ()
					{
						var t = new Date - n;
						return r(Math.min(1, t / e))
					}
				}(n, e),
				c = setInterval(function ()
				{
					v.a.write(function ()
					{
						return o.scrollTop(i + a() * s)
					})
				}, 15);
			setTimeout(function ()
			{
				clearInterval(c), v.a.write(function ()
				{
					return o.scrollTop(t)
				})
			}, e)
		},
		D = !1;
	try
	{
		var z = Object.defineProperty(
		{}, "passive",
		{
			get: function ()
			{
				D = !0
			}
		});
		window.addEventListener("test", null, z)
	}
	catch (t)
	{}
	var H = n(108),
		$ = n.n(H),
		R = n(107),
		B = n.n(R),
		G = n(66),
		q = n.n(G),
		V = 31,
		W = 1,
		U = 9,
		K = n(37),
		Q = n.n(K),
		Y = n(28),
		X = n.n(Y),
		J = n(106),
		Z = n.n(J),
		tt = function (t)
		{
			return "across" === t.direction
		},
		et = function (t, e)
		{
			return e.reduce(function (e, n)
			{
				var r = t.find(function (t)
				{
					return t.id === n
				});
				return r && e.push(r), e
			}, [])
		},
		nt = function (t)
		{
			return 1 !== t.group.length
		},
		rt = function (t, e)
		{
			if (nt(e))
			{
				var n = et(t, e.group);
				return {
					id: e.id,
					number: (i = n, i[0].humanNumber),
					length: function (t)
					{
						return t.reduce(function (t, e)
						{
							return t + e.length
						}, 0)
					}(n),
					separatorLocations: (r = n, o = {}, [",", "-"].forEach(function (t)
					{
						var e = 0,
							n = Q()(r.map(function (n)
							{
								var r = (n.separatorLocations[t] || []).map(function (t)
								{
									return t + e
								});
								return e += n.length, r
							}));
						o[t] = n
					}), o),
					direction: "",
					clue: function (t)
					{
						return t[0].clue
					}(n)
				}
			}
			var r, o, i;
			return e
		},
		ot = function (t)
		{
			return tt(t) ? X()(t.position.x, t.position.x + t.length).map(function (e)
			{
				return {
					x: e,
					y: t.position.y
				}
			}) : X()(t.position.y, t.position.y + t.length).map(function (e)
			{
				return {
					x: t.position.x,
					y: e
				}
			})
		},
		it = function (t, e)
		{
			return ot(e).every(function (e)
			{
				return /^.$/.test(t[e.x][e.y].value)
			})
		},
		st = function (t)
		{
			return "across" === t ? "down" : "across"
		},
		at = function (t, e)
		{
			return t + "_" + e
		},
		ct = function (t, e, n)
		{
			return t[at(e, n)]
		},
		ut = function (t, e, n, r)
		{
			var o = st(r.direction);
			return ot(r).filter(function (n)
			{
				var i = ct(e, n.x, n.y)[o];
				return !i || (function (t, e)
				{
					return e.group.includes(t.id)
				}(r, i) || !it(t, i))
			})
		},
		lt = function (t, e, n, r)
		{
			var o = X()(e).map(function (e)
			{
				return X()(t).map(function (t)
				{
					return {
						isHighlighted: !1,
						isEditable: !1,
						isError: !1,
						isAnimating: !1,
						value: r && r[e] && r[e][t] ? r[e][t] : ""
					}
				})
			});
			return n.forEach(function (t)
			{
				var e = t.position,
					n = e.x,
					r = e.y;
				o[n][r].number = t.number, ot(t).forEach(function (t)
				{
					o[t.x][t.y].isEditable = !0
				})
			}), o
		},
		ft = function (t)
		{
			var e = {};
			return t.forEach(function (t)
			{
				ot(t).forEach(function (n)
				{
					var r = at(n.x, n.y);
					void 0 === e[r] && (e[r] = {}), tt(t) ? e[r].across = t : e[r].down = t
				})
			}), e
		},
		dt = function (t, e, n)
		{
			return ot(t).some(function (t)
			{
				return t.x === e && t.y === n
			})
		},
		pt = function (t)
		{
			return t * (V + W) + W
		},
		ht = function (t, e)
		{
			return t.map(function (t, n)
			{
				return t.map(function (t, r)
				{
					return e(t, n, r)
				})
			})
		},
		vt = n(105),
		_t = n.n(vt),
		mt = function (t)
		{
			function e()
			{
				return o()(this, e), c()(this, (e.__proto__ || Object.getPrototypeOf(e)).apply(this, arguments))
			}
			return l()(e, t), s()(e, [
			{
				key: "componentDidMount",
				value: function ()
				{
					var t = Object(p.findDOMNode)(this);
					t && t.focus()
				}
			},
			{
				key: "componentDidUpdate",
				value: function ()
				{
					var t = Object(p.findDOMNode)(this);
					"" === this.props.value && t && t.focus()
				}
			},
			{
				key: "onInputChange",
				value: function (t)
				{
					t.target instanceof HTMLInputElement && this.props.onChange(t.target.value.toLowerCase())
				}
			},
			{
				key: "onKeyDown",
				value: function (t)
				{
					var e = Object(p.findDOMNode)(this);
					13 === t.keyCode && e && (e.blur(), this.props.onEnter())
				}
			},
			{
				key: "render",
				value: function ()
				{
					return d.a.createElement("input",
					{
						type: "text",
						className: "crossword__anagram-helper__clue-input",
						placeholder: "Enter letters",
						maxLength: this.props.clue.length,
						value: this.props.value,
						onChange: this.onInputChange.bind(this),
						onKeyDown: this.onKeyDown.bind(this)
					})
				}
			}]), e
		}(f.Component),
		gt = function (t)
		{
			function e()
			{
				return o()(this, e), c()(this, (e.__proto__ || Object.getPrototypeOf(e)).apply(this, arguments))
			}
			return l()(e, t), s()(e, [
			{
				key: "getEntries",
				value: function ()
				{
					var t = this,
						e = this.props.letters.filter(function (t)
						{
							return !t.entered
						});
					return this.props.entries.map(function (n)
					{
						n.solved = !!n.value;
						var r = t.props.hasShuffled ? n.value && n || e.shift() : n;
						return Object.assign(
						{},
						{
							key: n.key
						}, r)
					})
				}
			},
			{
				key: "render",
				value: function ()
				{
					var t = this,
						e = this.getEntries();
					return d.a.createElement("div",
					{
						className: "crossword__anagram-helper__clue-preview " + (e.length >= 9 ? "long" : "")
					}, d.a.createElement("div", null, d.a.createElement("strong", null, this.props.clue.number, " ", d.a.createElement("span",
					{
						className: "crossword__anagram-helper__direction"
					}, this.props.clue.direction)), " ", this.props.clue.clue), e.map(function (e, n)
					{
						var r = function (t, e)
						{
							var n = t[","],
								r = function (t)
								{
									return t.includes(e)
								};
							if (n && r(n)) return "crossword__anagram-helper__cell crossword__anagram-helper__cell--with-space";
							var o = t["-"];
							return o && r(o) ? "crossword__anagram-helper__cell crossword__anagram-helper__cell--with-hyphen" : "crossword__anagram-helper__cell"
						}(t.props.clue.separatorLocations, n + 1);
						return d.a.createElement("span",
						{
							className: r + (e.solved ? " has-value" : ""),
							key: e.key
						}, e.value || "")
					}))
				}
			}]), e
		}(f.Component),
		yt = function (t)
		{
			return Math.round(100 * t) / 100
		},
		wt = function (t)
		{
			function e()
			{
				return o()(this, e), c()(this, (e.__proto__ || Object.getPrototypeOf(e)).apply(this, arguments))
			}
			return l()(e, t), s()(e, [
			{
				key: "render",
				value: function ()
				{
					var t = 360 / this.props.letters.length;
					return d.a.createElement("div",
					{
						className: "crossword__anagram-helper-shuffler"
					}, this.props.letters.map(function (e, n)
					{
						return d.a.createElement("div",
						{
							className: "crossword__anagram-helper-shuffler__letter " + (e.entered ? "entered" : ""),
							style: function (t, e)
							{
								var n = t * Math.PI / 180 * e;
								return {
									left: 40 + yt(40 * Math.sin(n)) + "%",
									top: 40 + yt(40 * Math.cos(n)) + "%"
								}
							}(t, n),
							key: e.value + "-" + n
						}, e.value)
					}))
				}
			}]), e
		}(f.Component),
		bt = function (t)
		{
			function e()
			{
				o()(this, e);
				var t = c()(this, (e.__proto__ || Object.getPrototypeOf(e)).call(this));
				return t.state = {
					letters: [],
					clueInput: "",
					showInput: !0
				}, t
			}
			return l()(e, t), s()(e, [
			{
				key: "onClueInput",
				value: function (t)
				{
					/\s|\d/g.test(t) || this.setState(
					{
						clueInput: t
					})
				}
			},
			{
				key: "shuffleWord",
				value: function (t, e)
				{
					var n = e.map(function (t)
					{
						return t.value.toLowerCase()
					}).filter(function (e)
					{
						return t.includes(e)
					}).filter(Boolean).sort();
					return _t()(t.trim().split("").sort().reduce(function (t, e)
					{
						var n = $()(t.entries),
							r = n[0],
							o = n.slice(1),
							i = r === e.toLowerCase();
						return {
							letters: t.letters.concat(
							{
								value: e,
								entered: i
							}),
							entries: i ? o : t.entries
						}
					},
					{
						letters: [],
						entries: n
					}).letters)
				}
			},
			{
				key: "shuffle",
				value: function ()
				{
					this.canShuffle() && this.setState(
					{
						letters: this.shuffleWord(this.state.clueInput, this.entries()),
						showInput: !1
					})
				}
			},
			{
				key: "reset",
				value: function ()
				{
					this.state.clueInput && this.setState(
					{
						clueInput: "",
						showInput: !0
					})
				}
			},
			{
				key: "canShuffle",
				value: function ()
				{
					return !!this.state.clueInput && this.state.clueInput.length > 0
				}
			},
			{
				key: "entries",
				value: function ()
				{
					var t = this;
					return function (t, e)
					{
						if (nt(e))
						{
							var n = et(t, e.group);
							return Q()(n.map(function (t)
							{
								return ot(t)
							}))
						}
						return ot(e)
					}(this.props.entries, this.props.focussedEntry).map(function (e)
					{
						return Object.assign(
						{}, t.props.grid[e.x][e.y],
						{
							key: e.x + "," + e.y
						})
					})
				}
			},
			{
				key: "render",
				value: function ()
				{
					var t = {
							__html: B.a
						},
						e = rt(this.props.entries, this.props.focussedEntry),
						n = this.state.showInput ? d.a.createElement(mt,
						{
							value: this.state.clueInput,
							clue: e,
							onChange: this.onClueInput.bind(this),
							onEnter: this.shuffle.bind(this)
						}) : d.a.createElement(wt,
						{
							letters: this.state.letters
						});
					return d.a.createElement("div",
					{
						className: "crossword__anagram-helper-outer",
						"data-link-name": "Anagram Helper"
					}, d.a.createElement("div",
					{
						className: "crossword__anagram-helper-inner"
					}, n), d.a.createElement("button",
					{
						className: "button button--large button--tertiary crossword__anagram-helper-close",
						onClick: this.props.close.bind(this.props.crossword),
						dangerouslySetInnerHTML: t,
						"data-link-name": "Close"
					}), d.a.createElement("button",
					{
						className: "button button--large " + (this.state.clueInput ? "" : "button--tertiary"),
						onClick: this.reset.bind(this),
						"data-link-name": "Start Again"
					}, "start again"), d.a.createElement("button",
					{
						className: "button button--large " + (this.canShuffle() ? "" : "button--tertiary"),
						onClick: this.shuffle.bind(this),
						"data-link-name": "Shuffle"
					}, "shuffle"), d.a.createElement(gt,
					{
						clue: e,
						entries: this.entries(),
						letters: this.state.letters,
						hasShuffled: !this.state.showInput
					}))
				}
			}]), e
		}(f.Component),
		xt = n(65),
		kt = n.n(xt),
		Et = n(104),
		Ot = n.n(Et),
		Ct = n(103),
		jt = n.n(Ct),
		St = function (t)
		{
			return Object.keys(t).filter(function (e)
			{
				return !0 === t[e]
			}).join(" ")
		},
		Nt = function (t)
		{
			function e()
			{
				return o()(this, e), c()(this, (e.__proto__ || Object.getPrototypeOf(e)).apply(this, arguments))
			}
			return l()(e, t), s()(e, [
			{
				key: "onClick",
				value: function (t)
				{
					t.preventDefault(), this.props.setReturnPosition(), this.props.focusFirstCellInClueById(this.props.id)
				}
			},
			{
				key: "render",
				value: function ()
				{
					return d.a.createElement("li", null, d.a.createElement("a",
					{
						href: "#" + this.props.id,
						onClick: this.onClick.bind(this),
						className: St(
						{
							crossword__clue: !0,
							"crossword__clue--answered": this.props.hasAnswered,
							"crossword__clue--selected": this.props.isSelected,
							"crossword__clue--display-group-order": JSON.stringify(this.props.number) !== this.props.humanNumber
						})
					}, d.a.createElement("div",
					{
						className: "crossword__clue__number"
					}, this.props.humanNumber), d.a.createElement("div",
					{
						className: "crossword__clue__text",
						dangerouslySetInnerHTML:
						{
							__html: this.props.clue
						}
					})))
				}
			}]), e
		}(f.Component),
		It = function (t)
		{
			function e(t)
			{
				o()(this, e);
				var n = c()(this, (e.__proto__ || Object.getPrototypeOf(e)).call(this, t));
				return n.state = {
					showGradient: !0
				}, n
			}
			return l()(e, t), s()(e, [
			{
				key: "componentDidMount",
				value: function ()
				{
					var t = this;
					this.$cluesNode = Object(p.findDOMNode)(this.clues);
					var e = this.$cluesNode.scrollHeight - this.$cluesNode.clientHeight;
					jt.a.on(this.$cluesNode, "scroll", function (n)
					{
						var r = e - n.currentTarget.scrollTop > 25;
						t.state.showGradient !== r && t.setState(
						{
							showGradient: r
						})
					})
				}
			},
			{
				key: "componentDidUpdate",
				value: function (t)
				{
					var e = this;
					!A(
					{
						min: "tablet",
						max: "leftCol"
					}) || !this.props.focussed || t.focussed && t.focussed.id === this.props.focussed.id || v.a.read(function ()
					{
						e.scrollIntoView(e.props.focussed)
					})
				}
			},
			{
				key: "scrollIntoView",
				value: function (t)
				{
					var e = Object(p.findDOMNode)(this[t.id]);
					if (!(e.offsetTop - 100 > this.$cluesNode.scrollTop && e.offsetTop + 100 < this.$cluesNode.scrollTop + this.$cluesNode.clientHeight))
					{
						var n = e.offsetTop - this.$cluesNode.clientHeight / 2;
						F(n, 250, "easeOutQuad", this.$cluesNode)
					}
				}
			},
			{
				key: "render",
				value: function ()
				{
					var t = this,
						e = function (e)
						{
							return t.props.clues.filter(function (t)
							{
								return t.entry.direction === e
							}).map(function (e)
							{
								return d.a.createElement(Nt,
								{
									ref: function (n)
									{
										t[e.entry.id] = n
									},
									id: e.entry.id,
									key: e.entry.id,
									number: e.entry.number,
									humanNumber: e.entry.humanNumber,
									clue: e.entry.clue,
									hasAnswered: e.hasAnswered,
									isSelected: e.isSelected,
									focusFirstCellInClueById: t.props.focusFirstCellInClueById,
									setReturnPosition: function ()
									{
										t.props.setReturnPosition(window.scrollY)
									}
								})
							})
						};
					return d.a.createElement("div",
					{
						className: "crossword__clues--wrapper " + (this.state.showGradient ? "" : "hide-gradient")
					}, d.a.createElement("div",
					{
						className: "crossword__clues",
						ref: function (e)
						{
							t.clues = e
						}
					}, d.a.createElement("div",
					{
						className: "crossword__clues--across"
					}, d.a.createElement("h3",
					{
						className: "crossword__clues-header"
					}, "Across"), d.a.createElement("ol",
					{
						className: "crossword__clues-list"
					}, e("across"))), d.a.createElement("div",
					{
						className: "crossword__clues--down"
					}, d.a.createElement("h3",
					{
						className: "crossword__clues-header"
					}, "Down"), d.a.createElement("ol",
					{
						className: "crossword__clues-list"
					}, e("down")))), d.a.createElement("div",
					{
						className: "crossword__clues__gradient"
					}))
				}
			}]), e
		}(f.Component),
		At = function (t)
		{
			function e(t)
			{
				o()(this, e);
				var n = c()(this, (e.__proto__ || Object.getPrototypeOf(e)).call(this, t));
				return n.state = {
					confirming: !1
				}, n
			}
			return l()(e, t), s()(e, [
			{
				key: "confirm",
				value: function ()
				{
					var t = this;
					this.state.confirming ? (this.setState(
					{
						confirming: !1
					}), this.props.onClick()) : (this.setState(
					{
						confirming: !0
					}), setTimeout(function ()
					{
						t.setState(
						{
							confirming: !1
						})
					}, 2e3))
				}
			},
			{
				key: "render",
				value: function ()
				{
					var t = this.state.confirming ? "Confirm " + this.props.text.toLowerCase() : this.props.text,
						e = {},
						n = St((e["crossword__controls__button--confirm"] = this.state.confirming, e[this.props.className] = !0, e)),
						r = {
							"data-link-name": this.props["data-link-name"],
							onClick: this.confirm.bind(this),
							className: n
						};
					return d.a.createElement("button", r, t)
				}
			}]), e
		}(f.Component),
		Tt = function (t)
		{
			function e()
			{
				return o()(this, e), c()(this, (e.__proto__ || Object.getPrototypeOf(e)).apply(this, arguments))
			}
			return l()(e, t), s()(e, [
			{
				key: "render",
				value: function ()
				{
					var t = this.props.hasSolutions,
						e = this.props.clueInFocus,
						n = {
							clue: [],
							grid: []
						};
					return n.grid.unshift(d.a.createElement(At,
					{
						className: "button button--primary button--secondary",
						onClick: this.props.crossword.onClearAll.bind(this.props.crossword),
						key: "clear",
						"data-link-name": "Clear all",
						text: "Clear all"
					})), t && (n.grid.unshift(d.a.createElement(At,
					{
						className: "button button--primary button--secondary",
						onClick: this.props.crossword.onSolution.bind(this.props.crossword),
						key: "solution",
						"data-link-name": "Reveal all",
						text: "Reveal all"
					})), n.grid.unshift(d.a.createElement(At,
					{
						className: "button button--primary button--secondary",
						onClick: this.props.crossword.onCheckAll.bind(this.props.crossword),
						key: "checkAll",
						"data-link-name": "Check all",
						text: "Check all"
					}))), e && (n.clue.unshift(d.a.createElement("button",
					{
						className: "button button--primary button--crossword--current",
						onClick: this.props.crossword.onClearSingle.bind(this.props.crossword),
						key: "clear-single",
						"data-link-name": "Clear this"
					}, "Clear this")), n.clue.push(d.a.createElement("button",
					{
						className: "button button--primary button--crossword--current",
						onClick: this.props.crossword.onToggleAnagramHelper.bind(this.props.crossword),
						key: "anagram",
						"data-link-name": "Show anagram helper"
					}, "Anagram helper")), t && (n.clue.unshift(d.a.createElement("button",
					{
						className: "button button--primary button--crossword--current",
						onClick: this.props.crossword.onCheat.bind(this.props.crossword),
						key: "cheat",
						"data-link-name": "Reveal this"
					}, "Reveal this")), n.clue.unshift(d.a.createElement("button",
					{
						className: "button button--primary button--crossword--current",
						onClick: this.props.crossword.onCheck.bind(this.props.crossword),
						key: "check",
						"data-link-name": "Check this"
					}, "Check this")))), d.a.createElement("div",
					{
						className: "crossword__controls"
					}, d.a.createElement("div",
					{
						className: "crossword__controls__clue"
					}, n.clue), d.a.createElement("div",
					{
						className: "crossword__controls__grid"
					}, n.grid))
				}
			}]), e
		}(f.Component),
		Lt = function (t)
		{
			function e(t)
			{
				o()(this, e);
				var n = c()(this, (e.__proto__ || Object.getPrototypeOf(e)).call(this, t));
				return n.state = {
					value: n.props.value
				}, n
			}
			return l()(e, t), s()(e, [
			{
				key: "onClick",
				value: function (t)
				{
					this.props.crossword.onClickHiddenInput(t)
				}
			},
			{
				key: "onKeyDown",
				value: function (t)
				{
					this.props.crossword.onKeyDown(t)
				}
			},
			{
				key: "onBlur",
				value: function (t)
				{
					this.props.crossword.goToReturnPosition(t)
				}
			},
			{
				key: "onFocusPrevious",
				value: function ()
				{
					this.props.crossword.focusPreviousClue()
				}
			},
			{
				key: "onFocusNext",
				value: function ()
				{
					this.props.crossword.focusNextClue()
				}
			},
			{
				key: "touchStart",
				value: function (t)
				{
					this.props.crossword.onClickHiddenInput(t)
				}
			},
			{
				key: "handleChange",
				value: function (t)
				{
					this.props.crossword.insertCharacter(t.target.value), this.setState(
					{
						value: ""
					})
				}
			},
			{
				key: "render",
				value: function ()
				{
					var t = this;
					return d.a.createElement("div",
					{
						className: "crossword__hidden-input-wrapper",
						ref: function (e)
						{
							t.wrapper = e
						}
					}, d.a.createElement("input",
					{
						key: "1",
						type: "text",
						className: "crossword__hidden-input-prev-next",
						onFocus: this.onFocusPrevious.bind(this)
					}), d.a.createElement("input",
					{
						key: "2",
						type: "text",
						className: "crossword__hidden-input",
						maxLength: "1",
						onClick: this.onClick.bind(this),
						onChange: this.handleChange.bind(this),
						onTouchStart: this.touchStart.bind(this),
						onKeyDown: this.onKeyDown.bind(this),
						onBlur: this.onBlur.bind(this),
						value: this.state.value,
						autoComplete: "off",
						spellCheck: "false",
						autoCorrect: "off",
						ref: function (e)
						{
							t.input = e
						}
					}), d.a.createElement("input",
					{
						key: "3",
						type: "text",
						className: "crossword__hidden-input-prev-next",
						onFocus: this.onFocusNext.bind(this)
					}))
				}
			}]), e
		}(f.Component),
		Pt = function (t)
		{
			function e()
			{
				return o()(this, e), c()(this, (e.__proto__ || Object.getPrototypeOf(e)).apply(this, arguments))
			}
			return l()(e, t), s()(e, [
			{
				key: "shouldComponentUpdate",
				value: function (t)
				{
					return this.props.value !== t.value || this.props.isFocussed !== t.isFocussed || this.props.isHighlighted !== t.isHighlighted || this.props.isError !== t.isError
				}
			},
			{
				key: "onClick",
				value: function (t)
				{
					t.preventDefault(), this.props.handleSelect(this.props.x, this.props.y)
				}
			},
			{
				key: "render",
				value: function ()
				{
					var t = pt(this.props.y),
						e = pt(this.props.x),
						n = null;
					void 0 !== this.props.number && (n = d.a.createElement("text",
					{
						x: e + 1,
						y: t + U,
						key: "number",
						className: "crossword__cell-number"
					}, this.props.number));
					var r = null;
					return void 0 !== this.props.value && (r = d.a.createElement("text",
					{
						x: e + .5 * V,
						y: t + .675 * V,
						key: "entry",
						className: St(
						{
							"crossword__cell-text": !0,
							"crossword__cell-text--focussed": this.props.isFocussed,
							"crossword__cell-text--error": this.props.isError
						}),
						textAnchor: "middle"
					}, this.props.value)), d.a.createElement("g",
					{
						onClick: this.onClick.bind(this)
					}, d.a.createElement("rect",
					{
						x: e,
						y: t,
						width: V,
						height: V,
						className: St(
						{
							crossword__cell: !0,
							"crossword__cell--focussed": this.props.isFocussed,
							"crossword__cell--highlighted": this.props.isHighlighted
						})
					}), n, r)
				}
			}]), e
		}(f.Component),
		Mt = function (t, e, n)
		{
			if (n)
			{
				if ("," === n.separator) return function (t, e, n)
				{
					var r = pt(e),
						o = pt(t);
					if ("across" === n) return d.a.createElement("rect",
					{
						x: o - 1 - 1,
						y: r,
						key: ["sep", n, t, e].join("_"),
						width: 1,
						height: V
					});
					if ("down" === n) return d.a.createElement("rect",
					{
						x: o,
						y: r - 1 - 1,
						key: ["sep", n, t, e].join("_"),
						width: V,
						height: 1
					})
				}(t, e, n.direction);
				if ("-" === n.separator) return function (t, e, n)
				{
					var r = pt(e),
						o = pt(t),
						i = void 0,
						s = void 0;
					return "across" === n ? (i = V / 4, s = 1, d.a.createElement("rect",
					{
						x: o - .5 - i / 2,
						y: r + V / 2 + s / 2,
						key: ["sep", n, t, e].join("_"),
						width: i,
						height: s
					})) : "down" === n ? (i = 1, s = V / 4, d.a.createElement("rect",
					{
						x: o + V / 2 + i / 2,
						y: r - .5 - s / 2,
						key: ["sep", n, t, e].join("_"),
						width: i,
						height: s
					})) : void 0
				}(t, e, n.direction)
			}
		},
		Ft = function (t)
		{
			var e = function (e, n)
				{
					return t.crossword.onSelect(e, n)
				},
				n = pt(t.columns),
				r = pt(t.rows),
				o = [],
				i = [],
				s = function (t)
				{
					return Array.from(
					{
						length: t
					}, function (t, e)
					{
						return e
					})
				},
				a = t.cells;
			return s(t.rows).forEach(function (n)
			{
				return s(t.columns).forEach(function (r)
				{
					var s = a[r][n];
					if (s.isEditable)
					{
						var c = t.crossword.isHighlighted(r, n);
						o.push(d.a.createElement(Pt, Object.assign(
						{}, s,
						{
							handleSelect: e,
							x: r,
							y: n,
							key: "cell_" + r + "_" + n,
							isHighlighted: c,
							isFocussed: t.focussedCell && r === t.focussedCell.x && n === t.focussedCell.y
						}, void 0))), i = i.concat(Mt(r, n, function (e, n)
						{
							return t.separators[at(e, n)]
						}(r, n)))
					}
				})
			}), d.a.createElement("svg",
			{
				viewBox: "0 0 " + n + " " + r,
				className: St(
				{
					crossword__grid: !0,
					"crossword__grid--focussed": !!t.focussedCell
				})
			}, d.a.createElement("rect",
			{
				x: 0,
				y: 0,
				width: n,
				height: r,
				className: "crossword__grid-background"
			}), o, d.a.createElement("g",
			{
				className: "crossword__grid__separators"
			}, i))
		},
		Dt = 8,
		zt = 37,
		Ht = 38,
		$t = 39,
		Rt = 40,
		Bt = 46,
		Gt = function ()
		{
			function t(e)
			{
				o()(this, t), this.storage = window[e], this.available = this.isAvailable()
			}
			return s()(t, [
			{
				key: "isAvailable",
				value: function ()
				{
					var t = "local-storage-module-test";
					if (void 0 !== this.available) return this.available;
					try
					{
						this.storage.setItem(t, "graun"), this.storage.removeItem(t), this.available = !0
					}
					catch (t)
					{
						this.available = !1
					}
					return this.available
				}
			},
			{
				key: "get",
				value: function (t)
				{
					if (this.available)
					{
						var e = void 0;
						try
						{
							var n = this.getRaw(t);
							if (null === n || void 0 === n) return null;
							if (null === (e = JSON.parse(n))) return null
						}
						catch (e)
						{
							return this.remove(t), null
						}
						return e.expires && new Date > new Date(e.expires) ? (this.remove(t), null) : e.value
					}
				}
			},
			{
				key: "set",
				value: function (t, e)
				{
					var n = arguments.length > 2 && void 0 !== arguments[2] ? arguments[2] :
					{};
					if (this.available) return this.storage.setItem(t, JSON.stringify(
					{
						value: e,
						expires: n.expires
					}))
				}
			},
			{
				key: "setIfNotExists",
				value: function (t, e)
				{
					var n = arguments.length > 2 && void 0 !== arguments[2] ? arguments[2] :
					{};
					if (this.available && null === this.storage.getItem(t)) return this.storage.setItem(t, JSON.stringify(
					{
						value: e,
						expires: n.expires
					}))
				}
			},
			{
				key: "getRaw",
				value: function (t)
				{
					if (this.available) return this.storage.getItem(t)
				}
			},
			{
				key: "remove",
				value: function (t)
				{
					if (this.available) return this.storage.removeItem(t)
				}
			}]), t
		}(),
		qt = new Gt("localStorage"),
		Vt = (new Gt("sessionStorage"), function (t)
		{
			return "crosswords." + t
		}),
		Wt = function (t)
		{
			function e(t)
			{
				o()(this, e);
				var n = c()(this, (e.__proto__ || Object.getPrototypeOf(e)).call(this, t)),
					r = n.props.data.dimensions;
				return n.columns = r.cols, n.rows = r.rows, n.clueMap = ft(n.props.data.entries), n.state = {
					grid: lt(r.rows, r.cols, n.props.data.entries, n.props.loadGrid(n.props.id)),
					cellInFocus: null,
					directionOfEntry: null,
					showAnagramHelper: !1
				}, n
			}
			return l()(e, t), s()(e, [
			{
				key: "componentDidMount",
				value: function ()
				{
					var t = b(Object(p.findDOMNode)(this.stickyClueWrapper)),
						e = b(Object(p.findDOMNode)(this.game));
					k.on("window:resize", kt()(this.setGridHeight.bind(this), 200)), k.on("window:orientationchange", kt()(this.setGridHeight.bind(this), 200)), this.setGridHeight(),
						function (t, e, n)
						{
							var r = arguments.length > 3 && void 0 !== arguments[3] ? arguments[3] :
								{},
								o = r.passive,
								i = void 0 !== o && o,
								s = r.capture,
								a = void 0 !== s && s,
								c = r.once,
								u = void 0 !== c && c;
							D ? t.addEventListener(e, n,
							{
								passive: i,
								capture: a,
								once: u
							}) : u ? t.addEventListener(e, function r(o)
							{
								n.call(this, o), t.removeEventListener(e, r)
							}, a) : t.addEventListener(e, n, a)
						}(window, "scroll", function ()
						{
							var n = e.offset(),
								r = t.offset(),
								o = window.scrollY,
								i = o - n.top;
							i >= 0 ? o > n.top + n.height - r.height ? t.css(
							{
								top: "auto",
								bottom: 0
							}) : t.css(
							{
								top: i,
								bottom: ""
							}) : t.css(
							{
								top: "",
								bottom: ""
							})
						});
					var n = window.location.hash.replace("#", "");
					this.focusFirstCellInClueById(n)
				}
			},
			{
				key: "componentDidUpdate",
				value: function (t, e)
				{
					this.state.showAnagramHelper || this.state.showAnagramHelper === e.showAnagramHelper || this.focusCurrentCell()
				}
			},
			{
				key: "onKeyDown",
				value: function (t)
				{
					var e = this.state.cellInFocus;
					t.metaKey || t.ctrlKey || t.altKey || (t.keyCode === Dt || t.keyCode === Bt ? (t.preventDefault(), e && (this.cellIsEmpty(e.x, e.y) ? this.focusPrevious() : (this.setCellValue(e.x, e.y, ""), this.saveGrid()))) : t.keyCode === zt ? (t.preventDefault(), this.moveFocus(-1, 0)) : t.keyCode === Ht ? (t.preventDefault(), this.moveFocus(0, -1)) : t.keyCode === $t ? (t.preventDefault(), this.moveFocus(1, 0)) : t.keyCode === Rt && (t.preventDefault(), this.moveFocus(0, 1)))
				}
			},
			{
				key: "onSelect",
				value: function (t, e)
				{
					var n = this.state.cellInFocus,
						r = ct(this.clueMap, t, e),
						o = this.clueInFocus(),
						i = void 0;
					if (n && n.x === t && n.y === e && this.state.directionOfEntry) r[i = st(this.state.directionOfEntry)] && this.focusClue(t, e, i);
					else if (o && dt(o, t, e) && this.state.directionOfEntry) this.focusClue(t, e, this.state.directionOfEntry);
					else
					{
						this.state.cellInFocus = {
							x: t,
							y: e
						};
						var s = function (n)
						{
							return !!n && n.position.x === t && n.position.y === e
						};
						i = !s(r.across) && s(r.down) ? "down" : r.across ? "across" : "down", this.focusClue(t, e, i)
					}
				}
			},
			{
				key: "onCheat",
				value: function ()
				{
					var t = this;
					this.allHighlightedClues().forEach(function (e)
					{
						return t.cheat(e)
					}), this.saveGrid()
				}
			},
			{
				key: "onCheck",
				value: function ()
				{
					var t = this;
					this.allHighlightedClues().forEach(function (e)
					{
						return t.check(e)
					}), this.saveGrid()
				}
			},
			{
				key: "onSolution",
				value: function ()
				{
					var t = this;
					this.props.data.entries.forEach(function (e)
					{
						return t.cheat(e)
					}), this.saveGrid()
				}
			},
			{
				key: "onCheckAll",
				value: function ()
				{
					var t = this;
					this.props.data.entries.forEach(function (e)
					{
						return t.check(e)
					}), this.saveGrid()
				}
			},
			{
				key: "onClearAll",
				value: function ()
				{
					var t = this;
					this.setState(
					{
						grid: ht(this.state.grid, function (e, n, r)
						{
							var o = e.value;
							return e.value = "", t.props.onMove(
							{
								x: n,
								y: r,
								value: "",
								previousValue: o
							}), e
						})
					}), this.saveGrid()
				}
			},
			{
				key: "onClearSingle",
				value: function ()
				{
					var t = this,
						e = this.clueInFocus();
					if (e)
					{
						var n = function (t, e, n, r)
						{
							if (nt(r))
							{
								var o = et(n, r.group);
								return Z()(Q()(o.map(function (n)
								{
									return ut(t, e, 0, n)
								})), function (t)
								{
									return [t.x, t.y].join()
								})
							}
							return ut(t, e, 0, r)
						}(this.state.grid, this.clueMap, this.props.data.entries, e);
						this.setState(
						{
							grid: ht(this.state.grid, function (e, r, o)
							{
								if (n.some(function (t)
									{
										return t.x === r && t.y === o
									}))
								{
									var i = e.value;
									e.value = "", t.props.onMove(
									{
										x: r,
										y: o,
										value: "",
										previousValue: i
									})
								}
								return e
							})
						}), this.saveGrid()
					}
				}
			},
			{
				key: "onToggleAnagramHelper",
				value: function ()
				{
					this.state.showAnagramHelper ? this.setState(
					{
						showAnagramHelper: !1
					}) : this.clueInFocus() && this.setState(
					{
						showAnagramHelper: !0
					})
				}
			},
			{
				key: "onClickHiddenInput",
				value: function (t)
				{
					var e = this.state.cellInFocus;
					e && this.onSelect(e.x, e.y), "touchstart" === t.type && t.preventDefault()
				}
			},
			{
				key: "setGridHeight",
				value: function ()
				{
					var t = this;
					this.$gridWrapper || (this.$gridWrapper = b(Object(p.findDOMNode)(this.gridWrapper))), A(
					{
						max: "tablet"
					}) ? v.a.read(function ()
					{
						v.a.write(function ()
						{
							t.$gridWrapper.css("height", t.$gridWrapper.offset().width + "px")
						}), t.gridHeightIsSet = !0
					}) : this.gridHeightIsSet && this.$gridWrapper.attr("style", "")
				}
			},
			{
				key: "setCellValue",
				value: function (t, e, n)
				{
					var r = this,
						o = !(arguments.length > 3 && void 0 !== arguments[3]) || arguments[3];
					this.setState(
					{
						grid: ht(this.state.grid, function (i, s, a)
						{
							if (s === t && a === e)
							{
								var c = i.value;
								i.value = n, i.isError = !1, o && r.props.onMove(
								{
									x: t,
									y: e,
									value: n,
									previousValue: c
								})
							}
							return i
						})
					})
				}
			},
			{
				key: "getCellValue",
				value: function (t, e)
				{
					return this.state.grid[t][e].value
				}
			},
			{
				key: "setReturnPosition",
				value: function (t)
				{
					this.returnPosition = t
				}
			},
			{
				key: "updateGrid",
				value: function (t)
				{
					this.setState(
					{
						grid: lt(this.rows, this.columns, this.props.data.entries, t)
					})
				}
			},
			{
				key: "insertCharacter",
				value: function (t)
				{
					var e = t.toUpperCase(),
						n = this.state.cellInFocus;
					/[A-Za-z-0-9]/.test(e) && 1 === e.length && n && (this.setCellValue(n.x, n.y, e), this.saveGrid(), this.focusNext())
				}
			},
			{
				key: "cellIsEmpty",
				value: function (t, e)
				{
					return !this.getCellValue(t, e)
				}
			},
			{
				key: "goToReturnPosition",
				value: function ()
				{
					A(
					{
						max: "mobile"
					}) && (this.returnPosition && F(this.returnPosition, 250, "easeOutQuad"), this.returnPosition = null)
				}
			},
			{
				key: "indexOfClueInFocus",
				value: function ()
				{
					return this.props.data.entries.indexOf(this.clueInFocus())
				}
			},
			{
				key: "focusPreviousClue",
				value: function ()
				{
					var t = this.indexOfClueInFocus(),
						e = this.props.data.entries;
					if (-1 !== t)
					{
						var n = e[0 === t ? e.length - 1 : t - 1];
						this.focusClue(n.position.x, n.position.y, n.direction)
					}
				}
			},
			{
				key: "focusNextClue",
				value: function ()
				{
					var t = this.indexOfClueInFocus(),
						e = this.props.data.entries;
					if (-1 !== t)
					{
						var n = e[t === e.length - 1 ? 0 : t + 1];
						this.focusClue(n.position.x, n.position.y, n.direction)
					}
				}
			},
			{
				key: "moveFocus",
				value: function (t, e)
				{
					var n = this.state.cellInFocus;
					if (n)
					{
						var r = n.x + t,
							o = n.y + e,
							i = "down";
						this.state.grid[r] && this.state.grid[r][o] && this.state.grid[r][o].isEditable && (0 !== e ? i = "down" : 0 !== t && (i = "across"), this.focusClue(r, o, i))
					}
				}
			},
			{
				key: "isAcross",
				value: function ()
				{
					return "across" === this.state.directionOfEntry
				}
			},
			{
				key: "focusPrevious",
				value: function ()
				{
					var t = this.state.cellInFocus,
						e = this.clueInFocus();
					if (t && e)
						if (function (t, e)
							{
								var n = tt(e) ? "x" : "y";
								return t[n] === e.position[n]
							}(t, e))
						{
							var n = function (t, e)
							{
								var n = e.group[e.group.findIndex(function (t)
								{
									return t === e.id
								}) - 1];
								return t.find(function (t)
								{
									return t.id === n
								})
							}(this.props.data.entries, e);
							if (n)
							{
								var r = function (t)
								{
									var e, n = {
											true: "x",
											false: "y"
										},
										r = n[String(tt(t))],
										o = n[String(!tt(t))];
									return e = {}, q()(e, r, t.position[r] + (t.length - 1)), q()(e, o, t.position[o]), e
								}(n);
								this.focusClue(r.x, r.y, n.direction)
							}
						}
					else this.isAcross() ? this.moveFocus(-1, 0) : this.moveFocus(0, -1)
				}
			},
			{
				key: "focusNext",
				value: function ()
				{
					var t = this.state.cellInFocus,
						e = this.clueInFocus();
					if (t && e)
						if (function (t, e)
							{
								var n = tt(e) ? "x" : "y";
								return t[n] === e.position[n] + (e.length - 1)
							}(t, e))
						{
							var n = function (t, e)
							{
								var n = e.group[e.group.findIndex(function (t)
								{
									return t === e.id
								}) + 1];
								return t.find(function (t)
								{
									return t.id === n
								})
							}(this.props.data.entries, e);
							n && this.focusClue(n.position.x, n.position.y, n.direction)
						}
					else this.isAcross() ? this.moveFocus(1, 0) : this.moveFocus(0, 1)
				}
			},
			{
				key: "asPercentage",
				value: function (t, e)
				{
					return {
						x: 100 * t / pt(this.columns),
						y: 100 * e / pt(this.rows)
					}
				}
			},
			{
				key: "focusHiddenInput",
				value: function (t, e)
				{
					var n = Object(p.findDOMNode)(this.hiddenInputComponent.wrapper),
						r = pt(t),
						o = pt(e),
						i = this.asPercentage(r, o);
					n.style.left = i.x + "%", n.style.top = i.y + "%";
					var s = Object(p.findDOMNode)(this.hiddenInputComponent.input);
					document.activeElement !== s && s.focus()
				}
			},
			{
				key: "focusClue",
				value: function (t, e, n)
				{
					var r = ct(this.clueMap, t, e),
						o = r[n];
					r && o && (this.focusHiddenInput(t, e), this.setState(
					{
						grid: this.state.grid,
						cellInFocus:
						{
							x: t,
							y: e
						},
						directionOfEntry: n
					}), window.history.replaceState(void 0, document.title, "#" + o.id))
				}
			},
			{
				key: "focusFirstCellInClue",
				value: function (t)
				{
					this.focusClue(t.position.x, t.position.y, t.direction)
				}
			},
			{
				key: "focusFirstCellInClueById",
				value: function (t)
				{
					var e = this.props.data.entries.find(function (e)
					{
						return e.id === t
					});
					e && this.focusFirstCellInClue(e)
				}
			},
			{
				key: "focusCurrentCell",
				value: function ()
				{
					this.state.cellInFocus && this.focusHiddenInput(this.state.cellInFocus.x, this.state.cellInFocus.y)
				}
			},
			{
				key: "clueInFocus",
				value: function ()
				{
					if (this.state.cellInFocus)
					{
						var t = ct(this.clueMap, this.state.cellInFocus.x, this.state.cellInFocus.y);
						if (this.state.directionOfEntry) return t[this.state.directionOfEntry]
					}
					return null
				}
			},
			{
				key: "allHighlightedClues",
				value: function ()
				{
					var t = this;
					return this.props.data.entries.filter(function (e)
					{
						return t.clueIsInFocusGroup(e)
					})
				}
			},
			{
				key: "clueIsInFocusGroup",
				value: function (t)
				{
					if (this.state.cellInFocus)
					{
						var e = ct(this.clueMap, this.state.cellInFocus.x, this.state.cellInFocus.y);
						if (this.state.directionOfEntry && e[this.state.directionOfEntry]) return e[this.state.directionOfEntry].group.includes(t.id)
					}
					return !1
				}
			},
			{
				key: "cluesData",
				value: function ()
				{
					var t = this;
					return this.props.data.entries.map(function (e)
					{
						return {
							entry: e,
							hasAnswered: it(t.state.grid, e),
							isSelected: t.clueIsInFocusGroup(e)
						}
					})
				}
			},
			{
				key: "cheat",
				value: function (t)
				{
					var e = this,
						n = ot(t);
					t.solution && this.setState(
					{
						grid: ht(this.state.grid, function (r, o, i)
						{
							if (n.some(function (t)
								{
									return t.x === o && t.y === i
								}))
							{
								var s = "across" === t.direction ? o - t.position.x : i - t.position.y,
									a = r.value;
								r.value = t.solution[s], e.props.onMove(
								{
									x: o,
									y: i,
									value: r.value,
									previousValue: a
								})
							}
							return r
						})
					})
				}
			},
			{
				key: "check",
				value: function (t)
				{
					var e = this,
						n = ot(t);
					if (t.solution)
					{
						var r = Ot()(n, t.solution.split("")).filter(function (t)
						{
							var n = t[0],
								r = e.state.grid[n.x][n.y],
								o = t[1];
							return /^.$/.test(r.value) && r.value !== o
						}).map(function (t)
						{
							return t[0]
						});
						this.setState(
						{
							grid: ht(this.state.grid, function (t, n, o)
							{
								if (r.some(function (t)
									{
										return t.x === n && t.y === o
									}))
								{
									var i = t.value;
									t.value = "", e.props.onMove(
									{
										x: n,
										y: o,
										value: "",
										previousValue: i
									})
								}
								return t
							})
						})
					}
				}
			},
			{
				key: "hiddenInputValue",
				value: function ()
				{
					var t = this.state.cellInFocus,
						e = void 0;
					return t && (e = this.state.grid[t.x][t.y].value), e || ""
				}
			},
			{
				key: "hasSolutions",
				value: function ()
				{
					return "solution" in this.props.data.entries[0]
				}
			},
			{
				key: "isHighlighted",
				value: function (t, e)
				{
					var n = this,
						r = this.clueInFocus();
					return !!r && r.group.some(function (r)
					{
						var o = n.props.data.entries.find(function (t)
						{
							return t.id === r
						});
						return dt(o, t, e)
					})
				}
			},
			{
				key: "saveGrid",
				value: function ()
				{
					var t = this.state.grid.map(function (t)
					{
						return t.map(function (t)
						{
							return t.value
						})
					});
					this.props.saveGrid(this.props.id, t)
				}
			},
			{
				key: "render",
				value: function ()
				{
					var t, e = this,
						n = this.clueInFocus(),
						r = this.state.showAnagramHelper && d.a.createElement(bt,
						{
							key: n.id,
							crossword: this,
							focussedEntry: n,
							entries: this.props.data.entries,
							grid: this.state.grid,
							close: this.onToggleAnagramHelper
						}),
						o = {
							rows: this.rows,
							columns: this.columns,
							cells: this.state.grid,
							separators: (t = this.props.data.entries, t.map(function (t)
							{
								return Object.keys(t.separatorLocations).map(function (e)
								{
									var n = e;
									return t.separatorLocations[n].map(function (e)
									{
										return {
											key: tt(t) ? at(t.position.x + e, t.position.y) : at(t.position.x, t.position.y + e),
											direction: t.direction,
											separator: n
										}
									})
								})
							}).reduce(function t(e, n)
							{
								var r = n;
								return Array.isArray(r) && r.length && (r = r.reduce(t, [])), e.concat(r)
							}, []).reduce(function (t, e)
							{
								return e ? (void 0 === t[e.key] && (t[e.key] = {}), t[e.key] = e, t) : t
							},
							{})),
							crossword: this,
							focussedCell: this.state.cellInFocus,
							ref: function (t)
							{
								e.grid = t
							}
						};
					return d.a.createElement("div",
					{
						className: "crossword__container crossword__container--" + this.props.data.crosswordType + " crossword__container--react",
						"data-link-name": "Crosswords"
					}, d.a.createElement("div",
					{
						className: "crossword__container__game",
						ref: function (t)
						{
							e.game = t
						}
					}, d.a.createElement("div",
					{
						className: "crossword__sticky-clue-wrapper",
						ref: function (t)
						{
							e.stickyClueWrapper = t
						}
					}, d.a.createElement("div",
					{
						className: St(
						{
							"crossword__sticky-clue": !0,
							"is-hidden": !n
						})
					}, n && d.a.createElement("div",
					{
						className: "crossword__sticky-clue__inner"
					}, d.a.createElement("div",
					{
						className: "crossword__sticky-clue__inner__inner"
					}, d.a.createElement("strong", null, n.number, " ", d.a.createElement("span",
					{
						className: "crossword__sticky-clue__direction"
					}, n.direction)), " ", n.clue)))), d.a.createElement("div",
					{
						className: "crossword__container__grid-wrapper",
						ref: function (t)
						{
							e.gridWrapper = t
						}
					}, Ft(o), d.a.createElement(Lt,
					{
						crossword: this,
						value: this.hiddenInputValue(),
						ref: function (t)
						{
							e.hiddenInputComponent = t
						}
					}), r)), d.a.createElement(Tt,
					{
						hasSolutions: this.hasSolutions(),
						clueInFocus: n,
						crossword: this
					}), d.a.createElement(It,
					{
						clues: this.cluesData(),
						focussed: n,
						focusFirstCellInClueById: this.focusFirstCellInClueById.bind(this),
						setReturnPosition: this.setReturnPosition.bind(this)
					}))
				}
			}]), e
		}(f.Component);
	Wt.defaultProps = {
		onMove: function () {},
		loadGrid: function (t)
		{
			return function (t)
			{
				return qt.get(Vt(t))
			}(t)
		},
		saveGrid: function (t, e)
		{
			return function (t, e)
			{
				try
				{
					return qt.set(Vt(t), e)
				}
				catch (t)
				{
					return !1
				}
			}(t, e)
		}
	};
	e.default = Wt
}, function (t, e, n)
{
	var r = n(47),
		o = n(18);
	t.exports = function (t)
	{
		return o(t) && r(t)
	}
}, function (t, e, n)
{
	var r = n(79),
		o = n(39),
		i = n(69),
		s = n(78),
		a = n(111),
		c = Math.max;
	t.exports = function (t)
	{
		if (!t || !t.length) return [];
		var e = 0;
		return t = r(t, function (t)
		{
			if (a(t)) return e = c(t.length, e), !0
		}), s(e, function (e)
		{
			return o(t, i(e))
		})
	}
}, function (t, e)
{
	var n = 800,
		r = 16,
		o = Date.now;
	t.exports = function (t)
	{
		var e = 0,
			i = 0;
		return function ()
		{
			var s = o(),
				a = r - (s - i);
			if (i = s, a > 0)
			{
				if (++e >= n) return arguments[0]
			}
			else e = 0;
			return t.apply(void 0, arguments)
		}
	}
}, function (t, e, n)
{
	var r = n(14),
		o = function ()
		{
			try
			{
				var t = r(Object, "defineProperty");
				return t(
				{}, "",
				{}), t
			}
			catch (t)
			{}
		}();
	t.exports = o
}, function (t, e)
{
	t.exports = function (t)
	{
		return function ()
		{
			return t
		}
	}
}, function (t, e, n)
{
	var r = n(115),
		o = n(114),
		i = n(38),
		s = o ? function (t, e)
		{
			return o(t, "toString",
			{
				configurable: !0,
				enumerable: !1,
				value: r(e),
				writable: !0
			})
		} : i;
	t.exports = s
}, function (t, e, n)
{
	var r = n(116),
		o = n(113)(r);
	t.exports = o
}, function (t, e)
{
	t.exports = function (t, e, n)
	{
		switch (n.length)
		{
		case 0:
			return t.call(e);
		case 1:
			return t.call(e, n[0]);
		case 2:
			return t.call(e, n[0], n[1]);
		case 3:
			return t.call(e, n[0], n[1], n[2])
		}
		return t.apply(e, n)
	}
}, function (t, e, n)
{
	var r = n(118),
		o = Math.max;
	t.exports = function (t, e, n)
	{
		return e = o(void 0 === e ? t.length - 1 : e, 0),
			function ()
			{
				for (var i = arguments, s = -1, a = o(i.length - e, 0), c = Array(a); ++s < a;) c[s] = i[e + s];
				s = -1;
				for (var u = Array(e + 1); ++s < e;) u[s] = i[s];
				return u[e] = n(c), r(t, this, u)
			}
	}
}, function (t, e, n)
{
	var r = n(38),
		o = n(119),
		i = n(117);
	t.exports = function (t, e)
	{
		return i(o(t, e, r), t + "")
	}
}, function (t, e, n)
{
	var r = n(7);
	t.exports = function ()
	{
		return r.Date.now()
	}
}, function (t, e, n)
{
	var r = n(39);
	t.exports = function (t, e)
	{
		return r(e, function (e)
		{
			return t[e]
		})
	}
}, function (t, e, n)
{
	var r = n(122),
		o = n(41);
	t.exports = function (t)
	{
		return null == t ? [] : r(t, o(t))
	}
}, function (t, e, n)
{
	var r = n(68),
		o = n(123);
	t.exports = function (t)
	{
		return r(o(t))
	}
}, function (t, e)
{
	var n = Math.floor,
		r = Math.random;
	t.exports = function (t, e)
	{
		return t + n(r() * (e - t + 1))
	}
}, function (t, e)
{
	t.exports = function (t, e)
	{
		var n = -1,
			r = t.length;
		for (e || (e = Array(r)); ++n < r;) e[n] = t[n];
		return e
	}
}, function (t, e, n)
{
	var r = n(126),
		o = n(68);
	t.exports = function (t)
	{
		return o(r(t))
	}
}, function (t, e)
{
	t.exports = function () {}
}, function (t, e, n)
{
	var r = n(74),
		o = n(128),
		i = n(42),
		s = r && 1 / i(new r([, -0]))[1] == 1 / 0 ? function (t)
		{
			return new r(t)
		} : o;
	t.exports = s
}, function (t, e)
{
	t.exports = function (t, e, n)
	{
		for (var r = -1, o = null == t ? 0 : t.length; ++r < o;)
			if (n(e, t[r])) return !0;
		return !1
	}
}, function (t, e)
{
	t.exports = function (t, e, n)
	{
		for (var r = n - 1, o = t.length; ++r < o;)
			if (t[r] === e) return r;
		return -1
	}
}, function (t, e)
{
	t.exports = function (t)
	{
		return t != t
	}
}, function (t, e)
{
	t.exports = function (t, e, n, r)
	{
		for (var o = t.length, i = n + (r ? 1 : -1); r ? i-- : ++i < o;)
			if (e(t[i], i, t)) return i;
		return -1
	}
}, function (t, e, n)
{
	var r = n(133),
		o = n(132),
		i = n(131);
	t.exports = function (t, e, n)
	{
		return e == e ? i(t, e, n) : r(t, o, n)
	}
}, function (t, e, n)
{
	var r = n(134);
	t.exports = function (t, e)
	{
		return !(null == t || !t.length) && r(t, e, 0) > -1
	}
}, function (t, e, n)
{
	var r = n(81),
		o = n(135),
		i = n(130),
		s = n(80),
		a = n(129),
		c = n(42),
		u = 200;
	t.exports = function (t, e, n)
	{
		var l = -1,
			f = o,
			d = t.length,
			p = !0,
			h = [],
			v = h;
		if (n) p = !1, f = i;
		else if (d >= u)
		{
			var _ = e ? null : a(t);
			if (_) return c(_);
			p = !1, f = s, v = new r
		}
		else v = e ? [] : h;
		t: for (; ++l < d;)
		{
			var m = t[l],
				g = e ? e(m) : m;
			if (m = n || 0 !== m ? m : 0, p && g == g)
			{
				for (var y = v.length; y--;)
					if (v[y] === g) continue t;
				e && v.push(g), h.push(m)
			}
			else f(v, g, n) || (v !== h && v.push(g), h.push(m))
		}
		return h
	}
}, function (t, e, n)
{
	var r = n(71);
	t.exports = function (t)
	{
		return function (e)
		{
			return r(e, t)
		}
	}
}, function (t, e, n)
{
	var r = n(69),
		o = n(137),
		i = n(40),
		s = n(29);
	t.exports = function (t)
	{
		return i(t) ? r(s(t)) : o(t)
	}
}, function (t, e, n)
{
	var r = n(70),
		o = n(49),
		i = n(6),
		s = n(45),
		a = n(46),
		c = n(29);
	t.exports = function (t, e, n)
	{
		for (var u = -1, l = (e = r(e, t)).length, f = !1; ++u < l;)
		{
			var d = c(e[u]);
			if (!(f = null != t && n(t, d))) break;
			t = t[d]
		}
		return f || ++u != l ? f : !!(l = null == t ? 0 : t.length) && a(l) && s(d, l) && (i(t) || o(t))
	}
}, function (t, e)
{
	t.exports = function (t, e)
	{
		return null != t && e in Object(t)
	}
}, function (t, e, n)
{
	var r = n(140),
		o = n(139);
	t.exports = function (t, e)
	{
		return null != t && o(t, e, r)
	}
}, function (t, e, n)
{
	var r = n(25),
		o = n(39),
		i = n(6),
		s = n(34),
		a = 1 / 0,
		c = r ? r.prototype : void 0,
		u = c ? c.toString : void 0;
	t.exports = function t(e)
	{
		if ("string" == typeof e) return e;
		if (i(e)) return o(e, t) + "";
		if (s(e)) return u ? u.call(e) : "";
		var n = e + "";
		return "0" == n && 1 / e == -a ? "-0" : n
	}
}, function (t, e, n)
{
	var r = n(142);
	t.exports = function (t)
	{
		return null == t ? "" : r(t)
	}
}, function (t, e, n)
{
	var r = n(43),
		o = "Expected a function";

	function i(t, e)
	{
		if ("function" != typeof t || null != e && "function" != typeof e) throw new TypeError(o);
		var n = function ()
		{
			var r = arguments,
				o = e ? e.apply(this, r) : r[0],
				i = n.cache;
			if (i.has(o)) return i.get(o);
			var s = t.apply(this, r);
			return n.cache = i.set(o, s) || i, s
		};
		return n.cache = new(i.Cache || r), n
	}
	i.Cache = r, t.exports = i
}, function (t, e, n)
{
	var r = n(144),
		o = 500;
	t.exports = function (t)
	{
		var e = r(t, function (t)
			{
				return n.size === o && n.clear(), t
			}),
			n = e.cache;
		return e
	}
}, function (t, e, n)
{
	var r = /[^.[\]]+|\[(?:(-?\d+(?:\.\d+)?)|(["'])((?:(?!\2)[^\\]|\\.)*?)\2)\]|(?=(?:\.|\[\])(?:\.|\[\]|$))/g,
		o = /\\(\\)?/g,
		i = n(145)(function (t)
		{
			var e = [];
			return 46 === t.charCodeAt(0) && e.push(""), t.replace(r, function (t, n, r, i)
			{
				e.push(r ? i.replace(o, "$1") : n || t)
			}), e
		});
	t.exports = i
}, function (t, e, n)
{
	var r = n(71);
	t.exports = function (t, e, n)
	{
		var o = null == t ? void 0 : r(t, e);
		return void 0 === o ? n : o
	}
}, function (t, e, n)
{
	var r = n(83),
		o = n(147),
		i = n(141),
		s = n(40),
		a = n(73),
		c = n(72),
		u = n(29),
		l = 1,
		f = 2;
	t.exports = function (t, e)
	{
		return s(t) && a(e) ? c(u(t), e) : function (n)
		{
			var s = o(n, t);
			return void 0 === s && s === e ? i(n, t) : r(e, s, l | f)
		}
	}
}, function (t, e, n)
{
	var r = n(73),
		o = n(41);
	t.exports = function (t)
	{
		for (var e = o(t), n = e.length; n--;)
		{
			var i = e[n],
				s = t[i];
			e[n] = [i, s, r(s)]
		}
		return e
	}
}, function (t, e, n)
{
	var r = n(14)(n(7), "WeakMap");
	t.exports = r
}, function (t, e, n)
{
	var r = n(14)(n(7), "Promise");
	t.exports = r
}, function (t, e, n)
{
	var r = n(14)(n(7), "DataView");
	t.exports = r
}, function (t, e, n)
{
	var r = n(152),
		o = n(44),
		i = n(151),
		s = n(74),
		a = n(150),
		c = n(24),
		u = n(84),
		l = u(r),
		f = u(o),
		d = u(i),
		p = u(s),
		h = u(a),
		v = c;
	(r && "[object DataView]" != v(new r(new ArrayBuffer(1))) || o && "[object Map]" != v(new o) || i && "[object Promise]" != v(i.resolve()) || s && "[object Set]" != v(new s) || a && "[object WeakMap]" != v(new a)) && (v = function (t)
	{
		var e = c(t),
			n = "[object Object]" == e ? t.constructor : void 0,
			r = n ? u(n) : "";
		if (r) switch (r)
		{
		case l:
			return "[object DataView]";
		case f:
			return "[object Map]";
		case d:
			return "[object Promise]";
		case p:
			return "[object Set]";
		case h:
			return "[object WeakMap]"
		}
		return e
	}), t.exports = v
}, function (t, e)
{
	t.exports = function (t, e)
	{
		return function (n)
		{
			return t(e(n))
		}
	}
}, function (t, e, n)
{
	var r = n(154)(Object.keys, Object);
	t.exports = r
}, function (t, e)
{
	var n = Object.prototype;
	t.exports = function (t)
	{
		var e = t && t.constructor;
		return t === ("function" == typeof e && e.prototype || n)
	}
}, function (t, e, n)
{
	var r = n(156),
		o = n(155),
		i = Object.prototype.hasOwnProperty;
	t.exports = function (t)
	{
		if (!r(t)) return o(t);
		var e = [];
		for (var n in Object(t)) i.call(t, n) && "constructor" != n && e.push(n);
		return e
	}
}, function (t, e, n)
{
	(function (t)
	{
		var r = n(88),
			o = "object" == typeof e && e && !e.nodeType && e,
			i = o && "object" == typeof t && t && !t.nodeType && t,
			s = i && i.exports === o && r.process,
			a = function ()
			{
				try
				{
					var t = i && i.require && i.require("util").types;
					return t || s && s.binding && s.binding("util")
				}
				catch (t)
				{}
			}();
		t.exports = a
	}).call(this, n(76)(t))
}, function (t, e)
{
	t.exports = function (t)
	{
		return function (e)
		{
			return t(e)
		}
	}
}, function (t, e, n)
{
	var r = n(24),
		o = n(46),
		i = n(18),
		s = {};
	s["[object Float32Array]"] = s["[object Float64Array]"] = s["[object Int8Array]"] = s["[object Int16Array]"] = s["[object Int32Array]"] = s["[object Uint8Array]"] = s["[object Uint8ClampedArray]"] = s["[object Uint16Array]"] = s["[object Uint32Array]"] = !0, s["[object Arguments]"] = s["[object Array]"] = s["[object ArrayBuffer]"] = s["[object Boolean]"] = s["[object DataView]"] = s["[object Date]"] = s["[object Error]"] = s["[object Function]"] = s["[object Map]"] = s["[object Number]"] = s["[object Object]"] = s["[object RegExp]"] = s["[object Set]"] = s["[object String]"] = s["[object WeakMap]"] = !1, t.exports = function (t)
	{
		return i(t) && o(t.length) && !!s[r(t)]
	}
}, function (t, e)
{
	t.exports = function ()
	{
		return !1
	}
}, function (t, e, n)
{
	var r = n(78),
		o = n(49),
		i = n(6),
		s = n(77),
		a = n(45),
		c = n(75),
		u = Object.prototype.hasOwnProperty;
	t.exports = function (t, e)
	{
		var n = i(t),
			l = !n && o(t),
			f = !n && !l && s(t),
			d = !n && !l && !f && c(t),
			p = n || l || f || d,
			h = p ? r(t.length, String) : [],
			v = h.length;
		for (var _ in t) !e && !u.call(t, _) || p && ("length" == _ || f && ("offset" == _ || "parent" == _) || d && ("buffer" == _ || "byteLength" == _ || "byteOffset" == _) || a(_, v)) || h.push(_);
		return h
	}
}, function (t, e)
{
	t.exports = function ()
	{
		return []
	}
}, function (t, e, n)
{
	var r = n(79),
		o = n(163),
		i = Object.prototype.propertyIsEnumerable,
		s = Object.getOwnPropertySymbols,
		a = s ? function (t)
		{
			return null == t ? [] : (t = Object(t), r(s(t), function (e)
			{
				return i.call(t, e)
			}))
		} : o;
	t.exports = a
}, function (t, e, n)
{
	var r = n(89),
		o = n(6);
	t.exports = function (t, e, n)
	{
		var i = e(t);
		return o(t) ? i : r(i, n(t))
	}
}, function (t, e, n)
{
	var r = n(165),
		o = n(164),
		i = n(41);
	t.exports = function (t)
	{
		return r(t, i, o)
	}
}, function (t, e, n)
{
	var r = n(166),
		o = 1,
		i = Object.prototype.hasOwnProperty;
	t.exports = function (t, e, n, s, a, c)
	{
		var u = n & o,
			l = r(t),
			f = l.length;
		if (f != r(e).length && !u) return !1;
		for (var d = f; d--;)
		{
			var p = l[d];
			if (!(u ? p in e : i.call(e, p))) return !1
		}
		var h = c.get(t);
		if (h && c.get(e)) return h == e;
		var v = !0;
		c.set(t, e), c.set(e, t);
		for (var _ = u; ++d < f;)
		{
			var m = t[p = l[d]],
				g = e[p];
			if (s) var y = u ? s(g, m, p, e, t, c) : s(m, g, p, t, e, c);
			if (!(void 0 === y ? m === g || a(m, g, n, s, c) : y))
			{
				v = !1;
				break
			}
			_ || (_ = "constructor" == p)
		}
		if (v && !_)
		{
			var w = t.constructor,
				b = e.constructor;
			w != b && "constructor" in t && "constructor" in e && !("function" == typeof w && w instanceof w && "function" == typeof b && b instanceof b) && (v = !1)
		}
		return c.delete(t), c.delete(e), v
	}
}, function (t, e)
{
	t.exports = function (t)
	{
		var e = -1,
			n = Array(t.size);
		return t.forEach(function (t, r)
		{
			n[++e] = [r, t]
		}), n
	}
}, function (t, e, n)
{
	var r = n(7).Uint8Array;
	t.exports = r
}, function (t, e, n)
{
	var r = n(25),
		o = n(169),
		i = n(48),
		s = n(82),
		a = n(168),
		c = n(42),
		u = 1,
		l = 2,
		f = "[object Boolean]",
		d = "[object Date]",
		p = "[object Error]",
		h = "[object Map]",
		v = "[object Number]",
		_ = "[object RegExp]",
		m = "[object Set]",
		g = "[object String]",
		y = "[object Symbol]",
		w = "[object ArrayBuffer]",
		b = "[object DataView]",
		x = r ? r.prototype : void 0,
		k = x ? x.valueOf : void 0;
	t.exports = function (t, e, n, r, x, E, O)
	{
		switch (n)
		{
		case b:
			if (t.byteLength != e.byteLength || t.byteOffset != e.byteOffset) return !1;
			t = t.buffer, e = e.buffer;
		case w:
			return !(t.byteLength != e.byteLength || !E(new o(t), new o(e)));
		case f:
		case d:
		case v:
			return i(+t, +e);
		case p:
			return t.name == e.name && t.message == e.message;
		case _:
		case g:
			return t == e + "";
		case h:
			var C = a;
		case m:
			var j = r & u;
			if (C || (C = c), t.size != e.size && !j) return !1;
			var S = O.get(t);
			if (S) return S == e;
			r |= l, O.set(t, e);
			var N = s(C(t), C(e), r, x, E, O);
			return O.delete(t), N;
		case y:
			if (k) return k.call(t) == k.call(e)
		}
		return !1
	}
}, function (t, e)
{
	t.exports = function (t, e)
	{
		for (var n = -1, r = null == t ? 0 : t.length; ++n < r;)
			if (e(t[n], n, t)) return !0;
		return !1
	}
}, function (t, e)
{
	t.exports = function (t)
	{
		return this.__data__.has(t)
	}
}, function (t, e)
{
	var n = "__lodash_hash_undefined__";
	t.exports = function (t)
	{
		return this.__data__.set(t, n), this
	}
}, function (t, e, n)
{
	var r = n(85),
		o = n(82),
		i = n(170),
		s = n(167),
		a = n(153),
		c = n(6),
		u = n(77),
		l = n(75),
		f = 1,
		d = "[object Arguments]",
		p = "[object Array]",
		h = "[object Object]",
		v = Object.prototype.hasOwnProperty;
	t.exports = function (t, e, n, _, m, g)
	{
		var y = c(t),
			w = c(e),
			b = y ? p : a(t),
			x = w ? p : a(e),
			k = (b = b == d ? h : b) == h,
			E = (x = x == d ? h : x) == h,
			O = b == x;
		if (O && u(t))
		{
			if (!u(e)) return !1;
			y = !0, k = !1
		}
		if (O && !k) return g || (g = new r), y || l(t) ? o(t, e, n, _, m, g) : i(t, e, b, n, _, m, g);
		if (!(n & f))
		{
			var C = k && v.call(t, "__wrapped__"),
				j = E && v.call(e, "__wrapped__");
			if (C || j)
			{
				var S = C ? t.value() : t,
					N = j ? e.value() : e;
				return g || (g = new r), m(S, N, n, _, g)
			}
		}
		return !!O && (g || (g = new r), s(t, e, n, _, m, g))
	}
}, function (t, e, n)
{
	var r = n(30);
	t.exports = function (t, e)
	{
		var n = r(this, t),
			o = n.size;
		return n.set(t, e), this.size += n.size == o ? 0 : 1, this
	}
}, function (t, e, n)
{
	var r = n(30);
	t.exports = function (t)
	{
		return r(this, t).has(t)
	}
}, function (t, e, n)
{
	var r = n(30);
	t.exports = function (t)
	{
		return r(this, t).get(t)
	}
}, function (t, e)
{
	t.exports = function (t)
	{
		var e = typeof t;
		return "string" == e || "number" == e || "symbol" == e || "boolean" == e ? "__proto__" !== t : null === t
	}
}, function (t, e, n)
{
	var r = n(30);
	t.exports = function (t)
	{
		var e = r(this, t).delete(t);
		return this.size -= e ? 1 : 0, e
	}
}, function (t, e, n)
{
	var r = n(31),
		o = "__lodash_hash_undefined__";
	t.exports = function (t, e)
	{
		var n = this.__data__;
		return this.size += this.has(t) ? 0 : 1, n[t] = r && void 0 === e ? o : e, this
	}
}, function (t, e, n)
{
	var r = n(31),
		o = Object.prototype.hasOwnProperty;
	t.exports = function (t)
	{
		var e = this.__data__;
		return r ? void 0 !== e[t] : o.call(e, t)
	}
}, function (t, e, n)
{
	var r = n(31),
		o = "__lodash_hash_undefined__",
		i = Object.prototype.hasOwnProperty;
	t.exports = function (t)
	{
		var e = this.__data__;
		if (r)
		{
			var n = e[t];
			return n === o ? void 0 : n
		}
		return i.call(e, t) ? e[t] : void 0
	}
}, function (t, e)
{
	t.exports = function (t)
	{
		var e = this.has(t) && delete this.__data__[t];
		return this.size -= e ? 1 : 0, e
	}
}, function (t, e, n)
{
	var r = n(31);
	t.exports = function ()
	{
		this.__data__ = r ? r(null) :
		{}, this.size = 0
	}
}, function (t, e, n)
{
	var r = n(184),
		o = n(183),
		i = n(182),
		s = n(181),
		a = n(180);

	function c(t)
	{
		var e = -1,
			n = null == t ? 0 : t.length;
		for (this.clear(); ++e < n;)
		{
			var r = t[e];
			this.set(r[0], r[1])
		}
	}
	c.prototype.clear = r, c.prototype.delete = o, c.prototype.get = i, c.prototype.has = s, c.prototype.set = a, t.exports = c
}, function (t, e, n)
{
	var r = n(185),
		o = n(33),
		i = n(44);
	t.exports = function ()
	{
		this.size = 0, this.__data__ = {
			hash: new r,
			map: new(i || o),
			string: new r
		}
	}
}, function (t, e)
{
	t.exports = function (t, e)
	{
		return null == t ? void 0 : t[e]
	}
}, function (t, e, n)
{
	var r = n(7)["__core-js_shared__"];
	t.exports = r
}, function (t, e, n)
{
	var r, o = n(188),
		i = (r = /[^.]+$/.exec(o && o.keys && o.keys.IE_PROTO || "")) ? "Symbol(src)_1." + r : "";
	t.exports = function (t)
	{
		return !!i && i in t
	}
}, function (t, e, n)
{
	var r = n(87),
		o = n(189),
		i = n(17),
		s = n(84),
		a = /^\[object .+?Constructor\]$/,
		c = Function.prototype,
		u = Object.prototype,
		l = c.toString,
		f = u.hasOwnProperty,
		d = RegExp("^" + l.call(f).replace(/[\\^$.*+?()[\]{}|]/g, "\\$&").replace(/hasOwnProperty|(function).*?(?=\\\()| for .+?(?=\\\])/g, "$1.*?") + "$");
	t.exports = function (t)
	{
		return !(!i(t) || o(t)) && (r(t) ? d : a).test(s(t))
	}
}, function (t, e, n)
{
	var r = n(33),
		o = n(44),
		i = n(43),
		s = 200;
	t.exports = function (t, e)
	{
		var n = this.__data__;
		if (n instanceof r)
		{
			var a = n.__data__;
			if (!o || a.length < s - 1) return a.push([t, e]), this.size = ++n.size, this;
			n = this.__data__ = new i(a)
		}
		return n.set(t, e), this.size = n.size, this
	}
}, function (t, e)
{
	t.exports = function (t)
	{
		return this.__data__.has(t)
	}
}, function (t, e)
{
	t.exports = function (t)
	{
		return this.__data__.get(t)
	}
}, function (t, e)
{
	t.exports = function (t)
	{
		var e = this.__data__,
			n = e.delete(t);
		return this.size = e.size, n
	}
}, function (t, e, n)
{
	var r = n(33);
	t.exports = function ()
	{
		this.__data__ = new r, this.size = 0
	}
}, function (t, e, n)
{
	var r = n(32);
	t.exports = function (t, e)
	{
		var n = this.__data__,
			o = r(n, t);
		return o < 0 ? (++this.size, n.push([t, e])) : n[o][1] = e, this
	}
}, function (t, e, n)
{
	var r = n(32);
	t.exports = function (t)
	{
		return r(this.__data__, t) > -1
	}
}, function (t, e, n)
{
	var r = n(32);
	t.exports = function (t)
	{
		var e = this.__data__,
			n = r(e, t);
		return n < 0 ? void 0 : e[n][1]
	}
}, function (t, e, n)
{
	var r = n(32),
		o = Array.prototype.splice;
	t.exports = function (t)
	{
		var e = this.__data__,
			n = r(e, t);
		return !(n < 0 || (n == e.length - 1 ? e.pop() : o.call(e, n, 1), --this.size, 0))
	}
}, function (t, e)
{
	t.exports = function ()
	{
		this.__data__ = [], this.size = 0
	}
}, function (t, e, n)
{
	var r = n(85),
		o = n(83),
		i = 1,
		s = 2;
	t.exports = function (t, e, n, a)
	{
		var c = n.length,
			u = c,
			l = !a;
		if (null == t) return !u;
		for (t = Object(t); c--;)
		{
			var f = n[c];
			if (l && f[2] ? f[1] !== t[f[0]] : !(f[0] in t)) return !1
		}
		for (; ++c < u;)
		{
			var d = (f = n[c])[0],
				p = t[d],
				h = f[1];
			if (l && f[2])
			{
				if (void 0 === p && !(d in t)) return !1
			}
			else
			{
				var v = new r;
				if (a) var _ = a(p, h, d, t, e, v);
				if (!(void 0 === _ ? o(h, p, i | s, a, v) : _)) return !1
			}
		}
		return !0
	}
}, function (t, e, n)
{
	var r = n(201),
		o = n(149),
		i = n(72);
	t.exports = function (t)
	{
		var e = o(t);
		return 1 == e.length && e[0][2] ? i(e[0][0], e[0][1]) : function (n)
		{
			return n === t || r(n, t, e)
		}
	}
}, function (t, e, n)
{
	var r = n(202),
		o = n(148),
		i = n(38),
		s = n(6),
		a = n(138);
	t.exports = function (t)
	{
		return "function" == typeof t ? t : null == t ? i : "object" == typeof t ? s(t) ? o(t[0], t[1]) : r(t) : a(t)
	}
}, function (t, e, n)
{
	var r = n(86),
		o = 1 / 0,
		i = 1.7976931348623157e308;
	t.exports = function (t)
	{
		return t ? (t = r(t)) === o || t === -o ? (t < 0 ? -1 : 1) * i : t == t ? t : 0 : 0 === t ? t : 0
	}
}, function (t, e, n)
{
	var r = n(48),
		o = n(47),
		i = n(45),
		s = n(17);
	t.exports = function (t, e, n)
	{
		if (!s(n)) return !1;
		var a = typeof e;
		return !!("number" == a ? o(n) && i(e, n.length) : "string" == a && e in n) && r(n[e], t)
	}
}, function (t, e)
{
	var n = Math.ceil,
		r = Math.max;
	t.exports = function (t, e, o, i)
	{
		for (var s = -1, a = r(n((e - t) / (o || 1)), 0), c = Array(a); a--;) c[i ? a : ++s] = t, t += o;
		return c
	}
}, function (t, e, n)
{
	var r = n(206),
		o = n(205),
		i = n(204);
	t.exports = function (t)
	{
		return function (e, n, s)
		{
			return s && "number" != typeof s && o(e, n, s) && (n = s = void 0), e = i(e), void 0 === n ? (n = e, e = 0) : n = i(n), s = void 0 === s ? e < n ? 1 : -1 : i(s), r(e, n, s, t)
		}
	}
}, function (t, e)
{
	var n = Object.prototype.toString;
	t.exports = function (t)
	{
		return n.call(t)
	}
}, function (t, e, n)
{
	var r = n(25),
		o = Object.prototype,
		i = o.hasOwnProperty,
		s = o.toString,
		a = r ? r.toStringTag : void 0;
	t.exports = function (t)
	{
		var e = i.call(t, a),
			n = t[a];
		try
		{
			t[a] = void 0;
			var r = !0
		}
		catch (t)
		{}
		var o = s.call(t);
		return r && (e ? t[a] = n : delete t[a]), o
	}
}, function (t, e, n)
{
	var r = n(24),
		o = n(18),
		i = "[object Arguments]";
	t.exports = function (t)
	{
		return o(t) && r(t) == i
	}
}, function (t, e)
{
	var n;
	n = function ()
	{
		return this
	}();
	try
	{
		n = n || Function("return this")() || (0, eval)("this")
	}
	catch (t)
	{
		"object" == typeof window && (n = window)
	}
	t.exports = n
}, function (t, e, n)
{
	var r = n(25),
		o = n(49),
		i = n(6),
		s = r ? r.isConcatSpreadable : void 0;
	t.exports = function (t)
	{
		return i(t) || o(t) || !!(s && t && t[s])
	}
}, function (t, e, n)
{
	var r = n(89),
		o = n(212);
	t.exports = function t(e, n, i, s, a)
	{
		var c = -1,
			u = e.length;
		for (i || (i = o), a || (a = []); ++c < u;)
		{
			var l = e[c];
			n > 0 && i(l) ? n > 1 ? t(l, n - 1, i, s, a) : r(a, l) : s || (a[a.length] = l)
		}
		return a
	}
}, function (t, e, n)
{
	var r = n(8)("iterator"),
		o = !1;
	try
	{
		var i = [7][r]();
		i.return = function ()
		{
			o = !0
		}, Array.from(i, function ()
		{
			throw 2
		})
	}
	catch (t)
	{}
	t.exports = function (t, e)
	{
		if (!e && !o) return !1;
		var n = !1;
		try
		{
			var i = [7],
				s = i[r]();
			s.next = function ()
			{
				return {
					done: n = !0
				}
			}, i[r] = function ()
			{
				return s
			}, t(i)
		}
		catch (t)
		{}
		return n
	}
}, function (t, e, n)
{
	var r = n(57),
		o = n(8)("toStringTag"),
		i = "Arguments" == r(function ()
		{
			return arguments
		}());
	t.exports = function (t)
	{
		var e, n, s;
		return void 0 === t ? "Undefined" : null === t ? "Null" : "string" == typeof (n = function (t, e)
		{
			try
			{
				return t[e]
			}
			catch (t)
			{}
		}(e = Object(t), o)) ? n : i ? r(e) : "Object" == (s = r(e)) && "function" == typeof e.callee ? "Arguments" : s
	}
}, function (t, e, n)
{
	var r = n(215),
		o = n(8)("iterator"),
		i = n(26);
	t.exports = n(11).getIteratorMethod = function (t)
	{
		if (void 0 != t) return t[o] || t["@@iterator"] || i[r(t)]
	}
}, function (t, e, n)
{
	"use strict";
	var r = n(10),
		o = n(27);
	t.exports = function (t, e, n)
	{
		e in t ? r.f(t, e, o(0, n)) : t[e] = n
	}
}, function (t, e, n)
{
	var r = n(26),
		o = n(8)("iterator"),
		i = Array.prototype;
	t.exports = function (t)
	{
		return void 0 !== t && (r.Array === t || i[o] === t)
	}
}, function (t, e, n)
{
	var r = n(21);
	t.exports = function (t, e, n, o)
	{
		try
		{
			return o ? e(r(n)[0], n[1]) : e(n)
		}
		catch (e)
		{
			var i = t.return;
			throw void 0 !== i && r(i.call(t)), e
		}
	}
}, function (t, e, n)
{
	"use strict";
	var r = n(64),
		o = n(23),
		i = n(93),
		s = n(219),
		a = n(218),
		c = n(94),
		u = n(217),
		l = n(216);
	o(o.S + o.F * !n(214)(function (t)
	{
		Array.from(t)
	}), "Array",
	{
		from: function (t)
		{
			var e, n, o, f, d = i(t),
				p = "function" == typeof this ? this : Array,
				h = arguments.length,
				v = h > 1 ? arguments[1] : void 0,
				_ = void 0 !== v,
				m = 0,
				g = l(d);
			if (_ && (v = r(v, h > 2 ? arguments[2] : void 0, 2)), void 0 == g || p == Array && a(g))
				for (n = new p(e = c(d.length)); e > m; m++) u(n, m, _ ? v(d[m], m) : d[m]);
			else
				for (f = g.call(d), n = new p; !(o = f.next()).done; m++) u(n, m, _ ? s(f, v, [o.value, m], !0) : o.value);
			return n.length = m, n
		}
	})
}, function (t, e, n)
{
	n(98), n(220), t.exports = n(11).Array.from
}, function (t, e, n)
{
	t.exports = {
		default: n(221),
		__esModule: !0
	}
}, function (t, e)
{
	t.exports = function (t)
	{
		var e = "undefined" != typeof window && window.location;
		if (!e) throw new Error("fixUrls requires window.location");
		if (!t || "string" != typeof t) return t;
		var n = e.protocol + "//" + e.host,
			r = n + e.pathname.replace(/\/[^\/]*$/, "/");
		return t.replace(/url\s*\(((?:[^)(]|\((?:[^)(]+|\([^)(]*\))*\))*)\)/gi, function (t, e)
		{
			var o, i = e.trim().replace(/^"(.*)"$/, function (t, e)
			{
				return e
			}).replace(/^'(.*)'$/, function (t, e)
			{
				return e
			});
			return /^(#|data:|http:\/\/|https:\/\/|file:\/\/\/|\s*$)/i.test(i) ? t : (o = 0 === i.indexOf("//") ? i : 0 === i.indexOf("/") ? n + i : r + i.replace(/^\.\//, ""), "url(" + JSON.stringify(o) + ")")
		})
	}
}, function (t, e, n)
{
	var r, o, i = {},
		s = (r = function ()
		{
			return window && document && document.all && !window.atob
		}, function ()
		{
			return void 0 === o && (o = r.apply(this, arguments)), o
		}),
		a = function (t)
		{
			var e = {};
			return function (t, n)
			{
				if ("function" == typeof t) return t();
				if (void 0 === e[t])
				{
					var r = function (t, e)
					{
						return e ? e.querySelector(t) : document.querySelector(t)
					}.call(this, t, n);
					if (window.HTMLIFrameElement && r instanceof window.HTMLIFrameElement) try
					{
						r = r.contentDocument.head
					}
					catch (t)
					{
						r = null
					}
					e[t] = r
				}
				return e[t]
			}
		}(),
		c = null,
		u = 0,
		l = [],
		f = n(223);

	function d(t, e)
	{
		for (var n = 0; n < t.length; n++)
		{
			var r = t[n],
				o = i[r.id];
			if (o)
			{
				o.refs++;
				for (var s = 0; s < o.parts.length; s++) o.parts[s](r.parts[s]);
				for (; s < r.parts.length; s++) o.parts.push(g(r.parts[s], e))
			}
			else
			{
				var a = [];
				for (s = 0; s < r.parts.length; s++) a.push(g(r.parts[s], e));
				i[r.id] = {
					id: r.id,
					refs: 1,
					parts: a
				}
			}
		}
	}

	function p(t, e)
	{
		for (var n = [], r = {}, o = 0; o < t.length; o++)
		{
			var i = t[o],
				s = e.base ? i[0] + e.base : i[0],
				a = {
					css: i[1],
					media: i[2],
					sourceMap: i[3]
				};
			r[s] ? r[s].parts.push(a) : n.push(r[s] = {
				id: s,
				parts: [a]
			})
		}
		return n
	}

	function h(t, e)
	{
		var n = a(t.insertInto);
		if (!n) throw new Error("Couldn't find a style target. This probably means that the value for the 'insertInto' parameter is invalid.");
		var r = l[l.length - 1];
		if ("top" === t.insertAt) r ? r.nextSibling ? n.insertBefore(e, r.nextSibling) : n.appendChild(e) : n.insertBefore(e, n.firstChild), l.push(e);
		else if ("bottom" === t.insertAt) n.appendChild(e);
		else
		{
			if ("object" != typeof t.insertAt || !t.insertAt.before) throw new Error("[Style Loader]\n\n Invalid value for parameter 'insertAt' ('options.insertAt') found.\n Must be 'top', 'bottom', or Object.\n (https://github.com/webpack-contrib/style-loader#insertat)\n");
			var o = a(t.insertAt.before, n);
			n.insertBefore(e, o)
		}
	}

	function v(t)
	{
		if (null === t.parentNode) return !1;
		t.parentNode.removeChild(t);
		var e = l.indexOf(t);
		e >= 0 && l.splice(e, 1)
	}

	function _(t)
	{
		var e = document.createElement("style");
		if (void 0 === t.attrs.type && (t.attrs.type = "text/css"), void 0 === t.attrs.nonce)
		{
			var r = function ()
			{
				0;
				return n.nc
			}();
			r && (t.attrs.nonce = r)
		}
		return m(e, t.attrs), h(t, e), e
	}

	function m(t, e)
	{
		Object.keys(e).forEach(function (n)
		{
			t.setAttribute(n, e[n])
		})
	}

	function g(t, e)
	{
		var n, r, o, i;
		if (e.transform && t.css)
		{
			if (!(i = "function" == typeof e.transform ? e.transform(t.css) : e.transform.default(t.css))) return function () {};
			t.css = i
		}
		if (e.singleton)
		{
			var s = u++;
			n = c || (c = _(e)), r = b.bind(null, n, s, !1), o = b.bind(null, n, s, !0)
		}
		else t.sourceMap && "function" == typeof URL && "function" == typeof URL.createObjectURL && "function" == typeof URL.revokeObjectURL && "function" == typeof Blob && "function" == typeof btoa ? (n = function (t)
		{
			var e = document.createElement("link");
			return void 0 === t.attrs.type && (t.attrs.type = "text/css"), t.attrs.rel = "stylesheet", m(e, t.attrs), h(t, e), e
		}(e), r = function (t, e, n)
		{
			var r = n.css,
				o = n.sourceMap,
				i = void 0 === e.convertToAbsoluteUrls && o;
			(e.convertToAbsoluteUrls || i) && (r = f(r));
			o && (r += "\n/*# sourceMappingURL=data:application/json;base64," + btoa(unescape(encodeURIComponent(JSON.stringify(o)))) + " */");
			var s = new Blob([r],
				{
					type: "text/css"
				}),
				a = t.href;
			t.href = URL.createObjectURL(s), a && URL.revokeObjectURL(a)
		}.bind(null, n, e), o = function ()
		{
			v(n), n.href && URL.revokeObjectURL(n.href)
		}) : (n = _(e), r = function (t, e)
		{
			var n = e.css,
				r = e.media;
			r && t.setAttribute("media", r);
			if (t.styleSheet) t.styleSheet.cssText = n;
			else
			{
				for (; t.firstChild;) t.removeChild(t.firstChild);
				t.appendChild(document.createTextNode(n))
			}
		}.bind(null, n), o = function ()
		{
			v(n)
		});
		return r(t),
			function (e)
			{
				if (e)
				{
					if (e.css === t.css && e.media === t.media && e.sourceMap === t.sourceMap) return;
					r(t = e)
				}
				else o()
			}
	}
	t.exports = function (t, e)
	{
		if ("undefined" != typeof DEBUG && DEBUG && "object" != typeof document) throw new Error("The style-loader cannot be used in a non-browser environment");
		(e = e ||
		{}).attrs = "object" == typeof e.attrs ? e.attrs :
		{}, e.singleton || "boolean" == typeof e.singleton || (e.singleton = s()), e.insertInto || (e.insertInto = "head"), e.insertAt || (e.insertAt = "bottom");
		var n = p(t, e);
		return d(n, e),
			function (t)
			{
				for (var r = [], o = 0; o < n.length; o++)
				{
					var s = n[o];
					(a = i[s.id]).refs--, r.push(a)
				}
				t && d(p(t, e), e);
				for (o = 0; o < r.length; o++)
				{
					var a;
					if (0 === (a = r[o]).refs)
					{
						for (var c = 0; c < a.parts.length; c++) a.parts[c]();
						delete i[a.id]
					}
				}
			}
	};
	var y, w = (y = [], function (t, e)
	{
		return y[t] = e, y.filter(Boolean).join("\n")
	});

	function b(t, e, n, r)
	{
		var o = n ? "" : r.css;
		if (t.styleSheet) t.styleSheet.cssText = w(e, o);
		else
		{
			var i = document.createTextNode(o),
				s = t.childNodes;
			s[e] && t.removeChild(s[e]), s.length ? t.insertBefore(i, s[e]) : t.appendChild(i)
		}
	}
}, function (t, e)
{
	t.exports = function (t)
	{
		var e = [];
		return e.toString = function ()
		{
			return this.map(function (e)
			{
				var n = function (t, e)
				{
					var n = t[1] || "",
						r = t[3];
					if (!r) return n;
					if (e && "function" == typeof btoa)
					{
						var o = (s = r, "/*# sourceMappingURL=data:application/json;charset=utf-8;base64," + btoa(unescape(encodeURIComponent(JSON.stringify(s)))) + " */"),
							i = r.sources.map(function (t)
							{
								return "/*# sourceURL=" + r.sourceRoot + t + " */"
							});
						return [n].concat(i).concat([o]).join("\n")
					}
					var s;
					return [n].join("\n")
				}(e, t);
				return e[2] ? "@media " + e[2] + "{" + n + "}" : n
			}).join("")
		}, e.i = function (t, n)
		{
			"string" == typeof t && (t = [
				[null, t, ""]
			]);
			for (var r = {}, o = 0; o < this.length; o++)
			{
				var i = this[o][0];
				"number" == typeof i && (r[i] = !0)
			}
			for (o = 0; o < t.length; o++)
			{
				var s = t[o];
				"number" == typeof s[0] && r[s[0]] || (n && !s[2] ? s[2] = n : n && (s[2] = "(" + s[2] + ") and (" + n + ")"), e.push(s))
			}
		}, e
	}
}, function (t, e, n)
{
	(t.exports = n(225)(void 0)).push([t.i, '/**\n * 1. As on mobile devices we use Helvetica font which have a different baseline\n *    we have to change line-height depending on resolution.\n */\n/**\n * 1. As on mobile devices we use Helvetica font which have a different baseline\n *    we have to change line-height depending on resolution.\n */\n.button {\n  font-size: 13px;\n  line-height: 18px;\n  font-family: "Helvetica Neue", Helvetica, Arial, "Lucida Grande", sans-serif;\n  height: 30px;\n  padding: 0 10px;\n  margin-right: 10px;\n  line-height: 28px;\n  /* [1] */\n  color: #ffffff;\n  background-color: #00b2ff;\n  border-color: #005689;\n  display: inline-block;\n  vertical-align: top;\n  width: auto;\n  font-weight: bold;\n  text-decoration: none;\n  border-radius: 2px;\n  border-width: 1px;\n  border-style: solid;\n  box-sizing: border-box; }\n  .button svg {\n    width: 30px;\n    height: 30px; }\n  .button:hover, .button:focus, .button:active {\n    text-decoration: none; }\n\n.button--primary {\n  color: #ffffff;\n  background-color: #005689;\n  border-color: #005689; }\n  .button--primary:hover, .button--primary:focus, .button--primary:active {\n    background-color: #003656;\n    border-color: #003656; }\n\n.button--secondary {\n  color: #121212;\n  background-color: #f6f6f6;\n  border-color: #f6f6f6; }\n  .button--secondary:hover, .button--secondary:focus, .button--secondary:active {\n    background-color: #dddddd;\n    border-color: #dddddd; }\n\n.button--tertiary {\n  color: #121212;\n  background-color: transparent;\n  border-color: #dcdcdc; }\n  .button--tertiary:hover, .button--tertiary:focus, .button--tertiary:active {\n    background-color: transparent;\n    border-color: #c3c3c3; }\n\n.button--large {\n  height: 36px;\n  padding: 0 12px;\n  margin-right: 12px;\n  line-height: 34px;\n  /* [1] */ }\n  .button--large svg {\n    width: 36px;\n    height: 36px; }\n  @media (min-width: 61.25em) {\n    .button--large {\n      line-height: 38px; } }\n\n@media (min-width: 46.25em) {\n  .crossword__container--quick {\n    padding-left: 417px; }\n    .crossword__container--quick.crossword__container--accessible {\n      padding-left: 0; } }\n\n@media (min-width: 46.25em) {\n  .crossword__container--quick .crossword__container__game {\n    margin-left: -417px; } }\n\n.crossword__container--quick .crossword__container__grid-wrapper {\n  width: 100%;\n  height: 417px; }\n  @media (min-width: 46.25em) {\n    .crossword__container--quick .crossword__container__grid-wrapper {\n      width: 417px; } }\n\n@media (min-width: 46.25em) {\n  .crossword__container--quick .crossword__controls {\n    margin-left: -417px;\n    width: 417px; } }\n\n.crossword__container--quick .crossword__hidden-input-wrapper {\n  width: 7.69231%;\n  height: 7.69231%; }\n\n@media (min-width: 46.25em) {\n  .crossword__container--cryptic {\n    padding-left: 481px; }\n    .crossword__container--cryptic.crossword__container--accessible {\n      padding-left: 0; } }\n\n@media (min-width: 46.25em) {\n  .crossword__container--cryptic .crossword__container__game {\n    margin-left: -481px; } }\n\n.crossword__container--cryptic .crossword__container__grid-wrapper {\n  width: 100%;\n  height: 481px; }\n  @media (min-width: 46.25em) {\n    .crossword__container--cryptic .crossword__container__grid-wrapper {\n      width: 481px; } }\n\n@media (min-width: 46.25em) {\n  .crossword__container--cryptic .crossword__controls {\n    margin-left: -481px;\n    width: 481px; } }\n\n.crossword__container--cryptic .crossword__hidden-input-wrapper {\n  width: 6.66667%;\n  height: 6.66667%; }\n\n@media (min-width: 46.25em) {\n  .crossword__container--prize {\n    padding-left: 481px; }\n    .crossword__container--prize.crossword__container--accessible {\n      padding-left: 0; } }\n\n@media (min-width: 46.25em) {\n  .crossword__container--prize .crossword__container__game {\n    margin-left: -481px; } }\n\n.crossword__container--prize .crossword__container__grid-wrapper {\n  width: 100%;\n  height: 481px; }\n  @media (min-width: 46.25em) {\n    .crossword__container--prize .crossword__container__grid-wrapper {\n      width: 481px; } }\n\n@media (min-width: 46.25em) {\n  .crossword__container--prize .crossword__controls {\n    margin-left: -481px;\n    width: 481px; } }\n\n.crossword__container--prize .crossword__hidden-input-wrapper {\n  width: 6.66667%;\n  height: 6.66667%; }\n\n@media (min-width: 46.25em) {\n  .crossword__container--quiptic {\n    padding-left: 481px; }\n    .crossword__container--quiptic.crossword__container--accessible {\n      padding-left: 0; } }\n\n@media (min-width: 46.25em) {\n  .crossword__container--quiptic .crossword__container__game {\n    margin-left: -481px; } }\n\n.crossword__container--quiptic .crossword__container__grid-wrapper {\n  width: 100%;\n  height: 481px; }\n  @media (min-width: 46.25em) {\n    .crossword__container--quiptic .crossword__container__grid-wrapper {\n      width: 481px; } }\n\n@media (min-width: 46.25em) {\n  .crossword__container--quiptic .crossword__controls {\n    margin-left: -481px;\n    width: 481px; } }\n\n.crossword__container--quiptic .crossword__hidden-input-wrapper {\n  width: 6.66667%;\n  height: 6.66667%; }\n\n@media (min-width: 46.25em) {\n  .crossword__container--genius {\n    padding-left: 481px; }\n    .crossword__container--genius.crossword__container--accessible {\n      padding-left: 0; } }\n\n@media (min-width: 46.25em) {\n  .crossword__container--genius .crossword__container__game {\n    margin-left: -481px; } }\n\n.crossword__container--genius .crossword__container__grid-wrapper {\n  width: 100%;\n  height: 481px; }\n  @media (min-width: 46.25em) {\n    .crossword__container--genius .crossword__container__grid-wrapper {\n      width: 481px; } }\n\n@media (min-width: 46.25em) {\n  .crossword__container--genius .crossword__controls {\n    margin-left: -481px;\n    width: 481px; } }\n\n.crossword__container--genius .crossword__hidden-input-wrapper {\n  width: 6.66667%;\n  height: 6.66667%; }\n\n@media (min-width: 46.25em) {\n  .crossword__container--speedy {\n    padding-left: 417px; }\n    .crossword__container--speedy.crossword__container--accessible {\n      padding-left: 0; } }\n\n@media (min-width: 46.25em) {\n  .crossword__container--speedy .crossword__container__game {\n    margin-left: -417px; } }\n\n.crossword__container--speedy .crossword__container__grid-wrapper {\n  width: 100%;\n  height: 417px; }\n  @media (min-width: 46.25em) {\n    .crossword__container--speedy .crossword__container__grid-wrapper {\n      width: 417px; } }\n\n@media (min-width: 46.25em) {\n  .crossword__container--speedy .crossword__controls {\n    margin-left: -417px;\n    width: 417px; } }\n\n.crossword__container--speedy .crossword__hidden-input-wrapper {\n  width: 7.69231%;\n  height: 7.69231%; }\n\n@media (min-width: 46.25em) {\n  .crossword__container--everyman {\n    padding-left: 481px; }\n    .crossword__container--everyman.crossword__container--accessible {\n      padding-left: 0; } }\n\n@media (min-width: 46.25em) {\n  .crossword__container--everyman .crossword__container__game {\n    margin-left: -481px; } }\n\n.crossword__container--everyman .crossword__container__grid-wrapper {\n  width: 100%;\n  height: 481px; }\n  @media (min-width: 46.25em) {\n    .crossword__container--everyman .crossword__container__grid-wrapper {\n      width: 481px; } }\n\n@media (min-width: 46.25em) {\n  .crossword__container--everyman .crossword__controls {\n    margin-left: -481px;\n    width: 481px; } }\n\n.crossword__container--everyman .crossword__hidden-input-wrapper {\n  width: 6.66667%;\n  height: 6.66667%; }\n\n@media (min-width: 46.25em) {\n  .crossword__container--weekend {\n    padding-left: 417px; }\n    .crossword__container--weekend.crossword__container--accessible {\n      padding-left: 0; } }\n\n@media (min-width: 46.25em) {\n  .crossword__container--weekend .crossword__container__game {\n    margin-left: -417px; } }\n\n.crossword__container--weekend .crossword__container__grid-wrapper {\n  width: 100%;\n  height: 417px; }\n  @media (min-width: 46.25em) {\n    .crossword__container--weekend .crossword__container__grid-wrapper {\n      width: 417px; } }\n\n@media (min-width: 46.25em) {\n  .crossword__container--weekend .crossword__controls {\n    margin-left: -417px;\n    width: 417px; } }\n\n.crossword__container--weekend .crossword__hidden-input-wrapper {\n  width: 7.69231%;\n  height: 7.69231%; }\n\n.crossword__container {\n  clear: both;\n  position: relative; }\n\n.crossword__controls {\n  margin-top: 12px;\n  float: left;\n  clear: both; }\n\n@media (min-width: 46.25em) and (max-width: 71.24em) {\n  .crossword__container--react .crossword__clues--wrapper,\n  .crossword__container--react .crossword__clues {\n    height: 601px; } }\n\n@media (min-width: 46.25em) and (max-width: 71.24em) {\n  .crossword__container--react .crossword__clues {\n    position: relative;\n    overflow-y: scroll;\n    -webkit-overflow-scrolling: touch; } }\n\n.crossword__clues__gradient {\n  position: absolute;\n  bottom: 0;\n  width: 100%;\n  height: 120px;\n  background: linear-gradient(to bottom, rgba(255, 255, 255, 0), #ffffff);\n  display: none; }\n  @media (min-width: 46.25em) and (max-width: 71.24em) {\n    .crossword__clues__gradient {\n      display: block; }\n      .hide-gradient .crossword__clues__gradient {\n        display: none; } }\n\n.crossword__clues {\n  width: 100%;\n  clear: both; }\n  @media (min-width: 46.25em) {\n    .crossword__clues {\n      clear: none;\n      padding-left: 20px;\n      box-sizing: border-box; }\n      noscript .crossword__clues {\n        width: 85%; }\n      .crossword__container--accessible .crossword__clues {\n        display: table;\n        table-layout: fixed;\n        width: initial; } }\n  @media (min-width: 71.25em) {\n    .crossword__clues {\n      display: table;\n      table-layout: fixed;\n      height: auto;\n      background: none; } }\n\n.crossword__accessible-row-data-wrapper {\n  float: left; }\n\n@media (min-width: 61.25em) {\n  .crossword__clues--across {\n    display: table-cell;\n    padding-right: 10px; } }\n\n@media (min-width: 61.25em) {\n  .crossword__clues--down {\n    display: table-cell;\n    padding-left: 10px; } }\n\n.crossword__anagram-helper-outer {\n  position: fixed;\n  box-sizing: border-box;\n  top: 0;\n  left: 0;\n  width: 100%;\n  height: 100%;\n  padding-top: 2.5%;\n  background: #f6f6f6;\n  text-align: center; }\n  .crossword__anagram-helper-outer .button:last-of-type {\n    margin-right: 0; }\n  .crossword__anagram-helper-outer .button--large {\n    background: #bb3b80;\n    border: 1px solid #bb3b80; }\n    .crossword__anagram-helper-outer .button--large:hover, .crossword__anagram-helper-outer .button--large:visited, .crossword__anagram-helper-outer .button--large:focus {\n      background-color: #942f65;\n      border-color: #942f65; }\n  .crossword__anagram-helper-outer .button--tertiary {\n    background: transparent;\n    border: 1px solid #dcdcdc; }\n    .crossword__anagram-helper-outer .button--tertiary:hover, .crossword__anagram-helper-outer .button--tertiary:visited, .crossword__anagram-helper-outer .button--tertiary:focus {\n      background-color: transparent;\n      border-color: #b6b6b6; }\n  @media (max-width: 46.24em) {\n    .crossword__anagram-helper-outer {\n      z-index: 1030; } }\n  @media (min-width: 46.25em) {\n    .crossword__anagram-helper-outer {\n      position: absolute; } }\n\n.crossword__anagram-helper-inner {\n  height: 50%; }\n\n.crossword__anagram-helper__clue-input,\n.crossword__anagram-helper-shuffler__letter,\n.crossword__anagram-helper__clue-preview {\n  font-family: "Helvetica Neue", Helvetica, Arial, "Lucida Grande", sans-serif; }\n\n.crossword__anagram-helper__clue-input,\n.crossword__anagram-helper-shuffler__letter {\n  font-size: 24px; }\n\n.crossword__anagram-helper-shuffler__letter,\n.crossword__anagram-helper__cell {\n  text-transform: uppercase; }\n\n.crossword__anagram-helper__clue-input {\n  width: 100%;\n  background: none;\n  border: 0;\n  outline: none;\n  padding-top: 15%;\n  text-align: center; }\n  @media (min-width: 46.25em) {\n    .crossword__anagram-helper__clue-input {\n      padding-top: 25%; } }\n\n.crossword__anagram-helper__clue-preview {\n  margin-top: 5%;\n  font-size: 14px; }\n\n.crossword__anagram-helper__direction {\n  text-transform: capitalize; }\n\n.crossword__anagram-helper__cell {\n  display: inline-block;\n  vertical-align: middle;\n  width: 31px;\n  height: 31px;\n  line-height: 31px;\n  border: 1px solid #dcdcdc;\n  border-left: 1px solid #dcdcdc; }\n  @media (max-width: 29.99em) {\n    .crossword__anagram-helper__cell {\n      width: 24px;\n      height: 24px;\n      line-height: 24px; }\n      .long .crossword__anagram-helper__cell {\n        width: 20px;\n        height: 20px;\n        line-height: 20px; } }\n  .crossword__anagram-helper__cell:first-of-type {\n    border-left: 1px solid #dcdcdc; }\n  .crossword__anagram-helper__cell.has-value {\n    background: #dcdcdc; }\n\n.crossword__anagram-helper__cell--with-space {\n  border-right: 1px solid #121212; }\n\n.crossword__anagram-helper__cell--with-hyphen {\n  position: relative; }\n\n.crossword__anagram-helper__cell--with-hyphen::after {\n  content: \'-\';\n  position: absolute;\n  font-size: 20px;\n  right: -5px;\n  top: -2px;\n  z-index: 10; }\n\n.crossword__anagram-helper-shuffler {\n  position: relative;\n  top: 25px;\n  left: 10px;\n  width: 175px;\n  height: 175px;\n  margin: 0 auto; }\n  @media (min-width: 46.25em) {\n    .crossword__anagram-helper-shuffler {\n      top: 0;\n      left: 5%;\n      width: 50%;\n      height: 100%; } }\n\n.crossword__anagram-helper-shuffler__letter {\n  position: absolute; }\n  .crossword__anagram-helper-shuffler__letter.entered {\n    color: #dcdcdc; }\n\n.crossword__anagram-helper-close {\n  position: absolute;\n  top: 12px;\n  right: 0;\n  padding: 0; }\n  .crossword__anagram-helper-close svg {\n    position: relative;\n    top: 2px;\n    left: 3px;\n    fill: #121212; }\n\n.crossword__accessible-blank-data {\n  list-style-type: none; }\n\n.crossword__accessible-blank-data {\n  margin: 0;\n  margin-bottom: 12px; }\n\n.crossword__accessible-row-data--row-number {\n  font-weight: bold; }\n\n.crossword__accessible-row-data {\n  font-size: 14px;\n  line-height: 22px;\n  font-family: "Helvetica Neue", Helvetica, Arial, "Lucida Grande", sans-serif;\n  position: relative;\n  cursor: pointer;\n  padding: 3px 0;\n  list-style-type: none;\n  padding-left: 2em; }\n  .crossword__accessible-row-data:before {\n    content: attr(data-number);\n    position: absolute;\n    left: 0;\n    font-weight: bold;\n    -webkit-font-smoothing: initial; }\n\n.crossword__accessible-data--header {\n  font-size: 16px;\n  line-height: 20px;\n  font-family: "PT Serif", serif;\n  font-weight: 900;\n  border-top: 1px solid #ffe500;\n  border-bottom: 1px dotted #dcdcdc;\n  padding-bottom: 9px; }\n\n.crossword__cell {\n  cursor: pointer;\n  fill: #ffffff; }\n\n.crossword__cell--highlighted,\n.crossword__grid--focussed .crossword__cell--highlighted {\n  fill: #ffe500; }\n\n.crossword__cell--focussed,\n.crossword__grid--focussed .crossword__cell--focussed {\n  fill: #fff7b2; }\n\n.crossword__cell-number,\n.crossword__cell-text {\n  fill: #000000;\n  font-family: "Helvetica Neue", Helvetica, Arial, "Lucida Grande", sans-serif; }\n\n.crossword__cell-number {\n  font-size: 10px; }\n\n.crossword__cell-text {\n  display: inline-block;\n  opacity: 1;\n  -webkit-font-smoothing: subpixel-antialiased;\n  transition: opacity .15s ease-in; }\n  .crossword__container--quick .crossword__cell-text {\n    font-size: 16px; }\n    @media (min-width: 46.25em) {\n      .crossword__container--quick .crossword__cell-text {\n        font-size: 14.3px; } }\n  .crossword__container--cryptic .crossword__cell-text {\n    font-size: 18px; }\n    @media (min-width: 46.25em) {\n      .crossword__container--cryptic .crossword__cell-text {\n        font-size: 16.5px; } }\n  .crossword__container--prize .crossword__cell-text {\n    font-size: 18px; }\n    @media (min-width: 46.25em) {\n      .crossword__container--prize .crossword__cell-text {\n        font-size: 16.5px; } }\n  .crossword__container--quiptic .crossword__cell-text {\n    font-size: 18px; }\n    @media (min-width: 46.25em) {\n      .crossword__container--quiptic .crossword__cell-text {\n        font-size: 16.5px; } }\n  .crossword__container--genius .crossword__cell-text {\n    font-size: 18px; }\n    @media (min-width: 46.25em) {\n      .crossword__container--genius .crossword__cell-text {\n        font-size: 16.5px; } }\n  .crossword__container--speedy .crossword__cell-text {\n    font-size: 16px; }\n    @media (min-width: 46.25em) {\n      .crossword__container--speedy .crossword__cell-text {\n        font-size: 14.3px; } }\n  .crossword__container--everyman .crossword__cell-text {\n    font-size: 18px; }\n    @media (min-width: 46.25em) {\n      .crossword__container--everyman .crossword__cell-text {\n        font-size: 16.5px; } }\n  .crossword__container--weekend .crossword__cell-text {\n    font-size: 16px; }\n    @media (min-width: 46.25em) {\n      .crossword__container--weekend .crossword__cell-text {\n        font-size: 14.3px; } }\n\n.crossword__cell-text--focussed {\n  font-weight: bold; }\n\n.crossword__cell-text--error {\n  fill: #610000;\n  opacity: 0; }\n\n.crossword__clues-header {\n  font-size: 16px;\n  line-height: 20px;\n  font-family: "PT Serif", serif;\n  font-weight: 900;\n  border-bottom: 1px dotted #dcdcdc;\n  padding-bottom: 9px; }\n\n.crossword__clues-list, ol.crossword__clues-list {\n  padding: 0;\n  margin: 0;\n  margin-bottom: 12px;\n  list-style-type: none; }\n\n.crossword__clue {\n  font-size: 14px;\n  line-height: 22px;\n  font-family: "Helvetica Neue", Helvetica, Arial, "Lucida Grande", sans-serif;\n  text-decoration: none;\n  position: relative;\n  padding: 3px 12px 3px 0;\n  margin: 3px 0;\n  display: block; }\n  .crossword__clue:hover {\n    text-decoration: none; }\n  .crossword__clue, .crossword__clue:active {\n    color: inherit; }\n\n.crossword__clue__number,\n.crossword__clue__text {\n  display: table-cell; }\n\n.crossword__clue__number {\n  font-weight: bold;\n  -webkit-font-smoothing: initial;\n  width: 20px;\n  padding-right: 10px; }\n  .has-grouped-clues .crossword__clue__number {\n    width: 40px; }\n\n.crossword__clue--selected {\n  background-color: #fff7b2; }\n  .crossword__clue--selected > * {\n    font-weight: bold;\n    color: #000000; }\n\n.crossword__clue--answered {\n  color: #767676; }\n\n.crossword__clue--display-group-order:before {\n  padding-right: 14px; }\n\n.crossword__controls {\n  display: block;\n  margin-bottom: 6px; }\n  .crossword__controls .button {\n    font-size: 14px;\n    line-height: 22px;\n    font-family: "Helvetica Neue", Helvetica, Arial, "Lucida Grande", sans-serif;\n    margin-bottom: 12px; }\n  .crossword__controls .button--primary {\n    background: #bb3b80;\n    border: 1px solid #bb3b80; }\n    .crossword__controls .button--primary:hover, .crossword__controls .button--primary:visited, .crossword__controls .button--primary:focus {\n      background-color: #942f65;\n      border-color: #942f65; }\n  .crossword__controls .button--secondary {\n    color: #767676;\n    background: #ededed;\n    border: 1px solid #ededed; }\n    .crossword__controls .button--secondary:hover, .crossword__controls .button--secondary:visited, .crossword__controls .button--secondary:focus {\n      background-color: #d4d4d4;\n      border-color: #d4d4d4; }\n  .crossword__controls .button--secondary.crossword__controls__button--confirm {\n    background-color: #121212;\n    color: #ffffff;\n    border: 0; }\n\n.crossword__controls__clue .button:last-child {\n  margin-right: 0; }\n\n.crossword__container__game {\n  position: relative;\n  padding-top: 52px;\n  padding-bottom: 52px; }\n  @media (min-width: 46.25em) {\n    .crossword__container__game {\n      float: left;\n      padding-top: 0;\n      padding-bottom: 0; } }\n\n.crossword__sticky-clue-wrapper {\n  position: absolute;\n  top: 0;\n  left: 0;\n  right: 0;\n  z-index: 1;\n  will-change: transform; }\n  @media (min-width: 46.25em) {\n    .crossword__sticky-clue-wrapper {\n      display: none; } }\n  .crossword__sticky-clue-wrapper.is-fixed {\n    padding-left: 10px;\n    padding-right: 10px;\n    position: fixed; }\n    @media (min-width: 30em) {\n      .crossword__sticky-clue-wrapper.is-fixed {\n        padding-left: 20px;\n        padding-right: 20px; } }\n\n.crossword__sticky-clue {\n  font-size: 14px;\n  line-height: 20px;\n  font-family: "Helvetica Neue", Helvetica, Arial, "Lucida Grande", sans-serif;\n  background: #ffffff;\n  height: 52px;\n  border-bottom: 1px solid #000000;\n  box-sizing: border-box;\n  padding-top: 6px;\n  padding-bottom: 6px;\n  display: table;\n  width: 100%; }\n\n.crossword__sticky-clue__direction {\n  text-transform: capitalize; }\n\n.crossword__sticky-clue__inner {\n  display: table-cell;\n  vertical-align: middle; }\n\n.crossword__sticky-clue__inner__inner {\n  overflow: hidden;\n  display: -webkit-box;\n  -webkit-box-orient: vertical;\n  -webkit-line-clamp: 2;\n  max-height: 40; }\n\n.crossword__container__grid-wrapper {\n  position: relative; }\n\n.crossword__grid {\n  display: block; }\n\n.crossword__hidden-input-wrapper {\n  position: absolute; }\n\n.crossword__hidden-input {\n  border: 0;\n  padding: 0;\n  width: 100%;\n  height: 100%;\n  text-align: center;\n  background-color: transparent;\n  font-size: 16px;\n  font-family: "PT Serif", serif;\n  font-weight: 800;\n  text-anchor: middle; }\n\n.crossword__hidden-input-prev-next {\n  position: absolute;\n  left: -1000px;\n  opacity: 0; }\n\n.printable-crossword {\n  width: 18cm;\n  margin: 10px; }\n\n.printable-crossword__title {\n  font-size: 18px; }\n\n.printable-crossword__instructions {\n  font-weight: normal;\n  font-size: 14px; }\n\n.printable-crossword__grid {\n  width: 436px;\n  height: 436px; }\n  .printable-crossword__grid .crossword__grid-background {\n    fill: #7f7f7f; }\n\n.printable-crossword__clues {\n  width: 49%;\n  float: left;\n  font-size: 12px; }\n  .printable-crossword__clues:first-of-type {\n    margin-right: 2%; }\n\n.printable-crossword__clues__title {\n  font-size: 13px;\n  font-weight: bold; }\n\n.printable-crossword__clues__list {\n  margin: 0;\n  list-style-type: none; }\n\n.printable-crossword__clue__number {\n  font-weight: bold; }\n\n.printable-crossword__copyright {\n  clear: both; }\n', ""])
}, function (t, e, n)
{
	var r = n(226);
	"string" == typeof r && (r = [
		[t.i, r, ""]
	]);
	var o = {
		hmr: !0,
		transform: void 0,
		insertInto: void 0
	};
	n(224)(r, o);
	r.locals && (t.exports = r.locals)
}, function (t, e, n)
{
	var r = n(23);
	r(r.S, "Object",
	{
		create: n(59)
	})
}, function (t, e, n)
{
	n(228);
	var r = n(11).Object;
	t.exports = function (t, e)
	{
		return r.create(t, e)
	}
}, function (t, e, n)
{
	t.exports = {
		default: n(229),
		__esModule: !0
	}
}, function (t, e, n)
{
	var r = n(20),
		o = n(21),
		i = function (t, e)
		{
			if (o(t), !r(e) && null !== e) throw TypeError(e + ": can't set as prototype!")
		};
	t.exports = {
		set: Object.setPrototypeOf || ("__proto__" in
		{} ? function (t, e, r)
		{
			try
			{
				(r = n(64)(Function.call, n(90).f(Object.prototype, "__proto__").set, 2))(t, []), e = !(t instanceof Array)
			}
			catch (t)
			{
				e = !0
			}
			return function (t, n)
			{
				return i(t, n), e ? t.__proto__ = n : r(t, n), t
			}
		}(
		{}, !1) : void 0),
		check: i
	}
}, function (t, e, n)
{
	var r = n(23);
	r(r.S, "Object",
	{
		setPrototypeOf: n(231).set
	})
}, function (t, e, n)
{
	n(232), t.exports = n(11).Object.setPrototypeOf
}, function (t, e, n)
{
	t.exports = {
		default: n(233),
		__esModule: !0
	}
}, function (t, e, n)
{
	n(51)("observable")
}, function (t, e, n)
{
	n(51)("asyncIterator")
}, function (t, e) {}, function (t, e, n)
{
	var r = n(19),
		o = n(91).f,
		i = {}.toString,
		s = "object" == typeof window && window && Object.getOwnPropertyNames ? Object.getOwnPropertyNames(window) : [];
	t.exports.f = function (t)
	{
		return s && "[object Window]" == i.call(t) ? function (t)
		{
			try
			{
				return o(t)
			}
			catch (t)
			{
				return s.slice()
			}
		}(t) : o(r(t))
	}
}, function (t, e, n)
{
	var r = n(57);
	t.exports = Array.isArray || function (t)
	{
		return "Array" == r(t)
	}
}, function (t, e, n)
{
	var r = n(58),
		o = n(92),
		i = n(50);
	t.exports = function (t)
	{
		var e = r(t),
			n = o.f;
		if (n)
			for (var s, a = n(t), c = i.f, u = 0; a.length > u;) c.call(t, s = a[u++]) && e.push(s);
		return e
	}
}, function (t, e, n)
{
	var r = n(35)("meta"),
		o = n(20),
		i = n(15),
		s = n(10).f,
		a = 0,
		c = Object.isExtensible || function ()
		{
			return !0
		},
		u = !n(36)(function ()
		{
			return c(Object.preventExtensions(
			{}))
		}),
		l = function (t)
		{
			s(t, r,
			{
				value:
				{
					i: "O" + ++a,
					w:
					{}
				}
			})
		},
		f = t.exports = {
			KEY: r,
			NEED: !1,
			fastKey: function (t, e)
			{
				if (!o(t)) return "symbol" == typeof t ? t : ("string" == typeof t ? "S" : "P") + t;
				if (!i(t, r))
				{
					if (!c(t)) return "F";
					if (!e) return "E";
					l(t)
				}
				return t[r].i
			},
			getWeak: function (t, e)
			{
				if (!i(t, r))
				{
					if (!c(t)) return !0;
					if (!e) return !1;
					l(t)
				}
				return t[r].w
			},
			onFreeze: function (t)
			{
				return u && f.NEED && c(t) && !i(t, r) && l(t), t
			}
		}
}, function (t, e, n)
{
	"use strict";
	var r = n(12),
		o = n(15),
		i = n(16),
		s = n(23),
		a = n(96),
		c = n(241).KEY,
		u = n(36),
		l = n(55),
		f = n(53),
		d = n(35),
		p = n(8),
		h = n(52),
		v = n(51),
		_ = n(240),
		m = n(239),
		g = n(21),
		y = n(20),
		w = n(19),
		b = n(63),
		x = n(27),
		k = n(59),
		E = n(238),
		O = n(90),
		C = n(10),
		j = n(58),
		S = O.f,
		N = C.f,
		I = E.f,
		A = r.Symbol,
		T = r.JSON,
		L = T && T.stringify,
		P = p("_hidden"),
		M = p("toPrimitive"),
		F = {}.propertyIsEnumerable,
		D = l("symbol-registry"),
		z = l("symbols"),
		H = l("op-symbols"),
		$ = Object.prototype,
		R = "function" == typeof A,
		B = r.QObject,
		G = !B || !B.prototype || !B.prototype.findChild,
		q = i && u(function ()
		{
			return 7 != k(N(
			{}, "a",
			{
				get: function ()
				{
					return N(this, "a",
					{
						value: 7
					}).a
				}
			})).a
		}) ? function (t, e, n)
		{
			var r = S($, e);
			r && delete $[e], N(t, e, n), r && t !== $ && N($, e, r)
		} : N,
		V = function (t)
		{
			var e = z[t] = k(A.prototype);
			return e._k = t, e
		},
		W = R && "symbol" == typeof A.iterator ? function (t)
		{
			return "symbol" == typeof t
		} : function (t)
		{
			return t instanceof A
		},
		U = function (t, e, n)
		{
			return t === $ && U(H, e, n), g(t), e = b(e, !0), g(n), o(z, e) ? (n.enumerable ? (o(t, P) && t[P][e] && (t[P][e] = !1), n = k(n,
			{
				enumerable: x(0, !1)
			})) : (o(t, P) || N(t, P, x(1,
			{})), t[P][e] = !0), q(t, e, n)) : N(t, e, n)
		},
		K = function (t, e)
		{
			g(t);
			for (var n, r = _(e = w(e)), o = 0, i = r.length; i > o;) U(t, n = r[o++], e[n]);
			return t
		},
		Q = function (t)
		{
			var e = F.call(this, t = b(t, !0));
			return !(this === $ && o(z, t) && !o(H, t)) && (!(e || !o(this, t) || !o(z, t) || o(this, P) && this[P][t]) || e)
		},
		Y = function (t, e)
		{
			if (t = w(t), e = b(e, !0), t !== $ || !o(z, e) || o(H, e))
			{
				var n = S(t, e);
				return !n || !o(z, e) || o(t, P) && t[P][e] || (n.enumerable = !0), n
			}
		},
		X = function (t)
		{
			for (var e, n = I(w(t)), r = [], i = 0; n.length > i;) o(z, e = n[i++]) || e == P || e == c || r.push(e);
			return r
		},
		J = function (t)
		{
			for (var e, n = t === $, r = I(n ? H : w(t)), i = [], s = 0; r.length > s;) !o(z, e = r[s++]) || n && !o($, e) || i.push(z[e]);
			return i
		};
	R || (a((A = function ()
	{
		if (this instanceof A) throw TypeError("Symbol is not a constructor!");
		var t = d(arguments.length > 0 ? arguments[0] : void 0),
			e = function (n)
			{
				this === $ && e.call(H, n), o(this, P) && o(this[P], t) && (this[P][t] = !1), q(this, t, x(1, n))
			};
		return i && G && q($, t,
		{
			configurable: !0,
			set: e
		}), V(t)
	}).prototype, "toString", function ()
	{
		return this._k
	}), O.f = Y, C.f = U, n(91).f = E.f = X, n(50).f = Q, n(92).f = J, i && !n(60) && a($, "propertyIsEnumerable", Q, !0), h.f = function (t)
	{
		return V(p(t))
	}), s(s.G + s.W + s.F * !R,
	{
		Symbol: A
	});
	for (var Z = "hasInstance,isConcatSpreadable,iterator,match,replace,search,species,split,toPrimitive,toStringTag,unscopables".split(","), tt = 0; Z.length > tt;) p(Z[tt++]);
	for (var et = j(p.store), nt = 0; et.length > nt;) v(et[nt++]);
	s(s.S + s.F * !R, "Symbol",
	{
		for: function (t)
		{
			return o(D, t += "") ? D[t] : D[t] = A(t)
		},
		keyFor: function (t)
		{
			if (!W(t)) throw TypeError(t + " is not a symbol!");
			for (var e in D)
				if (D[e] === t) return e
		},
		useSetter: function ()
		{
			G = !0
		},
		useSimple: function ()
		{
			G = !1
		}
	}), s(s.S + s.F * !R, "Object",
	{
		create: function (t, e)
		{
			return void 0 === e ? k(t) : K(k(t), e)
		},
		defineProperty: U,
		defineProperties: K,
		getOwnPropertyDescriptor: Y,
		getOwnPropertyNames: X,
		getOwnPropertySymbols: J
	}), T && s(s.S + s.F * (!R || u(function ()
	{
		var t = A();
		return "[null]" != L([t]) || "{}" != L(
		{
			a: t
		}) || "{}" != L(Object(t))
	})), "JSON",
	{
		stringify: function (t)
		{
			for (var e, n, r = [t], o = 1; arguments.length > o;) r.push(arguments[o++]);
			if (n = e = r[1], (y(e) || void 0 !== t) && !W(t)) return m(e) || (e = function (t, e)
			{
				if ("function" == typeof n && (e = n.call(this, t, e)), !W(e)) return e
			}), r[1] = e, L.apply(T, r)
		}
	}), A.prototype[M] || n(22)(A.prototype, M, A.prototype.valueOf), f(A, "Symbol"), f(Math, "Math", !0), f(r.JSON, "JSON", !0)
}, function (t, e, n)
{
	n(242), n(237), n(236), n(235), t.exports = n(11).Symbol
}, function (t, e, n)
{
	t.exports = {
		default: n(243),
		__esModule: !0
	}
}, function (t, e)
{
	t.exports = function (t, e)
	{
		return {
			value: e,
			done: !!t
		}
	}
}, function (t, e)
{
	t.exports = function () {}
}, function (t, e, n)
{
	"use strict";
	var r = n(246),
		o = n(245),
		i = n(26),
		s = n(19);
	t.exports = n(97)(Array, "Array", function (t, e)
	{
		this._t = s(t), this._i = 0, this._k = e
	}, function ()
	{
		var t = this._t,
			e = this._k,
			n = this._i++;
		return !t || n >= t.length ? (this._t = void 0, o(1)) : o(0, "keys" == e ? n : "values" == e ? t[n] : [n, t[n]])
	}, "values"), i.Arguments = i.Array, r("keys"), r("values"), r("entries")
}, function (t, e, n)
{
	n(247);
	for (var r = n(12), o = n(22), i = n(26), s = n(8)("toStringTag"), a = "CSSRuleList,CSSStyleDeclaration,CSSValueList,ClientRectList,DOMRectList,DOMStringList,DOMTokenList,DataTransferItemList,FileList,HTMLAllCollection,HTMLCollection,HTMLFormElement,HTMLSelectElement,MediaList,MimeTypeArray,NamedNodeMap,NodeList,PaintRequestList,Plugin,PluginArray,SVGLengthList,SVGNumberList,SVGPathSegList,SVGPointList,SVGStringList,SVGTransformList,SourceBufferList,StyleSheetList,TextTrackCueList,TextTrackList,TouchList".split(","), c = 0; c < a.length; c++)
	{
		var u = a[c],
			l = r[u],
			f = l && l.prototype;
		f && !f[s] && o(f, s, u), i[u] = i.Array
	}
}, function (t, e, n)
{
	var r = n(15),
		o = n(93),
		i = n(56)("IE_PROTO"),
		s = Object.prototype;
	t.exports = Object.getPrototypeOf || function (t)
	{
		return t = o(t), r(t, i) ? t[i] : "function" == typeof t.constructor && t instanceof t.constructor ? t.constructor.prototype : t instanceof Object ? s : null
	}
}, function (t, e, n)
{
	var r = n(12).document;
	t.exports = r && r.documentElement
}, function (t, e, n)
{
	var r = n(62),
		o = Math.max,
		i = Math.min;
	t.exports = function (t, e)
	{
		return (t = r(t)) < 0 ? o(t + e, 0) : i(t, e)
	}
}, function (t, e, n)
{
	var r = n(19),
		o = n(94),
		i = n(251);
	t.exports = function (t)
	{
		return function (e, n, s)
		{
			var a, c = r(e),
				u = o(c.length),
				l = i(s, u);
			if (t && n != n)
			{
				for (; u > l;)
					if ((a = c[l++]) != a) return !0
			}
			else
				for (; u > l; l++)
					if ((t || l in c) && c[l] === n) return t || l || 0;
			return !t && -1
		}
	}
}, function (t, e, n)
{
	var r = n(57);
	t.exports = Object("z").propertyIsEnumerable(0) ? Object : function (t)
	{
		return "String" == r(t) ? t.split("") : Object(t)
	}
}, function (t, e, n)
{
	var r = n(10),
		o = n(21),
		i = n(58);
	t.exports = n(16) ? Object.defineProperties : function (t, e)
	{
		o(t);
		for (var n, s = i(e), a = s.length, c = 0; a > c;) r.f(t, n = s[c++], e[n]);
		return t
	}
}, function (t, e, n)
{
	"use strict";
	var r = n(59),
		o = n(27),
		i = n(53),
		s = {};
	n(22)(s, n(8)("iterator"), function ()
	{
		return this
	}), t.exports = function (t, e, n)
	{
		t.prototype = r(s,
		{
			next: o(1, n)
		}), i(t, e + " Iterator")
	}
}, function (t, e, n)
{
	var r = n(62),
		o = n(61);
	t.exports = function (t)
	{
		return function (e, n)
		{
			var i, s, a = String(o(e)),
				c = r(n),
				u = a.length;
			return c < 0 || c >= u ? t ? "" : void 0 : (i = a.charCodeAt(c)) < 55296 || i > 56319 || c + 1 === u || (s = a.charCodeAt(c + 1)) < 56320 || s > 57343 ? t ? a.charAt(c) : i : t ? a.slice(c, c + 2) : s - 56320 + (i - 55296 << 10) + 65536
		}
	}
}, function (t, e, n)
{
	n(98), n(248), t.exports = n(52).f("iterator")
}, function (t, e, n)
{
	t.exports = {
		default: n(257),
		__esModule: !0
	}
}, function (t, e)
{
	t.exports = function (t)
	{
		if ("function" != typeof t) throw TypeError(t + " is not a function!");
		return t
	}
}, function (t, e, n)
{
	var r = n(23);
	r(r.S + r.F * !n(16), "Object",
	{
		defineProperty: n(10).f
	})
}, function (t, e, n)
{
	n(260);
	var r = n(11).Object;
	t.exports = function (t, e, n)
	{
		return r.defineProperty(t, e, n)
	}
}]);
